From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Wed, 8 Nov 2017 15:22:02 +0100
Subject: Switch FreeBSD to poll

---
 cmake/DefinePlatformSpecifc.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cmake/DefinePlatformSpecifc.cmake b/cmake/DefinePlatformSpecifc.cmake
index e6d2c68..cc2c8f6 100644
--- a/cmake/DefinePlatformSpecifc.cmake
+++ b/cmake/DefinePlatformSpecifc.cmake
@@ -91,6 +91,8 @@ else (CYGWIN)
 	    if (QNX)
 	      add_definitions( -DPOCO_HAVE_FD_POLL)
 	      set(SYSLIBS  m socket)
+            elseif (CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+	      add_definitions( -D_XOPEN_SOURCE=500 -DPOCO_HAVE_FD_POLL)
 	    else (QNX)
 	      add_definitions( -D_XOPEN_SOURCE=500 -DPOCO_HAVE_FD_EPOLL)
 	      set(SYSLIBS  pthread ${CMAKE_DL_LIBS} rt)
