From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Thu, 9 Nov 2017 22:48:30 +0100
Subject: Link against dl and rt on FreeBSD

---
 Foundation/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Foundation/CMakeLists.txt b/Foundation/CMakeLists.txt
index 4380a38..ba9864b 100644
--- a/Foundation/CMakeLists.txt
+++ b/Foundation/CMakeLists.txt
@@ -105,6 +105,10 @@ if(ANDROID)
 	set(SYSLIBS ${SYSLIBS} log)
 endif(ANDROID)
 
+if (CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+	set(SYSLIBS ${SYSLIBS} dl rt)
+endif (CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
+
 # TODO: Why is this here?
 add_definitions( -DPCRE_STATIC)
 
