Description: Fix FTBFS when DSO linking. 
Bug-Origin: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=719902
Forwarded: no
Last-Updated: 2013-12-07

Index: slang-2.3.0/slsh/Makefile.in
===================================================================
--- slang-2.3.0.orig/slsh/Makefile.in
+++ slang-2.3.0/slsh/Makefile.in
@@ -24,6 +24,8 @@ RPATH		= @RPATH@
 INSTALL		= @INSTALL@
 INSTALL_DATA	= @INSTALL_DATA@
 MKINSDIR	= $(CONFIG_DIR)/autoconf/mkinsdir.sh
+DL_LIB          = @DYNAMIC_LINK_LIB@
+ELF_DEP_LIBS    = @ELF_DEP_LIBS@
 #----------------------------------------------------------------------------
 # Where system-wide slsh.rc, scripts, and library files reside
 #----------------------------------------------------------------------------
@@ -78,9 +80,9 @@ SRC_LIBS = $(SLANG_SRCLIB) -lslang $(REA
 all: $(OBJDIR)/slsh_exe
 slsh: $(OBJDIR)/slsh
 $(OBJDIR)/slsh_exe: $(OBJDIR)/slsh.o $(OBJDIR)/readline.o
-	$(CC) $(CFLAGS) $(OBJDIR)/slsh.o $(OBJDIR)/readline.o -o $(OBJDIR)/slsh_exe $(LDFLAGS) $(SRC_LIBS)
+	$(CC) $(CFLAGS) $(OBJDIR)/slsh.o $(OBJDIR)/readline.o -o $(OBJDIR)/slsh_exe $(LDFLAGS) $(SRC_LIBS) $(ELF_DEP_LIBS)
 $(OBJDIR)/slsh: $(OBJDIR)/slsh.o $(OBJDIR)/readline.o
-	$(CC) $(CFLAGS) $(OBJDIR)/slsh.o $(OBJDIR)/readline.o -o $(OBJDIR)/slsh $(LDFLAGS) $(INST_LIBS)
+	$(CC) $(CFLAGS) $(OBJDIR)/slsh.o $(OBJDIR)/readline.o -o $(OBJDIR)/slsh $(LDFLAGS) $(INST_LIBS) $(ELF_DEP_LIBS)
 $(OBJDIR)/slsh.o: $(OBJDIR) slsh.c slsh.h config.h Makefile
 	cd $(OBJDIR) && $(CC) $(SLANG_SRCINC) $(CFLAGS) -c $(DEFS) $(SRCDIR)/slsh.c
 $(OBJDIR)/readline.o: $(OBJDIR) readline.c slsh.h config.h Makefile
