Index: slang-2.3.0/slsh/Makefile.in
===================================================================
--- slang-2.3.0.orig/slsh/Makefile.in
+++ slang-2.3.0/slsh/Makefile.in
@@ -18,7 +18,7 @@ SLANG_INST_LIB	= -L$(INST_LIB_DIR)
 SLANG_SRCINC	= -I@SRCDIR@
 SLANG_SRCLIB	= -L@ELFDIR@
 #---------------------------------------------------------------------------
-OTHER_LIBS	= @TERMCAP@ @DYNAMIC_LINK_LIB@ @LIBS@ @M_LIB@
+OTHER_LIBS	= @TERMCAP@ @LIBS@ 
 RPATH		= @RPATH@
 #----------------------------------------------------------------------------
 INSTALL		= @INSTALL@
