Description: Use /run/nologin instead of /etc/nologin
Author: Roger Leigh <rleigh@debian.org>
Bug-Debian: http://bugs.debian.org/660862

diff --git a/man/shutdown.8 b/man/shutdown.8
index 919a77e..9ce56e4 100644
--- a/man/shutdown.8
+++ b/man/shutdown.8
@@ -119,7 +119,7 @@ Second, it can be in the format \fB+\fP\fIm\fP, in which \fIm\fP is the
 number of minutes to wait.  The word \fBnow\fP is an alias for \fB+0\fP.
 .PP
 If shutdown is called with a delay, it will create the advisory file
-.I /etc/nologin
+.I /run/nologin
 which causes programs such as \fIlogin(1)\fP to not allow new user
 logins. This file is created five minutes before the shutdown sequence
 starts. Shutdown removes this file if it is stopped before it
diff --git a/src/paths.h b/src/paths.h
index 232a944..74c867e 100644
--- a/src/paths.h
+++ b/src/paths.h
@@ -28,7 +28,7 @@
 #define SDALLOW		"/etc/shutdown.allow"	/* Users allowed to shutdown */
 #define INITTAB		"/etc/inittab"		/* Location of inittab */
 #define INIT		"/sbin/init"		/* Location of init itself. */
-#define NOLOGIN		"/etc/nologin"		/* Stop user logging in. */
+#define NOLOGIN		"/run/nologin"		/* Stop user logging in. */
 #define FASTBOOT	"/fastboot"		/* Enable fast boot. */
 #define FORCEFSCK	"/forcefsck"		/* Force fsck on boot */
 #define SDPID		"/var/run/shutdown.pid"	/* PID of shutdown program */
diff --git a/src/shutdown.c b/src/shutdown.c
index 7e997da..8c3345a 100644
--- a/src/shutdown.c
+++ b/src/shutdown.c
@@ -234,7 +234,7 @@ void warn(int mins)
 }
 
 /*
- *	Create the /etc/nologin file.
+ *	Create the /run/nologin file.
  */
 void donologin(int min)
 {
