From a0c01a537e46cd9d99a3cd4aab63cc5de3db9e30 Mon Sep 17 00:00:00 2001
From: Justin Maggard <justing.maggard@netgear.com>
Date: Thu, 2 Jun 2016 17:17:32 +0000
Subject: [PATCH] increase dhcp retries

---
 gdhcp/client.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdhcp/client.c b/gdhcp/client.c
index aab7437..157e33f 100644
--- a/gdhcp/client.c
+++ b/gdhcp/client.c
@@ -48,7 +48,7 @@
 #include "common.h"
 
 #define DISCOVER_TIMEOUT 5
-#define DISCOVER_RETRIES 6
+#define DISCOVER_RETRIES 12
 
 #define REQUEST_TIMEOUT 5
 #define REQUEST_RETRIES 3
-- 
2.1.4

