From 83d04e614864c32a1f7ce4c6c6b36f72f5e41d21 Mon Sep 17 00:00:00 2001
From: Justin Maggard <justin.maggard@netgear.com>
Date: Thu, 2 Jun 2016 17:37:53 +0000
Subject: [PATCH 06/21] set default route when service disconnects

---
 src/connection.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/connection.c b/src/connection.c
index 6b005e7..2e927c0 100644
--- a/src/connection.c
+++ b/src/connection.c
@@ -915,10 +915,10 @@ void __connman_connection_gateway_remove(struct connman_service *service,
 		return;
 
 	if (do_ipv4 && data->ipv4_gateway)
-		set_default4 = data->ipv4_gateway->vpn;
+		set_default4 = true;
 
 	if (do_ipv6 && data->ipv6_gateway)
-		set_default6 = data->ipv6_gateway->vpn;
+		set_default6 = true;
 
 	DBG("ipv4 gateway %s ipv6 gateway %s vpn %d/%d",
 		data->ipv4_gateway ? data->ipv4_gateway->gateway : "<null>",
-- 
2.1.4

