From da652ed682243f8c853f3e4e20d5206d1978967a Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Mon, 14 Aug 2017 13:58:52 -0700
Subject: [PATCH 07/21] disable unused features

---
 src/main.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/src/main.c b/src/main.c
index b78a046..7763140 100644
--- a/src/main.c
+++ b/src/main.c
@@ -743,9 +743,6 @@ int main(int argc, char *argv[])
 	__connman_device_init(option_device, option_nodevice);
 
 	__connman_ippool_init();
-	__connman_firewall_init();
-	__connman_nat_init();
-	__connman_tethering_init();
 	__connman_counter_init();
 	__connman_manager_init();
 	__connman_stats_init();
@@ -766,9 +763,6 @@ int main(int argc, char *argv[])
 	__connman_rtnl_start();
 	__connman_dhcp_init();
 	__connman_dhcpv6_init();
-	__connman_wpad_init();
-	__connman_wispr_init();
-	__connman_rfkill_init();
 	__connman_machine_init();
 
 	g_free(option_config);
@@ -782,9 +776,6 @@ int main(int argc, char *argv[])
 	g_source_remove(signal);
 
 	__connman_machine_cleanup();
-	__connman_rfkill_cleanup();
-	__connman_wispr_cleanup();
-	__connman_wpad_cleanup();
 	__connman_dhcpv6_cleanup();
 	__connman_session_cleanup();
 	__connman_plugin_cleanup();
@@ -802,9 +793,6 @@ int main(int argc, char *argv[])
 	__connman_config_cleanup();
 	__connman_manager_cleanup();
 	__connman_counter_cleanup();
-	__connman_tethering_cleanup();
-	__connman_nat_cleanup();
-	__connman_firewall_cleanup();
 	__connman_peer_service_cleanup();
 	__connman_peer_cleanup();
 	__connman_ippool_cleanup();
-- 
2.1.4

