From 6a39e2309f6f0c37007ae77cc00fb2e2d19a7c14 Mon Sep 17 00:00:00 2001
From: Justin Maggard <justin.maggard@netgear.com>
Date: Thu, 2 Jun 2016 17:39:02 +0000
Subject: [PATCH] default to no backtrace and let OS core dump

---
 src/main.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/src/main.c b/src/main.c
index 3abccaa..eb50e31 100644
--- a/src/main.c
+++ b/src/main.c
@@ -579,7 +579,7 @@ static gchar *option_noplugin = NULL;
 static gchar *option_wifi = NULL;
 static gboolean option_detach = TRUE;
 static gboolean option_dnsproxy = TRUE;
-static gboolean option_backtrace = TRUE;
+static gboolean option_backtrace = FALSE;
 static gboolean option_version = FALSE;
 
 static bool parse_debug(const char *key, const char *value,
@@ -640,9 +640,8 @@ static GOptionEntry options[] = {
 	{ "nodnsproxy", 'r', G_OPTION_FLAG_REVERSE,
 				G_OPTION_ARG_NONE, &option_dnsproxy,
 				"Don't support DNS resolving" },
-	{ "nobacktrace", 0, G_OPTION_FLAG_REVERSE,
-				G_OPTION_ARG_NONE, &option_backtrace,
-				"Don't print out backtrace information" },
+	{ "backtrace", 0, 0, G_OPTION_ARG_NONE, &option_backtrace,
+				"Print out backtrace information" },
 	{ "version", 'v', 0, G_OPTION_ARG_NONE, &option_version,
 				"Show version information and exit" },
 	{ NULL },
-- 
2.1.4

