Index: interfaces.c
===================================================================
--- interfaces.c	(revision 10327)
+++ interfaces.c	(working copy)
@@ -183,6 +183,7 @@
         if (ret < 0)
         {
                 perror("couldn't bind");
+		close(s);
                 return -1;
         }
 
Index: raidard_serve.c
===================================================================
--- raidard_serve.c	(revision 10327)
+++ raidard_serve.c	(working copy)
@@ -54,7 +54,7 @@
 int ifaceCount = 0;
 int quitting;
 int netlink_fd;
-int status_fd;
+int status_fd = -1;
 
 #define DISK_BUF_SZ 2048
 static char disk_buf[DISK_BUF_SZ];
@@ -409,7 +409,7 @@
 			flags &= ~FLAG_BOOTING;
 		}
 	}
-	if (!(flags & FLAG_BOOTING) &&
+	if (!(flags & (FLAG_BOOTING | FLAG_INITRD)) &&
 	    (!info->readynasd_online || time(NULL) >= info->next_readynasd_check))
 	{
 		info->readynasd_online = systemd_is_active("readynasd");
@@ -686,7 +686,7 @@
 
 	if (daemonize)
 	{
-		int ret = daemon(0, 0);
+		int ret = daemon(0, 1);
 		if (ret != 0)
 			fprintf(stderr, "main: failed to daemonize\n");
 	}
@@ -719,6 +719,7 @@
 	if (access("/run/readynasd/raidar.fifo", R_OK) == 0 ||
 	    mkfifo("/run/readynasd/raidar.fifo", 0600) == 0)
 		status_fd = open("/run/readynasd/raidar.fifo", O_RDWR);
+printf("status_fd = %d\n", status_fd);
 
 	/* We can just pass the mcast_ip as this
 	 * argument isn't used unless we are in multicast mode
