Author: Daniel Baumann <daniel@debian.org>
Description: Print information about the CSS README.

diff -Naurp libdvdread.orig/src/dvd_input.c libdvdread/src/dvd_input.c
--- libdvdread.orig/src/dvd_input.c	2008-09-06 21:55:51.000000000 +0000
+++ libdvdread/src/dvd_input.c	2008-09-09 05:31:07.000000000 +0000
@@ -348,7 +348,14 @@ int dvdinput_setup(void)
     return 1;
 
   } else {
-    fprintf(stderr, "libdvdread: Encrypted DVD support unavailable.\n");
+    fprintf(stderr, "libdvdread: Encrypted DVD support unavailable.\n"
+	    "************************************************\n"
+	    "**                                            **\n"
+	    "**  No css library available. See             **\n"
+	    "**  /usr/share/doc/libdvdread4/README.Debian  **\n"
+	    "**  for more information.                     **\n"
+	    "**                                            **\n"
+	    "************************************************\n");
 
     /* libdvdcss replacement functions */
     dvdinput_open  = file_open;
