Author: Aurelien Jarno <aurel32@debian.org>
Description: Adds support for Debian GNU/kFreeBSD.

diff -Naurp libdvdread.orig/src/dvd_reader.c libdvdread/src/dvd_reader.c
--- libdvdread.orig/src/dvd_reader.c	2008-09-06 21:55:51.000000000 +0000
+++ libdvdread/src/dvd_reader.c	2008-09-09 05:26:20.000000000 +0000
@@ -51,7 +51,7 @@ static inline int _private_gettimeofday(
 #define lseek64 _lseeki64
 #endif
 
-#if defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__bsdi__)|| defined(__DARWIN__)
+#if defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__bsdi__) || defined(__DARWIN__) || defined(__DragonFly__) || defined(__FreeBSD_kernel__)
 #define SYS_BSD 1
 #endif
 
