Author: Daniel Baumann <daniel@debian.org>
Description:
 Fixing symbol conflicts with other MD5 implementations (Closes: #522512).

diff -Naurp libdvdread.orig/src/md5.h libdvdread/src/md5.h
--- libdvdread.orig/src/md5.h	2008-06-01 15:16:10.000000000 +0000
+++ libdvdread/src/md5.h	2009-04-05 09:42:30.000000000 +0000
@@ -21,6 +21,15 @@
 #ifndef _MD5_H
 #define _MD5_H 1
 
+/* Change function names to prevent symbol conflicts */
+#define md5_init_ctx _DVD_md5_init_ctx
+#define md5_process_block _DVD_md5_process_block
+#define md5_process_bytes _DVD_md5_process_bytes
+#define md5_finish_ctx _DVD_md5_finish_ctx
+#define md5_read_ctx _DVD_md5_read_ctx
+#define md5_stream _DVD_md5_stream
+#define md5_buffer _DVD_md5_buffer
+
 #include <stdio.h>
 
 #if defined HAVE_LIMITS_H || _LIBC
