Author: Lorenz Minder <lminder@gmx.net>
Description: Use correct endianess on big-endian machines (Closes: #531621).

diff -Naurp libdvdread.orig//src/bswap.h libdvdread/src/bswap.h
--- libdvdread.orig//src/bswap.h	2009-10-02 14:14:50.000000000 +0200
+++ libdvdread/src/bswap.h	2010-02-20 12:59:41.000000000 +0100
@@ -20,6 +20,11 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
 
+#include <endian.h>
+#if BYTE_ORDER == BIG_ENDIAN
+#define WORDS_BIGENDIAN
+#endif
+
 #include <config.h>
 
 #if defined(WORDS_BIGENDIAN)
