Author: Daniel Baumann <daniel@debian.org>
Descriptino: Bumping version to match original upstream (Closes: #532438).

diff -Naurp libdvdread.orig/src/dvd_reader.h libdvdread/src/dvd_reader.h
--- libdvdread.orig/src/dvd_reader.h	2010-04-03 12:46:05.000000000 +0200
+++ libdvdread/src/dvd_reader.h	2010-07-12 20:43:19.530887071 +0200
@@ -41,7 +41,7 @@
 /**
  * The current version.
  */
-#define DVDREAD_VERSION 904
+#define DVDREAD_VERSION 907
 
 /**
  * The length of one Logical Block of a DVD.
