Subject: support 'now' 
Origin: vendor, http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=618651
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=618651
Forwarded: no
Reviewed-by: Herbert Parentes Fortes Neto <hpfn@ig.com.br>
Last-Update: 2014-12-22

--- a/gphoto2/actions.c
+++ b/gphoto2/actions.c
@@ -2028,8 +2028,10 @@
 		int	t = -1;
 		struct tm xtm;
 
+		if (!strcasecmp (value, "now")  || !strcasecmp (value, _("now")))
+			t = time(NULL);
 #ifdef HAVE_STRPTIME
-		if (strptime (value, "%c", &xtm) || strptime (value, "%Ec", &xtm))
+		else if (strptime (value, "%c", &xtm) || strptime (value, "%Ec", &xtm))
 			t = mktime (&xtm);
 #endif
 		if (t == -1) {
