Index: rpcbind-0.2.1/src/security.c
===================================================================
--- rpcbind-0.2.1.orig/src/security.c
+++ rpcbind-0.2.1/src/security.c
@@ -53,7 +53,7 @@ int deny_severity = LIBWRAP_DENY_FACILIT
 #ifndef PORTMAP_LOG_SEVERITY
 # define PORTMAP_LOG_SEVERITY LOG_INFO
 #endif
-#ifdef __linux__
+#if defined(INET6) && !defined(IPV6PORT_RESERVED)
  #define IPV6PORT_RESERVED IPPORT_RESERVED
 #endif
 
