From 6e950ac2539710ec847c8c91015f80036e0fe901 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Tue, 23 Jun 2015 09:43:44 -0700
Subject: [PATCH 4/6] btrfs-progs: restore: add "force" option to avoid endless
 "looping" warnings.

---
 cmds-restore.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/cmds-restore.c b/cmds-restore.c
index 8fc8b2a..8d2766f 100644
--- a/cmds-restore.c
+++ b/cmds-restore.c
@@ -54,6 +54,7 @@ static int ignore_errors = 0;
 static int overwrite = 0;
 static int get_xattrs = 0;
 static int dry_run = 0;
+static int force = 0;
 
 #define LZO_LEN 4
 #define PAGE_CACHE_SIZE 4096
@@ -630,7 +631,7 @@ static int copy_file(struct btrfs_root *root, int fd, struct btrfs_key *key,
 	int ret;
 	int extent_type;
 	int compression;
-	int loops = 0;
+	int loops = force ? -1 : 0;
 	u64 found_size = 0;
 	struct timespec times[2];
 	int times_ok = 0;
@@ -1400,6 +1401,7 @@ const char * const cmd_restore_usage[] = {
 	"-d                   find dir",
 	"-l|--list-roots      list tree roots",
 	"-D|--dry-run         dry run (only list files that would be recovered)",
+	"-F                   force (do not prompt on large files looping frequently)",
 	"--path-regex <regex>",
 	"                     restore only filenames matching regex,",
 	"                     you have to use following syntax (possibly quoted):",
@@ -1444,7 +1446,7 @@ int cmd_restore(int argc, char **argv)
 			{ NULL, 0, NULL, 0}
 		};
 
-		opt = getopt_long(argc, argv, "sSxviot:u:dmf:r:lDc", long_options,
+		opt = getopt_long(argc, argv, "sSxviot:u:dmf:r:lDFc", long_options,
 					NULL);
 		if (opt < 0)
 			break;
@@ -1499,6 +1501,9 @@ int cmd_restore(int argc, char **argv)
 			case 'D':
 				dry_run = 1;
 				break;
+			case 'F':
+				force = 1;
+				break;
 			case 'c':
 				match_cflags |= REG_ICASE;
 				break;
-- 
1.9.1

