From ed1d89ba5061534dc0868d0df7ac37a85fa4a60a Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Tue, 23 Jun 2015 09:48:21 -0700
Subject: [PATCH 6/6] btrfs-progs: scrub: move scrub status files to tmpfs.

---
 cmds-scrub.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmds-scrub.c b/cmds-scrub.c
index b7aa809..578127f 100644
--- a/cmds-scrub.c
+++ b/cmds-scrub.c
@@ -48,8 +48,8 @@ static const char * const scrub_cmd_group_usage[] = {
 	NULL
 };
 
-#define SCRUB_DATA_FILE "/var/lib/btrfs/scrub.status"
-#define SCRUB_PROGRESS_SOCKET_PATH "/var/lib/btrfs/scrub.progress"
+#define SCRUB_DATA_FILE "/run/btrfs/scrub.status"
+#define SCRUB_PROGRESS_SOCKET_PATH "/run/btrfs/scrub.progress"
 #define SCRUB_FILE_VERSION_PREFIX "scrub status"
 #define SCRUB_FILE_VERSION "1"
 
-- 
1.9.1

