diff -ur gsoap-2.8.17.orig/configure.in gsoap-2.8.17/configure.in
--- gsoap-2.8.17.orig/configure.in	2013-12-18 14:58:32.000000000 +0100
+++ gsoap-2.8.17/configure.in	2014-07-12 11:32:26.172570512 +0200
@@ -309,7 +309,6 @@
     gsoap/src/Makefile \
     gsoap/wsdl/Makefile \
     gsoap/samples/Makefile \
-    gsoap/samples/autotest/Makefile \
     gsoap/samples/calc/Makefile \
     gsoap/samples/calc++/Makefile \
     gsoap/samples/chaining/Makefile \
diff -ur gsoap-2.8.17.orig/gsoap/Makefile.am gsoap-2.8.17/gsoap/Makefile.am
--- gsoap-2.8.17.orig/gsoap/Makefile.am	2013-12-18 14:58:34.000000000 +0100
+++ gsoap-2.8.17/gsoap/Makefile.am	2014-07-12 07:12:01.372435821 +0200
@@ -13,7 +13,7 @@
 AM_CFLAGS = $(SOAPCPP2_DEBUG) -D$(platform)
 
 # Install all soapcpp2 and wsdl2h files into ${prefix}/share/gsoap
-nobase_pkgdata_DATA = $(srcdir)/import/* $(srcdir)/plugin/* $(srcdir)/WS/* $(srcdir)/custom/* $(srcdir)/extras/*
+nobase_pkgdata_DATA = $(srcdir)/import/* $(srcdir)/plugin/* $(srcdir)/WS/* $(srcdir)/custom/*
 
 ## we cannot build stdsoap2.o from 2 different sources (stdsoap2.cpp and stdsoap2.c), so we need an intermediate target:
 stdsoap2_ck.c: stdsoap2.cpp
diff -ur gsoap-2.8.17.orig/gsoap/samples/Makefile.am gsoap-2.8.17/gsoap/samples/Makefile.am
--- gsoap-2.8.17.orig/gsoap/samples/Makefile.am	2013-12-18 14:58:38.000000000 +0100
+++ gsoap-2.8.17/gsoap/samples/Makefile.am	2014-07-12 06:49:57.840419333 +0200
@@ -4,7 +4,7 @@
 AUTOMAKE_OPTIONS = foreign 1.4
 
 SUBDIRS = \
-	autotest calc calc++ databinding dime dom oneway oneway++ \
+	calc calc++ databinding dime dom oneway oneway++ \
 	factory factorytest \
 	gmt googleapi hello httpcookies lu magic mashup mashup++ \
 	chaining chaining++ primes \
