diff -ur gsoap-2.8.17.orig/gsoap/doc/dom/Doxyfile gsoap-2.8.17/gsoap/doc/dom/Doxyfile
--- gsoap-2.8.17.orig/gsoap/doc/dom/Doxyfile	2013-12-18 14:58:32.000000000 +0100
+++ gsoap-2.8.17/gsoap/doc/dom/Doxyfile	2014-07-09 18:09:23.619484794 +0200
@@ -25,7 +25,7 @@
 ALWAYS_DETAILED_SEC    = NO
 INLINE_INHERITED_MEMB  = NO
 FULL_PATH_NAMES        = YES
-STRIP_FROM_PATH        = /Applications/
+STRIP_FROM_PATH        = ../..
 STRIP_FROM_INC_PATH    = 
 SHORT_NAMES            = NO
 JAVADOC_AUTOBRIEF      = NO
diff -ur gsoap-2.8.17.orig/gsoap/doc/httpda/Doxyfile gsoap-2.8.17/gsoap/doc/httpda/Doxyfile
--- gsoap-2.8.17.orig/gsoap/doc/httpda/Doxyfile	2013-12-18 14:58:32.000000000 +0100
+++ gsoap-2.8.17/gsoap/doc/httpda/Doxyfile	2014-07-09 18:09:23.623484794 +0200
@@ -25,7 +25,7 @@
 ALWAYS_DETAILED_SEC    = NO
 INLINE_INHERITED_MEMB  = NO
 FULL_PATH_NAMES        = YES
-STRIP_FROM_PATH        = /Applications/
+STRIP_FROM_PATH        = ../..
 STRIP_FROM_INC_PATH    = 
 SHORT_NAMES            = NO
 JAVADOC_AUTOBRIEF      = NO
diff -ur gsoap-2.8.17.orig/gsoap/doc/uddi2/Doxyfile gsoap-2.8.17/gsoap/doc/uddi2/Doxyfile
--- gsoap-2.8.17.orig/gsoap/doc/uddi2/Doxyfile	2013-12-18 14:58:32.000000000 +0100
+++ gsoap-2.8.17/gsoap/doc/uddi2/Doxyfile	2014-07-09 18:09:23.623484794 +0200
@@ -25,7 +25,7 @@
 ALWAYS_DETAILED_SEC    = NO
 INLINE_INHERITED_MEMB  = NO
 FULL_PATH_NAMES        = YES
-STRIP_FROM_PATH        = /Applications/
+STRIP_FROM_PATH        = ../..
 STRIP_FROM_INC_PATH    = 
 SHORT_NAMES            = NO
 JAVADOC_AUTOBRIEF      = NO
diff -ur gsoap-2.8.17.orig/gsoap/doc/wsa/Doxyfile gsoap-2.8.17/gsoap/doc/wsa/Doxyfile
--- gsoap-2.8.17.orig/gsoap/doc/wsa/Doxyfile	2013-12-18 14:58:32.000000000 +0100
+++ gsoap-2.8.17/gsoap/doc/wsa/Doxyfile	2014-07-09 18:09:23.627484794 +0200
@@ -25,7 +25,7 @@
 ALWAYS_DETAILED_SEC    = NO
 INLINE_INHERITED_MEMB  = NO
 FULL_PATH_NAMES        = YES
-STRIP_FROM_PATH        = /Applications/
+STRIP_FROM_PATH        = ../..
 STRIP_FROM_INC_PATH    = 
 SHORT_NAMES            = NO
 JAVADOC_AUTOBRIEF      = NO
diff -ur gsoap-2.8.17.orig/gsoap/doc/wsdd/Doxyfile gsoap-2.8.17/gsoap/doc/wsdd/Doxyfile
--- gsoap-2.8.17.orig/gsoap/doc/wsdd/Doxyfile	2013-12-18 14:58:32.000000000 +0100
+++ gsoap-2.8.17/gsoap/doc/wsdd/Doxyfile	2014-07-09 18:09:23.627484794 +0200
@@ -25,7 +25,7 @@
 ALWAYS_DETAILED_SEC    = NO
 INLINE_INHERITED_MEMB  = NO
 FULL_PATH_NAMES        = YES
-STRIP_FROM_PATH        = /Applications/
+STRIP_FROM_PATH        = ../..
 STRIP_FROM_INC_PATH    = 
 SHORT_NAMES            = NO
 JAVADOC_AUTOBRIEF      = NO
diff -ur gsoap-2.8.17.orig/gsoap/doc/wsrm/Doxyfile gsoap-2.8.17/gsoap/doc/wsrm/Doxyfile
--- gsoap-2.8.17.orig/gsoap/doc/wsrm/Doxyfile	2013-12-18 14:58:32.000000000 +0100
+++ gsoap-2.8.17/gsoap/doc/wsrm/Doxyfile	2014-07-09 18:09:23.627484794 +0200
@@ -25,7 +25,7 @@
 ALWAYS_DETAILED_SEC    = NO
 INLINE_INHERITED_MEMB  = NO
 FULL_PATH_NAMES        = YES
-STRIP_FROM_PATH        = /Applications/
+STRIP_FROM_PATH        = ../..
 STRIP_FROM_INC_PATH    = 
 SHORT_NAMES            = NO
 JAVADOC_AUTOBRIEF      = NO
diff -ur gsoap-2.8.17.orig/gsoap/doc/wsse/Doxyfile gsoap-2.8.17/gsoap/doc/wsse/Doxyfile
--- gsoap-2.8.17.orig/gsoap/doc/wsse/Doxyfile	2013-12-18 14:58:32.000000000 +0100
+++ gsoap-2.8.17/gsoap/doc/wsse/Doxyfile	2014-07-09 18:09:23.639484793 +0200
@@ -25,7 +25,7 @@
 ALWAYS_DETAILED_SEC    = NO
 INLINE_INHERITED_MEMB  = NO
 FULL_PATH_NAMES        = YES
-STRIP_FROM_PATH        = /Applications/
+STRIP_FROM_PATH        = ../..
 STRIP_FROM_INC_PATH    = 
 SHORT_NAMES            = NO
 JAVADOC_AUTOBRIEF      = NO
diff -ur gsoap-2.8.17.orig/gsoap/doc/xml-rpc-json/Doxyfile gsoap-2.8.17/gsoap/doc/xml-rpc-json/Doxyfile
--- gsoap-2.8.17.orig/gsoap/doc/xml-rpc-json/Doxyfile	2013-12-18 14:58:32.000000000 +0100
+++ gsoap-2.8.17/gsoap/doc/xml-rpc-json/Doxyfile	2014-07-09 18:09:23.639484793 +0200
@@ -25,7 +25,7 @@
 ALWAYS_DETAILED_SEC    = NO
 INLINE_INHERITED_MEMB  = NO
 FULL_PATH_NAMES        = YES
-STRIP_FROM_PATH        = /Applications/
+STRIP_FROM_PATH        = ../..
 STRIP_FROM_INC_PATH    = 
 SHORT_NAMES            = NO
 JAVADOC_AUTOBRIEF      = NO
