diff -ur gsoap-2.8.17.orig/gsoap/samples/chaining/Makefile.am gsoap-2.8.17/gsoap/samples/chaining/Makefile.am
--- gsoap-2.8.17.orig/gsoap/samples/chaining/Makefile.am	2013-12-18 14:58:36.000000000 +0100
+++ gsoap-2.8.17/gsoap/samples/chaining/Makefile.am	2014-07-09 13:37:44.051096990 +0200
@@ -17,7 +17,7 @@
 
 BUILT_SOURCES = qServerLib.c cServerLib.c envC.c
 
-CLEANFILES = *.req.xml *.res.xml *.nsmap *.xsd *.wsdl envStub.h envH.h envC.c cStub.h cH.h cC.c Calc*Proxy.h Calc*Proxy.c Calc*Object.h Calc*Service.h Calc*Service.c cClient.c cServer.c cClientLib.c cServerLib.c qStub.h qH.h qC.c Quote*Proxy.h Quote*Proxy.c Quote*Object.h Quote*Service.h Quote*Service.c QuoteClient.c QuoteServer.c qClientLib.c qServerLib.c *~
+CLEANFILES = *.req.xml *.res.xml *.nsmap *.xsd *.wsdl envStub.h envH.h envC.c cStub.h cH.h cC.c cClient.c cServer.c cClientLib.c cServerLib.c qStub.h qH.h qC.c qClient.c qServer.c qClientLib.c qServerLib.c *~
 
 SOAP_NS = c.nsmap
 
diff -ur gsoap-2.8.17.orig/gsoap/samples/chaining++/Makefile.am gsoap-2.8.17/gsoap/samples/chaining++/Makefile.am
--- gsoap-2.8.17.orig/gsoap/samples/chaining++/Makefile.am	2013-12-18 14:58:36.000000000 +0100
+++ gsoap-2.8.17/gsoap/samples/chaining++/Makefile.am	2014-07-09 13:32:47.507096425 +0200
@@ -16,7 +16,7 @@
 
 BUILT_SOURCES = QuotequoteService.cpp CalccalcService.cpp envC.cpp
 
-CLEANFILES = *.req.xml *.res.xml *.nsmap *.xsd *.wsdl envStub.h envH.h envC.cpp calcStub.h calcH.h calcC.cpp Calc*Proxy.h Calc*Proxy.cpp Calc*Object.h Calc*Service.h Calc*Service.cpp calcClient.cpp calcServer.cpp calcClientLib.cpp calcServerLib.cpp quoteStub.h quoteH.h quoteC.cpp Quote*Proxy.h Quote*Proxy.cpp Quote*Object.h Quote*Service.h Quote*Service.cpp QuoteClient.cpp QuoteServer.cpp quoteClientLib.cpp quoteServerLib.cpp *~
+CLEANFILES = *.req.xml *.res.xml *.nsmap *.xsd *.wsdl envStub.h envH.h envC.cpp CalcStub.h CalcH.h CalcC.cpp Calc*Proxy.h Calc*Proxy.cpp Calc*Object.h Calc*Service.h Calc*Service.cpp CalcClient.cpp CalcServer.cpp CalcClientLib.cpp CalcServerLib.cpp QuoteStub.h QuoteH.h QuoteC.cpp Quote*Proxy.h Quote*Proxy.cpp Quote*Object.h Quote*Service.h Quote*Service.cpp QuoteClient.cpp QuoteServer.cpp QuoteClientLib.cpp QuoteServerLib.cpp *~
 
 SOAP_NS = Calc.nsmap
 
diff -ur gsoap-2.8.17.orig/gsoap/samples/databinding/Makefile.am gsoap-2.8.17/gsoap/samples/databinding/Makefile.am
--- gsoap-2.8.17.orig/gsoap/samples/databinding/Makefile.am	2013-12-18 14:58:36.000000000 +0100
+++ gsoap-2.8.17/gsoap/samples/databinding/Makefile.am	2014-07-09 14:09:27.207147420 +0200
@@ -30,4 +30,4 @@
 address_SOURCES=$(SOAP_CPP_SRC) address.cpp
 address_LDADD=$(SOAP_CPP_LIB)
 
-CLEANFILES= *~ *C.cpp *H.h *Stub.h *.nsmap
+CLEANFILES= *~ address.h *C.cpp *H.h *Stub.h *.nsmap
