diff -ur gsoap-2.8.orig/gsoap/stdsoap2.c gsoap-2.8/gsoap/stdsoap2.c
--- gsoap-2.8.orig/gsoap/stdsoap2.c	2017-07-11 03:51:16.000000000 +0200
+++ gsoap-2.8/gsoap/stdsoap2.c	2018-04-18 16:09:06.340071192 +0200
@@ -16965,7 +16965,6 @@
       return soap->error = SOAP_CHK_EOF;
     soap_unget(soap, soap_getchar(soap)); /* skip padding and get hdr */
     DBGLOG(TEST, SOAP_MESSAGE(fdebug, "... From chunked\n"));
-    return SOAP_OK;
   }
   s = (char*)tmp;
   for (i = 12; i > 0; i--)
diff -ur gsoap-2.8.orig/gsoap/stdsoap2.cpp gsoap-2.8/gsoap/stdsoap2.cpp
--- gsoap-2.8.orig/gsoap/stdsoap2.cpp	2017-07-11 03:51:16.000000000 +0200
+++ gsoap-2.8/gsoap/stdsoap2.cpp	2018-04-18 16:09:06.340071192 +0200
@@ -16965,7 +16965,6 @@
       return soap->error = SOAP_CHK_EOF;
     soap_unget(soap, soap_getchar(soap)); /* skip padding and get hdr */
     DBGLOG(TEST, SOAP_MESSAGE(fdebug, "... From chunked\n"));
-    return SOAP_OK;
   }
   s = (char*)tmp;
   for (i = 12; i > 0; i--)
