--- initscript/debian/avahi-daemon.in.old	2006-06-28 00:49:27.989393500 +0200
+++ initscript/debian/avahi-daemon.in	2006-06-28 01:11:54.653554750 +0200
@@ -87,6 +87,14 @@
 # Gracefully exit if the package has been removed.
 test -x $DAEMON || exit 0
 
+# don't start if /etc/avahi-daemon says so.
+AVAHI_DAEMON_START=1
+test -f /etc/default/avahi-daemon && . /etc/default/avahi-daemon
+
+if [ "$AVAHI_DAEMON_START" != "1" -a "$1" != "stop" ]; then
+	exit 0
+fi
+
 #
 #       Function that starts the daemon/service.
 #
--- initscript/debian/avahi-dnsconfd.in.old	2006-06-28 01:10:09.702995750 +0200
+++ initscript/debian/avahi-dnsconfd.in	2006-06-28 01:11:30.364036750 +0200
@@ -87,6 +87,14 @@
 # Gracefully exit if the package has been removed.
 test -x $DAEMON || exit 0
 
+# don't start if /etc/default/avahi-dnsconfd says so. 
+AVAHI_DNSCONFD_START=1
+test -f /etc/default/avahi-dnsconfd && . /etc/default/avahi-dnsconfd
+
+if [ "$AVAHI_DNSCONFD_START" != "1" -a "$1" != "stop" ]; then
+    exit 0
+fi
+
 #
 #       Function that starts the daemon/service.
 #
