--- initscript/debian/avahi-daemon.in.old	2006-10-17 18:58:43.005681000 +0200
+++ initscript/debian/avahi-daemon.in	2006-10-17 18:58:59.382704500 +0200
@@ -83,6 +83,7 @@
 DESC="Avahi mDNS/DNS-SD Daemon"
 NAME="avahi-daemon"
 DAEMON="@sbindir@/$NAME"
+SCRIPTNAME=/etc/init.d/$NAME
 
 # Gracefully exit if the package has been removed.
 test -x $DAEMON || exit 0
--- initscript/debian/avahi-dnsconfd.in.old	2006-10-17 18:58:47.705974750 +0200
+++ initscript/debian/avahi-dnsconfd.in	2006-10-17 18:58:53.890361250 +0200
@@ -83,6 +83,7 @@
 DESC="Avahi Unicast DNS Configuration Daemon"
 NAME="avahi-dnsconfd"
 DAEMON="@sbindir@/$NAME"
+SCRIPTNAME=/etc/init.d/$NAME
 
 # Gracefully exit if the package has been removed.
 test -x $DAEMON || exit 0
