Author: gregor herrmann <gregoa@debian.org>
Description: fix some minor spelling mistakes
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=59513

--- libdbi-perl.orig/lib/DBI/FAQ.pm
+++ libdbi-perl/lib/DBI/FAQ.pm
@@ -66,7 +66,7 @@
     ``DBI is a database access Application Programming Interface (API)
       for the Perl Language. The DBI API Specification defines a set
       of functions, variables and conventions that provide a consistent
-      database interface independant of the actual database being used.''
+      database interface independent of the actual database being used.''
 
 In simple language, the DBI interface allows users to access multiple database
 types transparently. So, if you connecting to an Oracle, Informix, mSQL, Sybase
--- libdbi-perl.orig/lib/DBI/ProxyServer.pm
+++ libdbi-perl/lib/DBI/ProxyServer.pm
@@ -651,7 +651,7 @@
 		},
 
 		# rule: employee_workplace
-		# desc: get detailled informations
+		# desc: get detailled information
 		{
 			# any IP-address is meant here
 			mask => '^10\.95\.81\.(\d+)$',
