diff --git a/main.cc- b/main.cc
index f906c8a..99c91ef 100644
--- a/main.cc-
+++ b/main.cc
@@ -369,7 +369,7 @@ int do_rescue( const long long offset, Domain & domain,
     { show_error( "Output file is not seekable." ); return 1; }
   while( preallocate )
     {
-#if defined _POSIX_ADVISORY_INFO && _POSIX_ADVISORY_INFO > 0
+#if 0 && defined _POSIX_ADVISORY_INFO && _POSIX_ADVISORY_INFO > 0
     if( posix_fallocate( odes, rescuebook.domain().pos() + rescuebook.offset(),
                          rescuebook.domain().size() ) == 0 ) break;
     if( errno != EINTR )
