Description: adjust makefile for hardening support
Author: Petter Reinholdtsen <pere@debian.org>
Forwarded: no
Reviewed-By: Petter Reinholdtsen <pere@hungry.com>
Last-Update: 2014-04-15
Status:  Applied upstream.

Index: startpar-git/Makefile
===================================================================
--- startpar-git.orig/Makefile	2014-04-15 12:03:41.733225880 +0200
+++ startpar-git/Makefile	2014-04-15 12:04:50.729746235 +0200
@@ -26,16 +26,20 @@
 # Enable large file support on GNU/Hurd
 CFLAGS += $(shell getconf LFS_CFLAGS)
 
+CPPFLAGS += $(EXTRACPPFLAGS)
+CFLAGS += $(EXTRACFLAGS)
+LDFLAGS += $(EXTRALDFLAGS)
+
 ifeq ($(MAKECMDGOALS),makeboot)
 CFLAGS += -DTEST
 endif
 
 .c.o:
-	$(CC) $(CFLAGS) -DVERSION=\"$(VERSION)\" $(ISSUSE) -c $<
+	$(CC) $(CPPFLAGS) $(CFLAGS) -DVERSION=\"$(VERSION)\" $(ISSUSE) -c $<
 
 all: startpar
 startpar: $(OBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -DVERSION=\"$(VERSION)\" $(ISSUSE) -o $@ $(OBJS) $(LIBS)
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -DVERSION=\"$(VERSION)\" $(ISSUSE) -o $@ $(OBJS) $(LIBS)
 
 makeboot: makeboot.c
 
