From 8e0a01e8afc6aeba59c6a5b3749b8e452a2b57bb Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard10@gmail.com>
Date: Tue, 26 Aug 2014 11:10:41 -0700
Subject: [PATCH] fsck: remove init-csum-tree from help text

---
 cmds-check.c |    1 -
 1 file changed, 1 deletion(-)

diff --git a/cmds-check.c b/cmds-check.c
index 05fe4a5..6b75367 100644
--- a/cmds-check.c
+++ b/cmds-check.c
@@ -6565,7 +6565,6 @@ const char * const cmd_check_usage[] = {
 	"-s|--super <superblock>     use this superblock copy",
 	"-b|--backup                 use the backup root copy",
 	"--repair                    try to repair the filesystem",
-	"--init-csum-tree            create a new CRC tree",
 	"--init-extent-tree          create a new extent tree",
 	"--check-data-csum           verify checkums of data blocks",
 	"--qgroup-report             print a report on qgroup consistency",
-- 
1.7.10.4

