--- freetype-2.1.7~/src/truetype/ttgload.c	Mon Sep 16 17:22:03 2002
+++ freetype-2.1.7/src/truetype/ttgload.c	Sat Mar  1 00:17:02 2002
@@ -138,7 +138,12 @@
     TT_Get_Metrics( &face->horizontal, idx, lsb, aw );
 
     if ( check && face->postscript.isFixedPitch )
-      *aw = face->horizontal.advance_Width_Max;
+    {
+      if ( *aw > face->horizontal.advance_Width_Max * 0.55 || *aw == 0 )
+	*aw = face->horizontal.advance_Width_Max;
+      else
+	*aw = face->horizontal.advance_Width_Max >> 1;
+    }
   }
 
 
