diff -ru /home/dchest/ft-clean/freetype2/src/pshinter/pshalgo3.c ./src/pshinter/pshalgo3.c
--- /home/dchest/ft-clean/freetype2/src/pshinter/pshalgo3.c	2003-01-30 18:24:18.000000000 -0500
+++ ./src/pshinter/pshalgo3.c	2003-02-01 21:54:36.000000000 -0500
@@ -1908,6 +1908,34 @@
     if ( error )
       goto Exit;

+   /* try to optimize the y_scale so that the top of non-capital letters
+    * is aligned on a pixel boundary whenever possible
+    */
+    {
+      PSH_Dimension  dim_x = &glyph->globals->dimension[0];
+      PSH_Dimension  dim_y = &glyph->globals->dimension[1];
+
+      FT_Fixed x_scale = dim_x->scale_mult;
+      FT_Fixed y_scale = dim_y->scale_mult;
+
+      FT_Fixed scaled;
+      FT_Fixed fitted;
+
+      scaled = FT_MulFix( globals->blues.normal_top.zones->org_ref, y_scale );
+      fitted = (scaled + 32) & -64;
+
+      if (scaled != fitted ) {
+
+        y_scale = FT_MulDiv( y_scale, fitted, scaled );
+
+        if ( fitted < scaled )
+          x_scale -= x_scale/50;
+
+        psh_globals_set_scale( glyph->globals, x_scale, y_scale, 0, 0);
+
+      }
+    }
+
     glyph->do_horz_hints = 1;
     glyph->do_vert_hints = 1;

