--- freetype-2.1.4~/include/freetype/config/ftoption.h	2003-05-21 02:24:06.000000000 +0800
+++ freetype-2.1.4/include/freetype/config/ftoption.h	2003-05-24 20:40:55.000000000 +0800
@@ -423,7 +423,7 @@
   /* For this to work you must also define                                 */
   /* TT_CONFIG_OPTION_COMPILE_UNPATENTED_HINTING.                          */
   /*                                                                       */
-/* #define TT_CONFIG_OPTION_FORCE_UNPATENTED_HINTING */
+#define TT_CONFIG_OPTION_FORCE_UNPATENTED_HINTING
 
 
   /*************************************************************************/
