--- freetype-2.1.4rc2~/src/base/ftobjs.c	2003-03-22 19:49:39.000000000 +0800
+++ freetype-2.1.4rc2/src/base/ftobjs.c	2003-03-22 21:24:23.000000000 +0800
@@ -1426,19 +1426,11 @@
       char_height = 1 * 64;
 
     /* Compute pixel sizes in 26.6 units */
-#ifdef FT_CONFIG_CHESTER_BLUE_SCALE
-    dim_x = ( char_width  * horz_resolution + 36 ) / 72; /* round div */
-    dim_y = ( char_height * vert_resolution + 36 ) / 72; /* round div */
-
-    metrics->x_ppem = (FT_UShort)( (dim_x+32) >> 6 ); /* round div */
-    metrics->y_ppem = (FT_UShort)( (dim_y+32) >> 6 ); /* round div */
-#else
     dim_x = ( ( char_width  * horz_resolution + (36+32*72) ) / 72 ) & -64; /* round div */
     dim_y = ( ( char_height * vert_resolution + (36+32*72) ) / 72 ) & -64; /* round div */
 
     metrics->x_ppem  = (FT_UShort)( dim_x >> 6 ); /* floor div, already rounded */
     metrics->y_ppem  = (FT_UShort)( dim_y >> 6 ); /* floor div, already rounded */
-#endif
 
     metrics->x_scale = 0x10000L;
     metrics->y_scale = 0x10000L;
