--- ft2demos-2.1.7~/graph/x11/rules.mk	2003-08-28 01:11:31.000000000 +0800
+++ ft2demos-2.1.7/graph/x11/rules.mk	2003-08-28 01:50:19.000000000 +0800
@@ -67,9 +67,11 @@
   # The GRAPH_LINK variable is expanded each time an executable is linked
   # against the graphics library.
   #
-  ifeq ($(PLATFORM),unix)
-    GRAPH_LINK += $(X11_LIB:%=-R%)
-  endif
+  # No, we do not want rpath on Debian; reversing YAMANO-UCHI Hidetoshi's
+  # 2003-06-13 change.  (Anthony Fok, 2003-08-28)
+  #ifeq ($(PLATFORM),unix)
+  #  GRAPH_LINK += $(X11_LIB:%=-R%)
+  #endif
   GRAPH_LINK += $(X11_LIB:%=-L%) -lX11
 
   # Solaris needs a -lsocket in GRAPH_LINK.
