diff -ur freetype-2.1.7/src/base/ftapi.c freetype-2.1.7.real/src/base/ftapi.c
--- freetype-2.1.7/src/base/ftapi.c-	2002-03-30 13:16:33.000000000 +0000
+++ freetype-2.1.7/src/base/ftapi.c	2004-07-28 15:20:23.000000000 +0100
@@ -118,4 +118,18 @@
   }
                    
 
+  FT_BASE_DEF( FT_Short )
+  FT_Get_Short( FT_Stream stream )
+  {
+    return FT_GET_SHORT();
+  }
+
+
+  FT_BASE_DEF( FT_Long )
+  FT_Get_Long( FT_Stream stream )
+  {
+    return FT_GET_LONG();
+  }
+
+
 /* END */
diff -ur freetype-2.1.7/src/base/ftbase.c freetype-2.1.7.real/src/base/ftbase.c
--- freetype-2.1.7/src/base/ftbase.c-	2003-05-11 08:12:26.000000000 +0100
+++ freetype-2.1.7/src/base/ftbase.c	2004-07-28 15:20:49.000000000 +0100
@@ -29,6 +29,7 @@
 #include "ftgloadr.c"
 #include "ftobjs.c"
 #include "ftnames.c"
+#include "ftapi.c"
 
 #if defined( __APPLE__ ) && !defined ( DARWIN_NO_CARBON )
 #include "ftmac.c"
