--- freetype-2.1.4/src/smooth/ftgrays.c.gamma	2003-05-19 12:25:51.000000000 +0800
+++ freetype-2.1.4/src/smooth/ftgrays.c	2003-05-19 13:16:28.000000000 +0800
@@ -83,7 +83,11 @@
 
 
 /* experimental support for gamma correction within the rasterizer */
-#define xxxGRAYS_USE_GAMMA
+/* Modify by Firefly (firefly@firefly.idv.tw)
+ * 使用 gamma table 讓筆劃清晰一些, 清晰度視顯示器而定
+ * 一般而言, 對筆劃複雜的 CJK 字型, 效果非常之好
+ */
+#define GRAYS_USE_GAMMA
 
 
   /*************************************************************************/
@@ -2040,7 +2044,10 @@
   /* but the results look pretty good for something that simple.         */
   /*                                                                     */
 #define M_MAX  255
-#define M_X    128
+/* Modify by Firefly (firefly@firefly.idv.tw)
+ * 純粹是吹毛求疵的修改, 讓筆劃更深一些 :-)
+ */
+#define M_X    112
 #define M_Y    192
 
   static void
