--- freetype-2.1.7/src/bdf/bdflib.c.old	2005-04-24 11:58:44.000000000 +0000
+++ freetype-2.1.7/src/bdf/bdflib.c	2005-04-24 11:56:37.000000000 +0000
@@ -1570,6 +1570,14 @@
         goto Exit;
       p->glyph_enc = _bdf_atol( p->list.field[1], 0, 10 );
 
+      /* Check that the encoding is in the range [0, 65535] because       */
+      /* otherwise p->have (a bitmap with static size) overflows.         */
+      if ( p->glyph_enc >= sizeof(p->have) * 8 )
+      {
+	error = BDF_Err_Invalid_File_Format; /* Not the ideal error code */
+	goto Exit;
+      }
+
       /* Check to see whether this encoding has already been encountered. */
       /* If it has then change it to unencoded so it gets added if        */
       /* indicated.                                                       */
