Index: alsa-lib-1.0.25/src/conf/cards/Makefile.am
===================================================================
--- alsa-lib-1.0.25.orig/src/conf/cards/Makefile.am	2012-02-15 16:19:51.444263639 +1100
+++ alsa-lib-1.0.25/src/conf/cards/Makefile.am	2012-02-15 16:20:48.900265439 +1100
@@ -45,6 +45,7 @@
 	RME9652.conf \
 	SI7018.conf \
 	SB-XFi.conf \
+	tegra-alc5632.conf \
 	TRID4DWAVENX.conf \
 	USB-Audio.conf \
 	YMF744.conf \
Index: alsa-lib-1.0.25/src/conf/cards/tegra-alc5632.conf
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ alsa-lib-1.0.25/src/conf/cards/tegra-alc5632.conf	2012-02-15 16:21:06.228265983 +1100
@@ -0,0 +1,23 @@
+# default with dmix/dsnoop
+tegra-alc5632.pcm.default {
+       @args [ CARD ]
+       @args.CARD {
+               type string
+       }
+       type asym
+       playback.pcm {
+               type plug
+               slave.pcm {
+                       @func concat
+                       strings [ "dmix:CARD=" $CARD ",RATE=44100" ]
+               }
+       }
+       capture.pcm {
+               type plug
+               slave.pcm {
+                       @func concat
+                       strings [ "dsnoop:CARD=" $CARD ",RATE=44100" ]
+               }
+       }
+}
+
