--- a/freedesktop.org.xml.in
+++ b/freedesktop.org.xml.in
@@ -2912,6 +2912,10 @@
     </magic>
     <glob pattern="*.psw"/>
   </mime-type>
+  <mime-type type="application/x-mps">
+    <_comment>Linear and integer program expression format</_comment>
+    <glob pattern="*.mps"/>
+  </mime-type>
   <mime-type type="application/x-profile">
     <_comment>profiler results</_comment>
     <sub-class-of type="text/plain"/>
