From: Bryan Liang <bliang@netgear.com>
Date: Mon, 27 Aug 2018 15:47:57 -0700
Subject: Disable SHA1 warning.

Signed-off-by: Bryan Liang <bliang@netgear.com>
---
 methods/gpgv.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/methods/gpgv.cc b/methods/gpgv.cc
index 3f16ac0..94adec7 100644
--- a/methods/gpgv.cc
+++ b/methods/gpgv.cc
@@ -66,7 +66,7 @@ struct Digest {
 static constexpr Digest Digests[] = {
    {Digest::State::Untrusted, "Invalid digest"},
    {Digest::State::Untrusted, "MD5"},
-   {Digest::State::Weak, "SHA1"},
+   {Digest::State::Trusted, "SHA1"},
    {Digest::State::Weak, "RIPE-MD/160"},
    {Digest::State::Trusted, "Reserved digest"},
    {Digest::State::Trusted, "Reserved digest"},
