diff --git a/programs/Makefile b/programs/Makefile
index f422902..883e323 100644
--- a/programs/Makefile
+++ b/programs/Makefile
@@ -141,9 +141,9 @@ uninstall:
 	rm -f $(DESTDIR)$(MANDIR)/unlz4.1
 	@echo lz4 programs successfully uninstalled
 
-test: test-lz4 test-lz4c test-frametest test-fullbench test-fuzzer test-mem
+test: test-lz4 test-lz4c test-frametest test-fullbench test-fuzzer
 
-test32: test-lz4c32 test-frametest32 test-fullbench32 test-fuzzer32 test-mem32
+test32: test-lz4c32 test-frametest32 test-fullbench32 test-fuzzer32
 
 test-all: test test32
 
