--- gdisk-1.0.1.orig/diskio-unix.cc
+++ gdisk-1.0.1/diskio-unix.cc
@@ -249,8 +249,8 @@ int DiskIO::DiskSync(void) {
       platformFound++;
 #endif
 #ifdef __linux__
-      sleep(1); // Theoretically unnecessary, but ioctl() fails sometimes if omitted....
       fsync(fd);
+      ioctl(fd, BLKFLSBUF);
       i = ioctl(fd, BLKRRPART);
       if (i) {
          cout << "Warning: The kernel is still using the old partition table.\n"
