From aa7ed748c9765e73d41e92713e2e196847d3da08 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Wed, 16 Sep 2015 13:21:34 -0700
Subject: [PATCH 4/8] Allow snapshot access to everyone if "guest" is in the
 allowed users list.

---
 server/Client.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/server/Client.cc b/server/Client.cc
index 8521a44..db45c38 100644
--- a/server/Client.cc
+++ b/server/Client.cc
@@ -387,6 +387,10 @@ Client::check_permission(DBus::Connection& conn, DBus::Message& msg,
     if (uid == 0)
 	return;
 
+    // Check if guest is allowed
+    if (contains(meta_snapper.uids, 99))
+	return;
+
     // Check if the uid of the dbus-user is included in the allowed uids.
     if (contains(meta_snapper.uids, uid))
 	return;
-- 
1.9.1

