From 1e30caa080b32b5601afbfde1e3778f655091980 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Wed, 16 Sep 2015 13:46:22 -0700
Subject: [PATCH 8/8] Don't fail snapper config creation if the snapshot subvol
 already exists.

---
 snapper/BtrfsUtils.cc | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/snapper/BtrfsUtils.cc b/snapper/BtrfsUtils.cc
index 7dd0e79..626deb2 100644
--- a/snapper/BtrfsUtils.cc
+++ b/snapper/BtrfsUtils.cc
@@ -107,7 +107,15 @@ namespace snapper
 	strncpy(args.name, name.c_str(), sizeof(args.name) - 1);
 
 	if (ioctl(fddst, BTRFS_IOC_SUBVOL_CREATE, &args) != 0)
+	{
+	    if (errno == EEXIST)
+	    {
+		struct stat x;
+		if (fstatat(fddst, args.name, &x, 0) == 0 && is_subvolume(x))
+		    return;
+	    }
 	    throw runtime_error_with_errno("ioctl(BTRFS_IOC_SUBVOL_CREATE) failed", errno);
+	}
     }
 
 
-- 
1.9.1

