--- snapper-0.2.4.orig/snapper/File.cc
+++ snapper-0.2.4/snapper/File.cc
@@ -392,7 +392,7 @@ namespace snapper
 	bool ret = clonefile(src_fd, dest_fd) || copyfile(src_fd, dest_fd);
 	if (!ret)
 	{
-	    y2err("clone and copy failed " << getAbsolutePath(LOC_SYSTEM));
+	    y2err("clone and copy failed " << getAbsolutePath(LOC_PRE) << " => " << getAbsolutePath(LOC_SYSTEM));
 	}
 
 	close(dest_fd);
--- snapper-0.2.4.orig/snapper/AppUtil.cc
+++ snapper-0.2.4/snapper/AppUtil.cc
@@ -80,8 +80,7 @@ namespace snapper
 	int r1 = ioctl(dest_fd, BTRFS_IOC_CLONE, src_fd);
 	if (r1 != 0)
 	{
-	    // TODO: too much logging with LVM
-	    // y2err("ioctl failed errno:" << errno << " (" << stringerror(errno) << ")");
+	    y2err("ioctl failed errno:" << errno << " (" << stringerror(errno) << ")");
 	}
 
 	return r1 == 0;
