diff --git a/pam/Makefile.in- b/pam/Makefile.in
index d09c9c7..03f0163 100644
--- a/pam/Makefile.in-
+++ b/pam/Makefile.in
@@ -340,7 +340,7 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 @HAVE_PAM_TRUE@AM_CFLAGS = -D_GNU_SOURCE -Wwrite-strings
 @HAVE_PAM_TRUE@AM_CPPFLAGS = -I$(top_srcdir) $(DBUS_CFLAGS)
-@HAVE_PAM_TRUE@securelibdir = $(shell echo /`basename $(libdir)`/security)
+@HAVE_PAM_TRUE@securelibdir = $(shell echo /lib/`basename $(libdir)`/security)
 @HAVE_PAM_TRUE@securelib_LTLIBRARIES = pam_snapper.la
 @HAVE_PAM_TRUE@pam_snapper_la_LDFLAGS = -no-undefined -avoid-version -module
 @HAVE_PAM_TRUE@pam_snapper_la_LIBADD = -lpam $(DBUS_LIBS)
