Description: Disable in-place editing in Makefile.PL
Author: Colin Watson <cjwatson@debian.org>
Forwarded: no
Last-Update: 2010-05-26

Index: b/Makefile.PL
===================================================================
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -95,7 +95,7 @@
 
 sub oldWarnings
 {
-    local ($^I) = ".bak" ;
+    local ($^I) = "" ;
     local (@ARGV) = @_ ;
 
     while (<>)
@@ -119,7 +119,7 @@
 
 sub newWarnings
 {
-    local ($^I) = ".bak" ;
+    local ($^I) = "" ;
     local (@ARGV) = @_ ;
 
     while (<>)
