Description: Install perlfilter manual page in section 1, not 3
Author: Colin Watson <cjwatson@debian.org>
Forwarded: no
Last-Update: 2010-05-26

Index: b/Makefile.PL
===================================================================
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -48,6 +48,21 @@
  { newWarnings(@files) }     
              
 
+{
+    package MY;
+
+    # Fun hack by cjwatson to get perlfilter.pod manified in section 1 with
+    # all the other perl*.pod pages.
+    sub constants {
+	my ($self) = @_;
+	$self->{MAN1PODS}->{'perlfilter.pod'} =
+	    $self->catfile("\$(INST_MAN1DIR)", "perlfilter.\$(MAN1EXT)");
+	delete $self->{MAN3PODS}->{'perlfilter.pod'};
+	$self->SUPER::constants();
+    }
+}
+
+
 WriteMakefile(
 	NAME      => 'Filter',
 	VERSION   => '1.37',
