Description: link with gtop-2.0
Forwarded: no
Author: Andres Salomon <dilinger@voxel.net>
Last-Update: 2010-04-21

--- a/lib/Apache2/Build.pm
+++ b/lib/Apache2/Build.pm
@@ -449,7 +449,7 @@ sub gtop_ldopts_old {
         $xlibs .= ' -lintl';
     }
 
-    return " -lgtop -lgtop_sysdeps -lgtop_common $xlibs";
+    return " -lgtop-2.0 -lgtop_sysdeps-2.0 -lgtop_common-2.0 $xlibs";
 }
 
 sub gtop_ldopts {
