Description: direct bug reports to Debian BTS
Forwarded: not-needed
Author: Gunnar Wolf <gwolf@debian.org>
Last-Update: 2010-04-21

--- a/lib/ModPerl/TestReport.pm
+++ b/lib/ModPerl/TestReport.pm
@@ -97,7 +97,11 @@ sub package2filename {
     return $package;
 }
 
-sub report_to { 'modperl@perl.apache.org' }
-
+# We are reporting to the Debian BTS, not directly to the upstream (Apache)
+sub report_to { 'submit@bugs.debian.org' }
+sub postit_note {
+    # No, no postit notes required :) This report will be typically
+    # embedded in reportbug's, so...
+}
 
 1;
