Description: Fix POD errors
Forwarded: http://mail-archives.apache.org/mod_mbox/perl-modperl/201004.mbox/thread
Author: Damyan Ivanov <dmn@debian.org>
Last-Update: 2010-04-20

--- a/Apache-Test/lib/Apache/Test.pm
+++ b/Apache-Test/lib/Apache/Test.pm
@@ -936,6 +936,8 @@ if no reason is given a default reason w
 
 =head1 Additional Configuration Variables
 
+=over
+
 =item basic_config
 
   my $basic_cfg = Apache::Test::basic_config();
@@ -968,6 +970,8 @@ If no arguments are passed, the referenc
 returned. If one or more arguments are passed the corresponding values
 are returned.
 
+=back
+
 =head1 Test::More Integration
 
 There are a few caveats if you want to use I<Apache::Test> with
