From e51795c8cd44de9eb3ddb175e992454b0c873352 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Wed, 15 Mar 2017 16:42:26 +0000
Subject: [PATCH 1/4] modify bonjour name

---
 etc/netatalk/afp_avahi.c | 20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/etc/netatalk/afp_avahi.c b/etc/netatalk/afp_avahi.c
index 7a14a49..2408f80 100644
--- a/etc/netatalk/afp_avahi.c
+++ b/etc/netatalk/afp_avahi.c
@@ -109,10 +109,11 @@ static void register_stuff(void) {
                                ctx->obj->options.hostname,
                                -1,
                                name,
-                               MAXINSTANCENAMELEN) <= 0) {
+                               MAXINSTANCENAMELEN - 6) <= 0) {
                 LOG(log_error, logtype_afpd, "Could not set Zeroconf instance name: %s", ctx->obj->options.hostname);
                 goto fail;
             }
+            strcat(name, " (AFP)");
         }
 
         LOG(log_info, logtype_afpd, "Registering server '%s' with Bonjour", name);            
@@ -132,6 +133,23 @@ static void register_stuff(void) {
             goto fail;
         }
 
+        if (i) {
+            char *c;
+            c = strrchr(name, ' ');
+            if (c)
+                *c = '\0';
+            avahi_entry_group_add_service(ctx->group,
+                                          AVAHI_IF_UNSPEC,
+                                          AVAHI_PROTO_UNSPEC,
+                                          0,
+                                          name,
+                                          AFP_DNS_SERVICE_TYPE,
+                                          NULL,
+                                          NULL,
+                                          port,
+                                          NULL);
+        }
+
         if (i && avahi_entry_group_add_service_strlst(ctx->group,
                                                       AVAHI_IF_UNSPEC,
                                                       AVAHI_PROTO_UNSPEC,
-- 
2.1.4

