From d69423050318fc581ad5c0f329024968873d36cb Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Wed, 15 Mar 2017 16:43:46 +0000
Subject: [PATCH 2/4] revert to appledb inside datadir

---
 libatalk/util/netatalk_conf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libatalk/util/netatalk_conf.c b/libatalk/util/netatalk_conf.c
index 23bc928..2fd06a2 100644
--- a/libatalk/util/netatalk_conf.c
+++ b/libatalk/util/netatalk_conf.c
@@ -700,7 +700,7 @@ static struct vol *creatvol(AFPObj *obj,
         if(tmpname[i] == '/') tmpname[i] = ':';
 
 
-    if (atalk_iniparser_getboolean(obj->iniconfig, INISEC_GLOBAL, "vol dbnest", 0)) {
+    if (atalk_iniparser_getboolean(obj->iniconfig, INISEC_GLOBAL, "vol dbnest", 1)) {
         EC_NULL( volume->v_dbpath = strdup(path) );
     } else {
         char *global_path;
-- 
2.1.4

