From ca64e528c2dcb89973a896fb1fdfbcbc591ba6fc Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Wed, 15 Mar 2017 16:44:12 +0000
Subject: [PATCH 3/4] fix recent external libtdb

---
 include/atalk/tdb.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/atalk/tdb.h b/include/atalk/tdb.h
index b13ee35..49d449e 100644
--- a/include/atalk/tdb.h
+++ b/include/atalk/tdb.h
@@ -1,4 +1,5 @@
 #ifndef USE_BUILTIN_TDB
+#  include <stdbool.h>
 #  include <tdb.h>
 #else
 #  ifndef __TDB_H__
-- 
2.1.4

