--- a/modprobe.c	2005-08-22 10:10:22.000000000 +0000
+++ b/modprobe.c	2005-08-22 15:49:51.000000000 +0000
@@ -725,8 +725,10 @@
 find_blacklist(const char *modname, const struct module_blacklist *blacklist)
 {
 	while (blacklist) {
-		if (strcmp(blacklist->modulename, modname) == 0)
+		if (strcmp(blacklist->modulename, modname) == 0) {
+			warn("Not loading blacklisted module %s\n", modname);
 			return 1;
+		}
 		blacklist = blacklist->next;
 	}
 	return 0;
