diff -ru a/agent/mibgroup/hardware/fsys/fsys_mntent.c b/agent/mibgroup/hardware/fsys/fsys_mntent.c
--- a/agent/mibgroup/hardware/fsys/fsys_mntent.c	2012-10-09 17:28:58.000000000 -0500
+++ b/agent/mibgroup/hardware/fsys/fsys_mntent.c	2013-10-12 15:50:53.217712757 -0500
@@ -139,7 +139,8 @@
               !strcmp(typename, MNTTYPE_REISERFS) ||
               !strcmp(typename, MNTTYPE_OCFS2) ||
               !strcmp(typename, MNTTYPE_CVFS) ||
-              !strcmp(typename, MNTTYPE_LOFS))
+              !strcmp(typename, MNTTYPE_LOFS) ||
+              !strcmp(typename, MNTTYPE_BTRFS))
        return NETSNMP_FS_TYPE_OTHER;
 
     /*    
diff -ru a/agent/mibgroup/hardware/fsys/mnttypes.h b/agent/mibgroup/hardware/fsys/mnttypes.h
--- a/agent/mibgroup/hardware/fsys/mnttypes.h	2012-10-09 17:28:58.000000000 -0500
+++ b/agent/mibgroup/hardware/fsys/mnttypes.h	2013-10-12 16:50:49.970633707 -0500
@@ -142,6 +142,9 @@
 #ifndef MNTTYPE_CVFS
 #define MNTTYPE_CVFS     "cvfs"
 #endif
+#ifndef MNTTYPE_BTRFS
+#define MNTTYPE_BTRFS    "btrfs"
+#endif
 
 /*
  *  File systems to skip
