diff --git a/avahi-daemon/main.c- b/avahi-daemon/main.c
index d46f40a..4e65837 100644
--- a/avahi-daemon/main.c-
+++ b/avahi-daemon/main.c
@@ -1101,9 +1101,6 @@ static int run_server(DaemonConfig *c) {
 
     ignore_signal(SIGPIPE);
 
-    if (!(nss_support = avahi_nss_support()))
-        avahi_log_warn("WARNING: No NSS support for mDNS detected, consider installing nss-mdns!");
-
     if (!(simple_poll_api = avahi_simple_poll_new())) {
         avahi_log_error("Failed to create main loop object.");
         goto finish;
