From 48313ba2455b6338be92960851bac5d455307bbd Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Wed, 12 Aug 2015 13:50:00 -0700
Subject: [PATCH 3/8] Hide common "new devices added" messages.

---
 minissdpd.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/minissdpd.c b/minissdpd.c
index 49e044a..277df69 100644
--- a/minissdpd.c
+++ b/minissdpd.c
@@ -1342,9 +1342,9 @@ int main(int argc, char * * argv)
 				if(i==0 || (i*deltadev < 0))
 				{
 					if(deltadev > 0)
-						syslog(LOG_NOTICE, "%d new devices added", deltadev);
+						syslog(LOG_INFO, "%d new devices added", deltadev);
 					else if(deltadev < 0)
-						syslog(LOG_NOTICE, "%d devices removed (good-bye!)", -deltadev);
+						syslog(LOG_INFO, "%d devices removed (good-bye!)", -deltadev);
 					deltadev = i;
 				}
 				else if((i*deltadev) >= 0)
@@ -1376,9 +1376,9 @@ int main(int argc, char * * argv)
 				if(i==0 || (i*deltadev < 0))
 				{
 					if(deltadev > 0)
-						syslog(LOG_NOTICE, "%d new devices added", deltadev);
+						syslog(LOG_INFO, "%d new devices added", deltadev);
 					else if(deltadev < 0)
-						syslog(LOG_NOTICE, "%d devices removed (good-bye!)", -deltadev);
+						syslog(LOG_INFO, "%d devices removed (good-bye!)", -deltadev);
 					deltadev = i;
 				}
 				else if((i*deltadev) >= 0)
-- 
1.9.1

