diff -urN db-3.2.9.orig/dist/aclocal/mutex.m4 db-3.2.9/dist/aclocal/mutex.m4
--- db-3.2.9.orig/dist/aclocal/mutex.m4	Thu Jan 11 13:28:31 2001
+++ db-3.2.9/dist/aclocal/mutex.m4	Sun Mar 11 10:26:22 2001
@@ -241,7 +241,7 @@
 dnl Alpha/gcc: OSF/1
 dnl The alpha/gcc code doesn't work as far as I know.  There are
 dnl two versions, both have problems.  See Support Request #1583.
-if test "$db_cv_mutex" = DOESNT_WORK; then
+if test "$db_cv_mutex" = no; then
 AC_TRY_RUN([main(){
 #if defined(__alpha)
 #if defined(__GNUC__)
@@ -279,7 +279,7 @@
 dnl Sparc/gcc: SunOS, Solaris
 dnl The sparc/gcc code doesn't always work, specifically, I've seen assembler
 dnl failures from the stbar instruction on SunOS 4.1.4/sun4c and gcc 2.7.2.2.
-if test "$db_cv_mutex" = DOESNT_WORK; then
+if test "$db_cv_mutex" = no; then
 AC_TRY_RUN([main(){
 #if defined(__sparc__)
 #if defined(__GNUC__)
