diff -urN db-3.2.9/db_dump185/db_dump185.c db-3.2.9.new/db_dump185/db_dump185.c
--- db-3.2.9/db_dump185/db_dump185.c	2001-01-11 11:28:26.000000000 -0700
+++ db-3.2.9.new/db_dump185/db_dump185.c	2002-08-12 06:34:57.000000000 -0600
@@ -21,7 +21,7 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include <db.h>
+#include <db_185.h>
 
 /* Hash Table Information */
 typedef struct hashhdr185 {		/* Disk resident portion */
diff -urN db-3.2.9.orig/dist/Makefile.in db-3.2.9/dist/Makefile.in
--- db-3.2.9.orig/dist/Makefile.in	Thu Nov 30 18:07:33 2000
+++ db-3.2.9/dist/Makefile.in	Sun Mar 11 10:30:49 2001
@@ -105,7 +105,7 @@
 # line, and -l options to the DB185LIB line.
 ##################################################
 DB185INC=	-c @CFLAGS@ -I$(srcdir)/include @CPPFLAGS@
-DB185LIB=
+DB185LIB=	-L.libs -ldb3
 
 ##################################################
 # INSTALLATION DIRECTORIES AND PERMISSIONS
