diff -urN db-3.2.9.orig/dist/Makefile.in db-3.2.9/dist/Makefile.in
--- db-3.2.9.orig/dist/Makefile.in	Thu Nov 30 18:07:33 2000
+++ db-3.2.9/dist/Makefile.in	Mon Mar 12 18:00:52 2001
@@ -409,12 +409,13 @@
 
 install_include:
 	@echo "Installing DB include files: $(includedir) ..."
-	@test -d $(includedir) || \
-	    ($(mkdir) -p $(includedir) && $(chmod) $(dmode) $(includedir))
-	@cd $(includedir) && $(rm) -f db.h db_185.h db_cxx.h
+	@test -d $(DESTDIR)$(includedir) || \
+	    ($(mkdir) -p $(DESTDIR)$(includedir) && \
+	    	$(chmod) $(dmode) $(DESTDIR)$(includedir))
+	@cd $(DESTDIR)$(includedir) && $(rm) -f db.h db_185.h db_cxx.h
 	@$(cp) -p db.h \
-	    $(srcdir)/include/db_cxx.h @ADDITIONAL_INCS@ $(includedir)
-	@cd $(includedir) && $(chmod) $(fmode) db.h db_cxx.h @ADDITIONAL_INCS@
+	    $(srcdir)/include/db_cxx.h @ADDITIONAL_INCS@ $(DESTDIR)$(includedir)
+	@cd $(DESTDIR)$(includedir) && $(chmod) $(fmode) db.h db_cxx.h @ADDITIONAL_INCS@
 
 uninstall_include:
 	-cd $(includedir) && $(rm) -f db.h db_185.h db_cxx.h
@@ -431,81 +432,81 @@
 
 install_static:
 	@echo "Installing DB library: $(libdir) ..."
-	@test -d $(libdir) || \
-	    ($(mkdir) -p $(libdir) && $(chmod) $(dmode) $(libdir))
-	@cd $(libdir) && $(rm) -f $(libdb)
-	@$(cp) -p $(libdb) $(libdir)
-	@cd $(libdir) && $(chmod) $(fmode) $(libdb)
+	@test -d $(DESTDIR)$(libdir) || \
+	    ($(mkdir) -p $(DESTDIR)$(libdir) && $(chmod) $(dmode) $(DESTDIR)$(libdir))
+	@cd $(DESTDIR)$(libdir) && $(rm) -f $(libdb)
+	@$(cp) -p $(libdb) $(DESTDIR)$(libdir)
+	@cd $(DESTDIR)$(libdir) && $(chmod) $(fmode) $(libdb)
 
 install_static_cxx:
 	@echo "Installing DB C++ static library: $(libdir) ..."
-	@test -d $(libdir) || \
-	    ($(mkdir) -p $(libdir) && $(chmod) $(dmode) $(libdir))
-	@cd $(libdir) && $(rm) -f $(libcxx)
-	@$(cp) -p $(libcxx) $(libdir)
-	@cd $(libdir) && $(chmod) $(fmode) $(libcxx)
+	@test -d $(DESTDIR)$(libdir) || \
+	    ($(mkdir) -p $(DESTDIR)$(libdir) && $(chmod) $(dmode) $(DESTDIR)$(libdir))
+	@cd $(DESTDIR)$(libdir) && $(rm) -f $(libcxx)
+	@$(cp) -p $(libcxx) $(DESTDIR)$(libdir)
+	@cd $(DESTDIR)$(libdir) && $(chmod) $(fmode) $(libcxx)
 
 install_dynamic:
 	@echo "Installing DB library: $(libdir) ..."
-	@test -d $(libdir) || \
-	    ($(mkdir) -p $(libdir) && $(chmod) $(dmode) $(libdir))
-	@cd $(libdir) && $(rm) -f \
+	@test -d $(DESTDIR)$(libdir) || \
+	    ($(mkdir) -p $(DESTDIR)$(libdir) && $(chmod) $(dmode) $(DESTDIR)$(libdir))
+	@cd $(DESTDIR)$(libdir) && $(rm) -f \
 	    $(libso_target) $(libso) $(libso_default) $(libso_major)
-	@$(INSTALLER) $(libso_target) $(libdir)
-	@cd $(libdir) && $(ln) -s $(libso) $(libso_default)
-	@cd $(libdir) && $(ln) -s $(libso) $(libso_major)
-	@$(LIBTOOL) --mode=finish $(libdir)
+	@$(INSTALLER) $(libso_target) $(DESTDIR)$(libdir)
+	@cd $(DESTDIR)$(libdir) && $(ln) -s $(libso) $(libso_default)
+	@cd $(DESTDIR)$(libdir) && $(ln) -s $(libso) $(libso_major)
+	@$(LIBTOOL) --mode=finish $(DESTDIR)$(libdir)
 
 install_dynamic_cxx:
 	@echo "Installing DB C++ library: $(libdir) ..."
-	@test -d $(libdir) || \
-	    ($(mkdir) -p $(libdir) && $(chmod) $(dmode) $(libdir))
-	@cd $(libdir) && $(rm) -f \
+	@test -d $(DESTDIR)$(libdir) || \
+	    ($(mkdir) -p $(DESTDIR)$(libdir) && $(chmod) $(dmode) $(DESTDIR)$(libdir))
+	@cd $(DESTDIR)$(libdir) && $(rm) -f \
 	    $(libxso_target) $(libxso) $(libxso_default) $(libxso_major)
-	@$(INSTALLER) $(libxso_target) $(libdir)
-	@cd $(libdir) && $(ln) -s $(libxso) $(libxso_default)
-	@cd $(libdir) && $(ln) -s $(libxso) $(libxso_major)
+	@$(INSTALLER) $(libxso_target) $(DESTDIR)$(libdir)
+	@cd $(DESTDIR)$(libdir) && $(ln) -s $(libxso) $(libxso_default)
+	@cd $(DESTDIR)$(libdir) && $(ln) -s $(libxso) $(libxso_major)
 
 install_tcl:
 	@echo "Installing DB Tcl library: $(libdir) ..."
-	@test -d $(libdir) || \
-	    ($(mkdir) -p $(libdir) && $(chmod) $(dmode) $(libdir))
-	@cd $(libdir) && $(rm) -f \
+	@test -d $(DESTDIR)$(libdir) || \
+	    ($(mkdir) -p $(DESTDIR)$(libdir) && $(chmod) $(dmode) $(DESTDIR)$(libdir))
+	@cd $(DESTDIR)$(libdir) && $(rm) -f \
 	    $(libtso_target) $(libtso) $(libtso_default) $(libtso_major)
-	@$(INSTALLER) $(libtso_target) $(libdir)
-	@cd $(libdir) && $(ln) -s $(libtso) $(libtso_default)
-	@cd $(libdir) && $(ln) -s $(libtso) $(libtso_major)
+	@$(INSTALLER) $(libtso_target) $(DESTDIR)$(libdir)
+	@cd $(DESTDIR)$(libdir) && $(ln) -s $(libtso) $(libtso_default)
+	@cd $(DESTDIR)$(libdir) && $(ln) -s $(libtso) $(libtso_major)
 
 install_java:
 	@echo "Installing DB Java library: $(libdir) ..."
-	@test -d $(libdir) || \
-	    ($(mkdir) -p $(libdir) && $(chmod) $(dmode) $(libdir))
-	@cd $(libdir) && $(rm) -f \
+	@test -d $(DESTDIR)$(libdir) || \
+	    ($(mkdir) -p $(DESTDIR)$(libdir) && $(chmod) $(dmode) $(DESTDIR)$(libdir))
+	@cd $(DESTDIR)$(libdir) && $(rm) -f \
 	    $(libjso_target) $(libjso) $(libjso_default) $(libjso_major)
-	@$(INSTALLER) $(libjso_target) $(libdir)
-	@cd $(libdir) && $(ln) -s $(libjso) $(libjso_default)
-	@cd $(libdir) && $(ln) -s $(libjso) $(libjso_major)
-	@cd $(libdir) && $(ln) -s $(libjso) $(libjso_g)
-	@$(cp) $(JAVA_CLASSTOP)/$(libj_jarfile) $(libdir)
+	@$(INSTALLER) $(libjso_target) $(DESTDIR)$(libdir)
+	@cd $(DESTDIR)$(libdir) && $(ln) -s $(libjso) $(libjso_default)
+	@cd $(DESTDIR)$(libdir) && $(ln) -s $(libjso) $(libjso_major)
+	@cd $(DESTDIR)$(libdir) && $(ln) -s $(libjso) $(libjso_g)
+	@$(cp) $(JAVA_CLASSTOP)/$(libj_jarfile) $(DESTDIR)$(libdir)
 
 install_utilities:
 	@echo "Installing DB utilities: $(bindir) ..."
-	@test -d $(bindir) || \
-	    ($(mkdir) -p $(bindir) && $(chmod) $(dmode) $(bindir))
-	@cd $(bindir) && $(rm) -f $(PROGS)
-	@$(INSTALLER) -fp $(PROGS) $(bindir)
-	@cd $(bindir) && (test ! -f $(strip) || $(strip) $(PROGS))
-	@cd $(bindir) && $(chmod) $(emode) $(PROGS)
+	@test -d $(DESTDIR)$(bindir) || \
+	    ($(mkdir) -p $(DESTDIR)$(bindir) && $(chmod) $(dmode) $(DESTDIR)$(bindir))
+	@cd $(DESTDIR)$(bindir) && $(rm) -f $(PROGS)
+	@$(INSTALLER) -fp $(PROGS) $(DESTDIR)$(bindir)
+	@cd $(DESTDIR)$(bindir) && (test ! -f $(strip) || $(strip) $(PROGS))
+	@cd $(DESTDIR)$(bindir) && $(chmod) $(emode) $(PROGS)
 
 uninstall_utilities:
 	-cd $(bindir) && $(rm) -f $(PROGS)
 
 install_docs:
 	@echo "Installing documentation: $(docdir) ..."
-	@test -d $(docdir) || \
-	    ($(mkdir) -p $(docdir) && $(chmod) $(dmode) $(docdir))
-	@cd $(docdir) && $(rm) -rf *
-	@cd $(srcdir)/docs && $(cp) -pr * $(docdir)/
+	@test -d $(DESTDIR)$(docdir) || \
+	    ($(mkdir) -p $(DESTDIR)$(docdir) && $(chmod) $(dmode) $(DESTDIR)$(docdir))
+	@cd $(DESTDIR)$(docdir) && $(rm) -rf *
+	@cd $(srcdir)/docs && $(cp) -pr * $(DESTDIR)$(docdir)/
 
 uninstall_docs:
 	-cd $(docdir) && $(rm) -rf *
