diff -urN db-3.2.9.orig/dist/aclocal/tcl.m4 db-3.2.9/dist/aclocal/tcl.m4
--- db-3.2.9.orig/dist/aclocal/tcl.m4	Tue Jun 27 16:38:41 2000
+++ db-3.2.9/dist/aclocal/tcl.m4	Mon Mar 12 18:48:38 2001
@@ -49,11 +49,11 @@
 	AC_CACHE_VAL(ac_cv_c_tclconfig,[
 
 	    # First check to see if --with-tclconfig was specified.
-	    if test "${with_tclconfig}" != no; then
-		if test -f "${with_tclconfig}/tclConfig.sh" ; then
-		    ac_cv_c_tclconfig=`(cd ${with_tclconfig}; pwd)`
+	    if test "${with_tcl}" != no; then
+		if test -f "${with_tcl}/tclConfig.sh" ; then
+		    ac_cv_c_tclconfig=`(cd ${with_tcl}; pwd)`
 		else
-		    AC_MSG_ERROR([${with_tclconfig} directory doesn't contain tclConfig.sh])
+		    AC_MSG_ERROR([${with_tcl} directory doesn't contain tclConfig.sh])
 		fi
 	    fi
 
@@ -110,17 +110,17 @@
 		AC_MSG_ERROR([--with-tcl requires --enable-dynamic])
 	fi
 
-	AC_SUBST(TCFLAGS)
-
 	SC_PATH_TCLCONFIG
 	SC_LOAD_TCLCONFIG
 
-	if test x"$TCL_PREFIX" != x && test -f "$TCL_PREFIX/include/tcl.h"; then
-		TCFLAGS="-I$TCL_PREFIX/include"
+	if test x"$TCL_PREFIX" != x && test -f "$TCL_PREFIX/include/tcl$TCL_VERSION/tcl.h"; then
+		TCFLAGS="-I$TCL_PREFIX/include/tcl$TCL_VERSION"
 	fi
 
-	LIBS="$LIBS $TCL_LIB_SPEC $TCL_LIBS"
+	LIBTSO_LIBS="$LIBTSO_LIBS $TCL_LIB_SPEC $TCL_LIBS"
 
 	ADDITIONAL_LIBS="$ADDITIONAL_LIBS \$(libtso_target)"
 	DEFAULT_INSTALL="${DEFAULT_INSTALL} install_tcl"
+
+	AC_SUBST(TCFLAGS)
 fi])
