diff -urN db-3.2.9.orig/dist/configure.in db-3.2.9/dist/configure.in
--- db-3.2.9.orig/dist/configure.in	Fri Jan 19 12:36:43 2001
+++ db-3.2.9/dist/configure.in	Wed May  2 10:38:04 2001
@@ -222,9 +222,9 @@
 	|| AC_MSG_ERROR([libtool configure failed])
 
 	SOSUFFIX=`sed -e '/^library_names_spec=/!d' -e 's/.*\.\([[a-zA-Z0-9_]]*\).*/\1/' ./libtool`
-	DEFAULT_LIB="\$(libso_target)"
-	DEFAULT_INSTALL="install_dynamic"
-	LIBDB_ARGS="\$(libso_linkname)"
+	DEFAULT_LIB="\$(libso_target) ${DEFAULT_LIB}"
+	DEFAULT_INSTALL="install_dynamic ${DEFAULT_INSTALL}"
+	LIBDB_ARGS="${LIBDB_ARGS} \$(libso_linkname)"
 	LIBTOOL="\$(SHELL) ./libtool"
 
 	MAKEFILE_CC="\$(LIBTOOL) --mode=compile ${SAVE_CC}"
@@ -241,8 +241,8 @@
 dnl Optional C++ API.
 if test "$db_cv_cxx" = "yes"; then
 	if test "$db_cv_dynamic" = "yes"; then
-		ADDITIONAL_LIBS="$ADDITIONAL_LIBS \$(libxso_target)"
-		DEFAULT_INSTALL="${DEFAULT_INSTALL} install_dynamic_cxx"
+		ADDITIONAL_LIBS="$ADDITIONAL_LIBS \$(libxso_target) \$(libcxx)"
+		DEFAULT_INSTALL="${DEFAULT_INSTALL} install_dynamic_cxx install_static_cxx"
 	else
 		ADDITIONAL_LIBS="$ADDITIONAL_LIBS \$(libcxx)"
 		DEFAULT_INSTALL="${DEFAULT_INSTALL} install_static_cxx"
