diff -urN db-3.2.9.orig/dist/Makefile.in db-3.2.9/dist/Makefile.in
--- db-3.2.9.orig/dist/Makefile.in	Tue Mar 13 00:12:26 2001
+++ db-3.2.9/dist/Makefile.in	Tue Mar 13 00:39:00 2001
@@ -26,19 +26,14 @@
 LIBXSO_LIBS=	@LIBXSO_LIBS@
 
 POSTLINK=	@POSTLINK@
-SOLINK=		@SOLINK@
+SOLINK=		@SOLINK@ -version-info @DB_VERSION_MAJOR@:@DB_VERSION_MINOR@
 SOFLAGS=	@SOFLAGS@
 SOMAJOR=	@DB_VERSION_MAJOR@
 SOVERSION=	@DB_VERSION_MAJOR@.@DB_VERSION_MINOR@
 
 libdb=		libdb.a
 
-libso_base=	libdb
-libso_linkname=	$(libso_base)-$(SOVERSION).la
-libso=		$(libso_base)-$(SOVERSION).@SOSUFFIX@
-libso_target=	$(libso_base)-$(SOVERSION).la
-libso_default=	$(libso_base).@SOSUFFIX@
-libso_major=	$(libso_base)-$(SOMAJOR).@SOSUFFIX@
+libso=		libdb3.la
 
 ##################################################
 # C++
@@ -48,11 +43,7 @@
 #	--enable-cxx
 ##################################################
 libcxx=		libdb_cxx.a
-libxso_base=	libdb_cxx
-libxso=		$(libxso_base)-$(SOVERSION).@SOSUFFIX@
-libxso_target=	$(libxso_base)-$(SOVERSION).la
-libxso_default=	$(libxso_base).@SOSUFFIX@
-libxso_major=	$(libxso_base)-$(SOMAJOR).@SOSUFFIX@
+libxso=		libdb_cxx.la
 
 ##################################################
 # JAVA
@@ -73,12 +64,7 @@
 JAVA_EXDIR=	$(JAVA_SRCDIR)/com/sleepycat/examples
 
 libj_jarfile=	db.jar
-libjso_base=	libdb_java
-libjso=		$(libjso_base)-$(SOVERSION).@SOSUFFIX@
-libjso_target=	$(libjso_base)-$(SOVERSION).la
-libjso_default=	$(libjso_base).@SOSUFFIX@
-libjso_major=	$(libjso_base)-$(SOMAJOR).@SOSUFFIX@
-libjso_g=	$(libjso_base)-$(SOVERSION)_g.@SOSUFFIX@
+libjso=		libdb_java.la
 
 ##################################################
 # TCL
@@ -88,11 +74,7 @@
 #	--enable-tcl --with-tcl=DIR --enable-dynamic
 ##################################################
 TCFLAGS=	@TCFLAGS@
-libtso_base=	libdb_tcl
-libtso=		$(libtso_base)-$(SOVERSION).@SOSUFFIX@
-libtso_target=	$(libtso_base)-$(SOVERSION).la
-libtso_default=	$(libtso_base).@SOSUFFIX@
-libtso_major=	$(libtso_base)-$(SOMAJOR).@SOSUFFIX@
+libtso=		libdb_tcl.la
 
 ##################################################
 # DB_DUMP185 UTILITY
@@ -238,20 +220,20 @@
 	$(ar) $@ $(COBJS) $(OBJS)
 	test ! -f $(ranlib) || $(ranlib) $@
 
-$(libso_target): $(OBJS)
-	$(SOLINK) $(SOFLAGS) -o $(libso_target) \
+$(libso): $(OBJS)
+	$(SOLINK) $(SOFLAGS) -o $(libso) \
 	    $(OBJS) $(LDFLAGS) $(LIBSO_LIBS)
 
-$(libxso_target): $(COBJS) $(OBJS)
-	$(SOLINK) $(SOFLAGS) -o $(libxso_target) \
+$(libxso): $(COBJS) $(OBJS)
+	$(SOLINK) $(SOFLAGS) -o $(libxso) \
 	    $(COBJS) $(OBJS) $(LDFLAGS) $(LIBXSO_LIBS)
 
-$(libjso_target): $(JOBJS) $(OBJS)
-	$(SOLINK) $(SOFLAGS) -o $(libjso_target) \
+$(libjso): $(JOBJS) $(OBJS)
+	$(SOLINK) $(SOFLAGS) -o $(libjso) \
 	    $(JOBJS) $(OBJS) $(LDFLAGS) $(LIBJSO_LIBS)
 
-$(libtso_target): $(TOBJS) $(OBJS)
-	$(SOLINK) $(SOFLAGS) -o $(libtso_target) \
+$(libtso): $(TOBJS) $(OBJS)
+	$(SOLINK) $(SOFLAGS) -o $(libtso) \
 	    $(TOBJS) $(OBJS) $(LDFLAGS) $(LIBTSO_LIBS)
 
 ##################################################
@@ -385,10 +367,7 @@
 	$(rm) -f *@o@ *.o *.lo core *.core
 	$(rm) -rf ALL.OUT TESTDIR
 	$(rm) -rf .libs $(libdb) $(libcxx)
-	$(rm) -rf $(libso_target) $(libso) $(libso_default) $(libso_major)
-	$(rm) -rf $(libxso_target) $(libxso) $(libxso_default) $(libxso_major)
-	$(rm) -rf $(libtso_target) $(libtso) $(libtso_default) $(libtso_major)
-	$(rm) -rf $(libjso_target) $(libjso) $(libjso_default) $(libjso_major)
+	$(rm) -rf $(libso) libdb* db.jar
 
 depend obj:
 
@@ -450,43 +429,29 @@
 	@echo "Installing DB library: $(libdir) ..."
 	@test -d $(DESTDIR)$(libdir) || \
 	    ($(mkdir) -p $(DESTDIR)$(libdir) && $(chmod) $(dmode) $(DESTDIR)$(libdir))
-	@cd $(DESTDIR)$(libdir) && $(rm) -f \
-	    $(libso_target) $(libso) $(libso_default) $(libso_major)
-	@$(INSTALLER) $(libso_target) $(DESTDIR)$(libdir)
-	@cd $(DESTDIR)$(libdir) && $(ln) -s $(libso) $(libso_default)
-	@cd $(DESTDIR)$(libdir) && $(ln) -s $(libso) $(libso_major)
+	@$(INSTALLER) $(libso) $(DESTDIR)$(libdir)
 	@$(LIBTOOL) --mode=finish $(DESTDIR)$(libdir)
 
 install_dynamic_cxx:
 	@echo "Installing DB C++ library: $(libdir) ..."
 	@test -d $(DESTDIR)$(libdir) || \
 	    ($(mkdir) -p $(DESTDIR)$(libdir) && $(chmod) $(dmode) $(DESTDIR)$(libdir))
-	@cd $(DESTDIR)$(libdir) && $(rm) -f \
-	    $(libxso_target) $(libxso) $(libxso_default) $(libxso_major)
-	@$(INSTALLER) $(libxso_target) $(DESTDIR)$(libdir)
-	@cd $(DESTDIR)$(libdir) && $(ln) -s $(libxso) $(libxso_default)
-	@cd $(DESTDIR)$(libdir) && $(ln) -s $(libxso) $(libxso_major)
+	@$(INSTALLER) $(libxso) $(DESTDIR)$(libdir)
+	@$(LIBTOOL) --mode=finish $(DESTDIR)$(libdir)
 
 install_tcl:
 	@echo "Installing DB Tcl library: $(libdir) ..."
 	@test -d $(DESTDIR)$(libdir) || \
 	    ($(mkdir) -p $(DESTDIR)$(libdir) && $(chmod) $(dmode) $(DESTDIR)$(libdir))
-	@cd $(DESTDIR)$(libdir) && $(rm) -f \
-	    $(libtso_target) $(libtso) $(libtso_default) $(libtso_major)
-	@$(INSTALLER) $(libtso_target) $(DESTDIR)$(libdir)
-	@cd $(DESTDIR)$(libdir) && $(ln) -s $(libtso) $(libtso_default)
-	@cd $(DESTDIR)$(libdir) && $(ln) -s $(libtso) $(libtso_major)
+	@$(INSTALLER) $(libtso) $(DESTDIR)$(libdir)
+	@$(LIBTOOL) --mode=finish $(DESTDIR)$(libdir)
 
 install_java:
 	@echo "Installing DB Java library: $(libdir) ..."
 	@test -d $(DESTDIR)$(libdir) || \
 	    ($(mkdir) -p $(DESTDIR)$(libdir) && $(chmod) $(dmode) $(DESTDIR)$(libdir))
-	@cd $(DESTDIR)$(libdir) && $(rm) -f \
-	    $(libjso_target) $(libjso) $(libjso_default) $(libjso_major)
-	@$(INSTALLER) $(libjso_target) $(DESTDIR)$(libdir)
-	@cd $(DESTDIR)$(libdir) && $(ln) -s $(libjso) $(libjso_default)
-	@cd $(DESTDIR)$(libdir) && $(ln) -s $(libjso) $(libjso_major)
-	@cd $(DESTDIR)$(libdir) && $(ln) -s $(libjso) $(libjso_g)
+	@$(INSTALLER) $(libjso) $(DESTDIR)$(libdir)
+	@$(LIBTOOL) --mode=finish $(DESTDIR)$(libdir)
 	@$(cp) $(JAVA_CLASSTOP)/$(libj_jarfile) $(DESTDIR)$(libdir)
 
 install_utilities:
diff -urN db-3.2.9.orig/dist/aclocal/tcl.m4 db-3.2.9/dist/aclocal/tcl.m4
--- db-3.2.9.orig/dist/aclocal/tcl.m4	Tue Mar 13 00:12:26 2001
+++ db-3.2.9/dist/aclocal/tcl.m4	Tue Mar 13 00:23:16 2001
@@ -119,7 +119,7 @@
 
 	LIBTSO_LIBS="$LIBTSO_LIBS $TCL_LIB_SPEC $TCL_LIBS"
 
-	ADDITIONAL_LIBS="$ADDITIONAL_LIBS \$(libtso_target)"
+	ADDITIONAL_LIBS="$ADDITIONAL_LIBS \$(libtso)"
 	DEFAULT_INSTALL="${DEFAULT_INSTALL} install_tcl"
 
 	AC_SUBST(TCFLAGS)
diff -urN db-3.2.9.orig/dist/configure.in db-3.2.9/dist/configure.in
--- db-3.2.9.orig/dist/configure.in	Tue Mar 13 00:12:26 2001
+++ db-3.2.9/dist/configure.in	Tue Mar 13 00:23:10 2001
@@ -218,12 +218,11 @@
 	${CONFIG_SHELL-/bin/sh} $srcdir/ltconfig \
 		--no-verify $srcdir/ltmain.sh \
 		--output=./libtool $host_os \
-		--disable-static \
 	|| AC_MSG_ERROR([libtool configure failed])
 
 	SOSUFFIX=`sed -e '/^library_names_spec=/!d' -e 's/.*\.\([[a-zA-Z0-9_]]*\).*/\1/' ./libtool`
-	DEFAULT_LIB="\$(libso_target) ${DEFAULT_LIB}"
-	DEFAULT_INSTALL="install_dynamic ${DEFAULT_INSTALL}"
-	LIBDB_ARGS="${LIBDB_ARGS} \$(libso_linkname)"
+	DEFAULT_LIB="\$(libso)"
+	DEFAULT_INSTALL="install_dynamic"
+	LIBDB_ARGS="\$(libso)"
 	LIBTOOL="\$(SHELL) ./libtool"
 
@@ -233,7 +232,7 @@
 
 	INSTALLER="\$(LIBTOOL) --mode=install cp"
 	POSTLINK="\$(LIBTOOL) --mode=execute true"
-	SOLINK="\$(LIBTOOL) --mode=link ${SAVE_CC} -avoid-version"
+	SOLINK="\$(LIBTOOL) --mode=link ${SAVE_CC}"
 	SOFLAGS="-rpath \$(libdir)"
 	o=".lo"
 fi
@@ -241,8 +240,8 @@
 dnl Optional C++ API.
 if test "$db_cv_cxx" = "yes"; then
 	if test "$db_cv_dynamic" = "yes"; then
-		ADDITIONAL_LIBS="$ADDITIONAL_LIBS \$(libxso_target) \$(libcxx)"
-		DEFAULT_INSTALL="${DEFAULT_INSTALL} install_dynamic_cxx install_static_cxx"
+		ADDITIONAL_LIBS="$ADDITIONAL_LIBS \$(libxso)"
+		DEFAULT_INSTALL="${DEFAULT_INSTALL} install_dynamic_cxx"
 	else
 		ADDITIONAL_LIBS="$ADDITIONAL_LIBS \$(libcxx)"
 		DEFAULT_INSTALL="${DEFAULT_INSTALL} install_static_cxx"
@@ -264,7 +263,7 @@
 		AC_MSG_ERROR([no jar utility in PATH])
 	fi
 	AC_PATH_PROG(JAVACABS, javac, nojavac)
-	ADDITIONAL_LIBS="$ADDITIONAL_LIBS \$(libjso_target)"
+	ADDITIONAL_LIBS="$ADDITIONAL_LIBS \$(libjso)"
 	ADDITIONAL_LANG="$ADDITIONAL_LANG java"
 	DEFAULT_INSTALL="${DEFAULT_INSTALL} install_java"
 
