diff -urN db-3.2.9.orig/include/mutex.h db-3.2.9/include/mutex.h
--- db-3.2.9.orig/include/mutex.h	Thu Jan 11 13:28:56 2001
+++ db-3.2.9/include/mutex.h	Mon Mar 26 23:40:46 2001
@@ -327,7 +327,7 @@
  */
 #define	MUTEX_SET(tsl) ({						\
 	register tsl_t *__l = (tsl);					\
-	int __r;							\
+	unsigned char __r;							\
 	    asm volatile("tas  %1; \n					\
 			  seq  %0"					\
 		: "=dm" (__r), "=m" (*__l)				\
