--- db-3.2.9/dist/Makefile.in~	Tue Mar 27 00:01:58 2001
+++ db-3.2.9/dist/Makefile.in	Tue Mar 27 00:03:00 2001
@@ -42,8 +42,8 @@
 # You must configure it using:
 #	--enable-cxx
 ##################################################
-libcxx=		libdb_cxx.a
-libxso=		libdb_cxx.la
+libcxx=		libdb3_cxx.a
+libxso=		libdb3_cxx.la
 
 ##################################################
 # JAVA
@@ -216,17 +216,17 @@
 	$(ar) $@ $(OBJS)
 	test ! -f $(ranlib) || $(ranlib) $@
 
-$(libcxx): $(COBJS) $(OBJS)
-	$(ar) $@ $(COBJS) $(OBJS)
+$(libcxx): $(COBJS)
+	$(ar) $@ $(COBJS)
 	test ! -f $(ranlib) || $(ranlib) $@
 
 $(libso): $(OBJS)
 	$(SOLINK) $(SOFLAGS) -o $(libso) \
 	    $(OBJS) $(LDFLAGS) $(LIBSO_LIBS)
 
-$(libxso): $(COBJS) $(OBJS)
+$(libxso): $(COBJS)
 	$(SOLINK) $(SOFLAGS) -o $(libxso) \
-	    $(COBJS) $(OBJS) $(LDFLAGS) $(LIBXSO_LIBS)
+	    $(COBJS) $(LDFLAGS) $(LIBXSO_LIBS) -L.libs/ -ldb3
 
 $(libjso): $(JOBJS) $(OBJS)
 	$(SOLINK) $(SOFLAGS) -o $(libjso) \
