diff -urN db-3.2.9.orig/dist/aclocal/mutex.m4 db-3.2.9/dist/aclocal/mutex.m4
--- db-3.2.9.orig/dist/aclocal/mutex.m4	Thu Apr  5 01:09:42 2001
+++ db-3.2.9/dist/aclocal/mutex.m4	Thu Apr  5 01:10:28 2001
@@ -255,7 +255,7 @@
 dnl PaRisc/gcc: HP/UX
 if test "$db_cv_mutex" = no; then
 AC_TRY_RUN([main(){
-#if defined(__hppa)
+#if defined(__hppa) || defined(__hppa__)
 #if defined(__GNUC__)
 exit(0);
 #endif
