--- db-3.2.9/dist/ltconfig~	Mon Apr 30 23:56:03 2001
+++ db-3.2.9/dist/ltconfig	Mon Apr 30 23:55:34 2001
@@ -1232,7 +1232,7 @@
 
   *)
     if $LD --help 2>&1 | egrep ': supported targets:.* elf' > /dev/null; then
-      archive_cmds='$CC -shared $libobjs $deplibs $linkopts ${wl}-soname $wl$soname -o $lib'
+      archive_cmds='$CC -shared -Wl,-Bsymbolic -Wl,--version-script=Versions $libobjs $deplibs $linkopts ${wl}-soname $wl$soname -o $lib'
       archive_expsym_cmds='$CC -shared $libobjs $deplibs $linkopts ${wl}-soname $wl$soname ${wl}-retain-symbols-file $wl$export_symbols -o $lib'
     else
       ld_shlibs=no
--- /dev/null   Mon Mar  4 22:44:31 2002
+++ db-3.2.9/build_unix/Versions        Sat Mar 16 21:09:38 2002
@@ -0,0 +1,4 @@
+DB3_2 {
+  global:
+    *;
+};
