diff -urN db-3.2.9.orig/include/mutex.h db-3.2.9/include/mutex.h
--- db-3.2.9.orig/include/mutex.h	Fri May  4 00:58:04 2001
+++ db-3.2.9/include/mutex.h	Fri May  4 01:05:16 2001
@@ -558,14 +558,23 @@
  *
  * For gcc/sparc, 0 is clear, 1 is set.
  */
-#define	MUTEX_SET(tsl) ({						\
-	register tsl_t *__l = (tsl);					\
-	register tsl_t __r;						\
-	__asm__ volatile						\
-	    ("ldstub [%1],%0; stbar"					\
-	    : "=r"( __r) : "r" (__l));					\
-	!__r;								\
+
+#ifndef __arch64__
+#define MUTEX_SET(tsl) ({					\
+	register tsl_t *__l = (tsl), __r;		\
+	__asm__ volatile ("ldstub [%1], %0"		\
+	    : "=r"( __r) : "r" (__l) : "memory");	\
+	!__r;						\
 })
+#else
+#define MUTEX_SET(tsl) ({					\
+	register tsl_t *__l = (tsl), __r;		\
+	__asm__ volatile ("ldstub [%1], %0"		\
+	    "membar #StoreLoad | #StoreStore"		\
+	    : "=r"( __r) : "r" (__l) : "memory");	\
+	!__r;						\
+})
+#endif
 
 #define	MUTEX_UNSET(tsl)	(*(tsl) = 0)
 #define	MUTEX_INIT(tsl)		MUTEX_UNSET(tsl)
