diff -urN db-3.2.9.orig/dist/Makefile.in db-3.2.9/dist/Makefile.in
--- db-3.2.9.orig/dist/Makefile.in	Thu Jun 21 09:35:01 2001
+++ db-3.2.9/dist/Makefile.in	Thu Jun 21 09:45:41 2001
@@ -27,6 +27,7 @@
 
 POSTLINK=	@POSTLINK@
 SOLINK=		@SOLINK@ -version-info @DB_VERSION_MAJOR@:@DB_VERSION_MINOR@
+SOLINKX=	@SOLINKX@ -version-info @DB_VERSION_MAJOR@:@DB_VERSION_MINOR@
 SOFLAGS=	@SOFLAGS@
 SOMAJOR=	@DB_VERSION_MAJOR@
 SOVERSION=	@DB_VERSION_MAJOR@.@DB_VERSION_MINOR@
@@ -225,7 +226,7 @@
 	    $(OBJS) $(LDFLAGS) $(LIBSO_LIBS)
 
 $(libxso): $(COBJS)
-	$(SOLINK) $(SOFLAGS) -o $(libxso) \
+	$(SOLINKX) $(SOFLAGS) -o $(libxso) \
 	    $(COBJS) $(LDFLAGS) $(LIBXSO_LIBS) -L.libs/ -ldb3
 
 $(libjso): $(JOBJS) $(OBJS)
diff -urN db-3.2.9.orig/dist/configure.in db-3.2.9/dist/configure.in
--- db-3.2.9.orig/dist/configure.in	Thu Jun 21 09:35:01 2001
+++ db-3.2.9/dist/configure.in	Thu Jun 21 09:46:02 2001
@@ -48,6 +48,7 @@
 AC_SUBST(RPC_OBJS)
 AC_SUBST(SOFLAGS)
 AC_SUBST(SOLINK)
+AC_SUBST(SOLINKX)
 AC_SUBST(SOSUFFIX)
 
 dnl $o is set to ".o" or ".lo", and is the file suffix used in the
@@ -233,6 +234,7 @@
 	INSTALLER="\$(LIBTOOL) --mode=install cp"
 	POSTLINK="\$(LIBTOOL) --mode=execute true"
 	SOLINK="\$(LIBTOOL) --mode=link ${SAVE_CC}"
+	SOLINKX="\$(LIBTOOL) --mode=link ${SAVE_CXX}"
 	SOFLAGS="-rpath \$(libdir)"
 	o=".lo"
 fi
