diff -u db-3.2.9/include/mutex.h db-3.3.11/include/mutex.h
--- db-3.2.9/include/mutex.h	Thu Nov 29 08:18:30 2001
+++ db-3.3.11/include/mutex.h	Wed Nov 28 13:54:02 2001
@@ -480,7 +480,7 @@
  * common case of a locked mutex without wasting cycles making a reservation.
  *
  * 'set' mutexes have the value 1, like on Intel; the returned value from
- * MUTEX_SET() is 1 if the mutex previously had its low bit set, 0 otherwise.
+ * MUTEX_SET() is 1 if the mutex previously had its low bit clear, 0 otherwise.
  */
 #define	MUTEX_SET(tsl)	({		\
 	int __one = 1;			\
@@ -496,7 +496,7 @@
 1:"					\
 	: "=&r" (__r)			\
 	: "r" (__l), "r" (__one));	\
-	__r & 1;			\
+	!(__r & 1);			\
 })
 
 #define	MUTEX_UNSET(tsl)	(*(tsl) = 0)
