--- db-3.2.9/dist/Makefile.in.orig	Fri Apr  5 22:42:44 2002
+++ db-3.2.9/dist/Makefile.in	Fri Apr  5 22:43:50 2002
@@ -221,7 +221,7 @@
 	$(ar) $@ $(COBJS)
 	test ! -f $(ranlib) || $(ranlib) $@
 
-$(libso): $(OBJS)
+$(libso): $(OBJS) Versions
 	$(SOLINK) $(SOFLAGS) -o $(libso) \
 	    $(OBJS) $(LDFLAGS) $(LIBSO_LIBS)
 
@@ -236,6 +236,13 @@
 $(libtso): $(TOBJS) $(OBJS)
 	$(SOLINK) $(SOFLAGS) -o $(libtso) \
 	    $(TOBJS) $(OBJS) $(LDFLAGS) $(LIBTSO_LIBS)
+
+Versions: $(OBJS) 
+	rm -f $@
+	echo "DB3_2 {" > $@
+	echo "  global:" >> $@
+	nm $(OBJS) | grep " [Tt] " | cut -d" " -f3 | sed -e 's/$$/;/' >> $@
+	echo "};" >> $@
 
 ##################################################
 # Creating individual dependencies and actions for building class

