diff -Nur db-3.2.9/dist/configure.in db-3.2.9/dist/configure.in
--- db-3.2.9/dist/configure.in	2004-04-18 13:01:57.000000000 +0200
+++ db-3.2.9/dist/configure.in	2004-04-18 12:52:37.000000000 +0200
@@ -374,13 +374,6 @@
 AC_REPLACE_FUNCS(getcwd getopt memcmp memcpy memmove)
 AC_REPLACE_FUNCS(raise snprintf strcasecmp strerror vsnprintf)
 
-dnl XXX
-dnl Nasty hack.  AC_REPLACE_FUNCS added entries of the form xxx.o to the
-dnl LIBOBJS variable.  They have to be xxx.lo if we are building shared
-dnl libraries.  Use sed, configure already requires it.
-tmp="`echo \"$LIBOBJS\" | sed \"s/\.o/${o}/g\"`"
-LIBOBJS="$tmp"
-
 dnl Check for system functions we optionally use.
 AC_CHECK_FUNCS(getuid pstat_getdynamic sysconf sched_yield strtoul yield)
 
