diff --git a/configure.ac b/configure.ac
index 4e99deb..883a9aa 100644
--- a/configure.ac
+++ b/configure.ac
@@ -48,17 +48,10 @@ AC_TYPE_UINT8_T
 # Checks for library functions.
 AC_CHECK_FUNCS([asprintf strcasecmp strdup strerror strndup stpcpy vasprintf])
 
-AC_CHECK_HEADER(endian.h, [ac_cv_have_endian_h="yes"], [ac_cv_have_endian_h="no"])
-if test "x$ac_cv_have_endian_h" = "xno"; then
-  AC_DEFINE(__LITTLE_ENDIAN,1234,[little endian])
-  AC_DEFINE(__BIG_ENDIAN,4321,[big endian])
-  AC_C_BIGENDIAN([ac_cv_c_bigendian="yes"], [ac_cv_c_bigendian="no"], [], [])
-  if test "x$ac_cv_c_bigendian" = "xyes"; then
-    AC_DEFINE(__BYTE_ORDER,4321,[big endian byte order])
-  else
-    AC_DEFINE(__BYTE_ORDER,1234,[little endian byte order])
-  fi
-fi
+# Checking endianness
+AC_C_BIGENDIAN([AC_DEFINE([__BIG_ENDIAN__], [1], [big endian])],
+               [AC_DEFINE([__LITTLE_ENDIAN__], [1], [little endian])])
+
 
 # Check for operating system
 AC_MSG_CHECKING([whether to enable WIN32 build settings])
diff --git a/src/common.h b/src/common.h
index beb4a2b..c71ef4a 100644
--- a/src/common.h
+++ b/src/common.h
@@ -1,6 +1,10 @@
 #ifndef COMMON_H
 #define COMMON_H
 
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
 #define PLIST_LITTLE_ENDIAN 0
 #define PLIST_BIG_ENDIAN 1
 
