--- libgcrypt-1.2.2-upstream/mpi/config.links	2005-07-29 15:45:42.000000000 +0200
+++ libgcrypt11-1.2.2/mpi/config.links	2006-06-25 13:30:26.876070760 +0200
@@ -221,6 +221,11 @@
 	path="m68k/mc68020 m68k"
 	;;
 
+    ppc620-*-*	    | \
+    powerpc64*-*-*)
+	mpi_sflags="-Wa,-mppc"
+	path="powerpc64"
+	;;
     powerpc*-*-linux*)
 	echo '/* configured for powerpc/ELF */' >>./mpi/asm-syntax.h
 	echo '#define ELF_SYNTAX'               >>./mpi/asm-syntax.h
@@ -264,11 +269,6 @@
 	mpi_sflags="-Wa,-mppc"
 	path="powerpc32"
 	;;
-    ppc620-*-*	    | \
-    powerpc64*-*-*)
-	mpi_sflags="-Wa,-mppc"
-	path="powerpc64"
-	;;
     powerpc*-*-*)
 	mpi_sflags="-Wa,-mppc"
 	path="powerpc32"
