From: Marcus Meissner <meissner@suse.de>
Subject: Added 2 cifs helpers to request-key.conf

For CIFS DFS support

Forwarded: not-needed
Last-Update: 2014-05-18

diff -Naurp keyutils.orig/request-key.conf keyutils/request-key.conf
--- keyutils.orig/request-key.conf
+++ keyutils/request-key.conf
@@ -38,4 +38,6 @@ create  user    debug:*         expired
 create  user    debug:*         revoked         /bin/keyctl reject %k 30 %c %S
 create	user	debug:loop:*	*		|/bin/cat
 create	user	debug:*		*		/usr/share/keyutils/request-key-debug.sh %k %d %c %S
+create	cifs.spnego	*	*		/usr/sbin/cifs.upcall -c %k
+create	dns_resolver	*	*		/usr/sbin/cifs.upcall %k
 negate	*	*		*		/bin/keyctl negate %k 30 %S
