From: Nobuhiro Iwamatsu <iwamatsu@debian.org>
Date: Tue, 23 Aug 2011 09:22:39 +0900
Subject: Avoid setting Intel Architecture specific CFLAGS

Bug-Debian: http://bugs.debian.org/638925
Forwarded: not-needed
Last-Update: 2014-05-18

Index: keyutils-1.5.9/Makefile
===================================================================
--- keyutils-1.5.9.orig/Makefile
+++ keyutils-1.5.9/Makefile
@@ -66,12 +66,10 @@ LNS		:= ln -sf
 
 ifeq ($(origin CFLAGS),undefined)
 ifeq ($(BUILDFOR),32-bit)
-CFLAGS		+= -m32
 LIBDIR		:= /lib
 USRLIBDIR	:= /usr/lib
 else
 ifeq ($(BUILDFOR),64-bit)
-CFLAGS		+= -m64
 LIBDIR		:= /lib64
 USRLIBDIR	:= /usr/lib64
 endif
