diff --git a/mapfile.c b/mapfile.c
index 41599df..5c11fba 100644
--- a/mapfile.c
+++ b/mapfile.c
@@ -356,10 +356,8 @@ void RebuildMap(void)
 	char *homehost = conf_get_homehost(&require_homehost);
 
 	if (homehost == NULL || strcmp(homehost, "<system>")==0) {
-		if (gethostname(sys_hostname, sizeof(sys_hostname)) == 0) {
-			sys_hostname[sizeof(sys_hostname)-1] = 0;
-			homehost = sys_hostname;
-		}
+		sprintf(sys_hostname, "%08lx", (long)gethostid());
+		homehost = sys_hostname;
 	}
 
 	for (md = mdstat ; md ; md = md->next) {
diff --git a/mdadm.c b/mdadm.c
index 93732a8..338b0b4 100644
--- a/mdadm.c
+++ b/mdadm.c
@@ -1250,10 +1250,8 @@ int main(int argc, char *argv[])
 	if (c.homehost == NULL && c.require_homehost)
 		c.homehost = conf_get_homehost(&c.require_homehost);
 	if (c.homehost == NULL || strcasecmp(c.homehost, "<system>")==0) {
-		if (gethostname(sys_hostname, sizeof(sys_hostname)) == 0) {
-			sys_hostname[sizeof(sys_hostname)-1] = 0;
-			c.homehost = sys_hostname;
-		}
+		sprintf(sys_hostname, "%08lx", (long)gethostid());
+		c.homehost = sys_hostname;
 	}
 	if (c.homehost && (!c.homehost[0] || strcasecmp(c.homehost, "<none>") == 0)) {
 		c.homehost = NULL;
