# http://svn.apache.org/viewvc?view=revision&revision=1745863
--- apache2.orig/support/logresolve.c
+++ apache2/support/logresolve.c
@@ -284,7 +284,7 @@ int main(int argc, const char * const ar
              */
             status = apr_sockaddr_info_get(&ipdouble, hostname, ip->family, 0,
                                            0, pline);
-            if (status == APR_SUCCESS ||
+            if (status != APR_SUCCESS ||
                 memcmp(ipdouble->ipaddr_ptr, ip->ipaddr_ptr, ip->ipaddr_len)) {
                 /* Double-lookup failed  */
                 *space = ' ';
