# https://svn.apache.org/r1830419
# http://bugs.debian.org/897218
--- apache2.orig/modules/http2/h2_bucket_beam.c
+++ apache2/modules/http2/h2_bucket_beam.c
@@ -924,6 +924,7 @@ apr_status_t h2_beam_send(h2_bucket_beam
             while (!APR_BRIGADE_EMPTY(sender_bb) && APR_SUCCESS == rv) {
                 if (space_left <= 0) {
                     report_prod_io(beam, force_report, &bl);
+                    r_purge_sent(beam);
                     rv = wait_not_full(beam, block, &space_left, &bl);
                     if (APR_SUCCESS != rv) {
                         break;
