# https://bz.apache.org/bugzilla/show_bug.cgi?id=60487
--- apache2.orig/server/mpm/event/event.c
+++ apache2/server/mpm/event/event.c
@@ -681,7 +681,8 @@ static void ap_start_shutdown(int gracef
         return;
     }
     shutdown_pending = 1;
-    retained->is_graceful = graceful;
+    if (retained)
+        retained->is_graceful = graceful;
 }
 
 /* do a graceful restart if graceful == 1 */
@@ -693,7 +694,8 @@ static void ap_start_restart(int gracefu
         return;
     }
     restart_pending = 1;
-    retained->is_graceful = graceful;
+    if (retained)
+        retained->is_graceful = graceful;
 }
 
 static void sig_term(int sig)
