--- pngconf.h.orig	2004-10-26 13:25:05.000000000 +0200
+++ pngconf.h	2004-10-26 13:26:19.000000000 +0200
@@ -255,11 +255,13 @@
 #      undef _BSD_SOURCE
 #    endif
 #    ifdef _SETJMP_H
-     /* If you encounter a compiler error here, see the explanation
-      * near the end of INSTALL.
-      */
-         __png.h__ already includes setjmp.h;
-         __dont__ include it again.;
+/* Explanation added by debian maintainer Philippe Troin <phil@fifi.org>.
+   There are two versions of setjmp, depending wether or not we compile for
+   BSD. They are incompatible and can cause crashes. The PNG people force
+   here a unique behavior for setjmp. It fou get the error below,
+   then include <png.h> before <setjmp.h>.
+*/
+#error png.h already includes setjmp.h with some additional fixup.
 #    endif
 #  endif /* __linux__ */
 
