From 099fe96e484816df1f7463832d2e5c656fbfdfaa Mon Sep 17 00:00:00 2001
From: Luk Claes <luk@debian.org>
Date: Sat, 19 Apr 2014 11:40:36 +0100
Subject: Fixes FTBFS with --no-add-needed.

Bug-Debian: http://bugs.debian.org/554059
Signed-off-by: Dimitri John Ledkov <xnox@debian.org>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index da05197..0120815 100644
--- a/Makefile
+++ b/Makefile
@@ -25,7 +25,7 @@ TESTS = fsck-tests.sh
 INSTALL = install
 prefix ?= /usr/local
 bindir = $(prefix)/bin
-lib_LIBS = -luuid -lblkid -lm -lz -llzo2 -L.
+lib_LIBS = -luuid -lblkid -lm -lz -llzo2 -lcom_err -L.
 libdir ?= $(prefix)/lib
 incdir = $(prefix)/include/btrfs
 LIBS = $(lib_LIBS) $(libs_static)
