diff --git a/cmds-restore.c- b/cmds-restore.c
index 1748262..94eee39 100644
--- a/cmds-restore.c-
+++ b/cmds-restore.c
@@ -407,17 +407,19 @@ static int ask_to_continue(const char *file)
 	char *ret;
 
 	printf("We seem to be looping a lot on %s, do you want to keep going "
-	       "on ? (y/N): ", file);
+	       "on ? (y/N/a[lways]): ", file);
 again:
 	ret = fgets(buf, 2, stdin);
 	if (*ret == '\n' || tolower(*ret) == 'n')
-		return 1;
+		return 0;
+	if (tolower(*ret) == 'a')
+		return 2;
 	if (tolower(*ret) != 'y') {
 		printf("Please enter either 'y' or 'n': ");
 		goto again;
 	}
 
-	return 0;
+	return 1;
 }
 
 
@@ -585,11 +587,14 @@ static int copy_file(struct btrfs_root *root, int fd, struct btrfs_key *key,
 	}
 
 	while (1) {
-		if (loops++ >= 1024) {
+		if (loops >= 0 && loops++ >= 1024) {
 			ret = ask_to_continue(file);
-			if (ret)
+			if (ret == 0)
 				break;
-			loops = 0;
+			else if (ret == 1)
+				loops = 0;
+			else if (ret == 2)
+				loops = -1;
 		}
 		if (path->slots[0] >= btrfs_header_nritems(leaf)) {
 			do {
