--- btrfs-tools-3.14.2.orig/cmds-check.c
+++ btrfs-tools-3.14.2/cmds-check.c
@@ -6446,7 +6446,9 @@ const char * const cmd_check_usage[] = {
 	"-s|--super <superblock>     use this superblock copy",
 	"-b|--backup                 use the backup root copy",
 	"--repair                    try to repair the filesystem",
+#if 0
 	"--init-csum-tree            create a new CRC tree",
+#endif
 	"--init-extent-tree          create a new extent tree",
 #ifdef __arm__
 	"--fix-crc                   try to fix bad CRCs",
