diff -Naur avahi-0.6.22/avahi-daemon/avahi-daemon.conf avahi-0.6.22.patched/avahi-daemon/avahi-daemon.conf
--- avahi-0.6.22/avahi-daemon/avahi-daemon.conf	2007-05-09 00:51:01.000000000 +0200
+++ avahi-0.6.22.patched/avahi-daemon/avahi-daemon.conf	2008-04-26 13:17:07.000000000 +0200
@@ -23,9 +23,9 @@
 [server]
 #host-name=foo
 #domain-name=local
-browse-domains=0pointer.de, zeroconf.org
+#browse-domains=0pointer.de, zeroconf.org
 use-ipv4=yes
-use-ipv6=no
+use-ipv6=yes
 #check-response-ttl=no
 #use-iff-running=no
 #enable-dbus=yes
