From: Mark Baker <mark@mnb.org.uk>
Description: Change soname to what debian use
--- a/configure.ac
+++ b/configure.ac
@@ -17,10 +17,10 @@
 # 50 lines of this file. Please update that if the variables above are moved.
 
 # Libtool shared library interface versions (current:revision:age)
-m4_define(libpcre_version, [3:3:2])
-m4_define(libpcre16_version, [2:3:2])
-m4_define(libpcre32_version, [0:3:0])
-m4_define(libpcreposix_version, [0:2:0])
+m4_define(libpcre_version, [16:1:13])
+m4_define(libpcre16_version, [16:1:13])
+m4_define(libpcre32_version, [16:1:13])
+m4_define(libpcreposix_version, [16:1:13])
 m4_define(libpcrecpp_version, [0:0:0])
 
 AC_PREREQ(2.57)
