From: Chris Butler <chrisb@debian.org>
Date: Fri, 8 Apr 2011 01:10:45 +0100
Subject: fix g++-4.3 FTBFS

---
 cgicc/HTMLElement.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/cgicc/HTMLElement.cpp b/cgicc/HTMLElement.cpp
index a76b05c..c0e2ed2 100644
--- a/cgicc/HTMLElement.cpp
+++ b/cgicc/HTMLElement.cpp
@@ -30,6 +30,8 @@
 #include "cgicc/HTMLElement.h"
 #include "cgicc/HTMLElementList.h"
 
+#include <cstring>
+
 // ============================================================
 // Class HTMLElement
 // ============================================================
-- 
