From: Chris Butler <chrisb@debian.org>
Date: Fri, 8 Apr 2011 01:11:20 +0100
Subject: fix autoconf/automake files to fit policy

---
 configure.ac    |   12 ++----------
 doc/Makefile.am |    2 +-
 2 files changed, 3 insertions(+), 11 deletions(-)

diff --git a/configure.ac b/configure.ac
index 23ae071..8477b45 100644
--- a/configure.ac
+++ b/configure.ac
@@ -77,10 +77,6 @@ AC_MSG_CHECKING(whether to build the cgicc demos)
 AC_MSG_RESULT($cgicc_demos)
 AM_CONDITIONAL(DEMOS, test "$cgicc_demos" = yes)
 
-if test "$cgicc_demos" = yes; then
-  AC_CONFIG_FILES(demo/Makefile)
-fi
-
 dnl enable/disable the FastCgi demos, if specified
 AC_ARG_ENABLE(fcgi-demos, 
   [  --enable-fcgi-demos     build FastCgi demos [[default=no]]],
@@ -99,13 +95,9 @@ if test "$cgicc_fcgi_demos" = yes; then
 fi
 AC_SUBST(FCGI_LIBS)
 
-dnl Output contrib/Makefile if either of the demos are enabled
-if test "$cgicc_demos" = yes || test "$cgicc_fcgi_demos" = yes; then
-  AC_CONFIG_FILES(contrib/Makefile)
-fi
-
 dnl Always output certain files
 AC_CONFIG_FILES([Makefile cgicc/Makefile cgicc/CgiDefs.h 
-  support/Makefile support/cgicc-config doc/Makefile doc/Doxyfile])
+  support/Makefile support/cgicc-config doc/Makefile doc/Doxyfile
+	demo/Makefile contrib/Makefile support/cgicc.pc])
 
 AC_OUTPUT
diff --git a/doc/Makefile.am b/doc/Makefile.am
index 7071315..11d3ff1 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -2,7 +2,7 @@
 
 CLEANFILES = *~ stamp
 
-docdir = $(prefix)/doc/$(PACKAGE)-$(VERSION)
+# docdir = $(prefix)/doc/$(PACKAGE)-$(VERSION)
 
 FILES = index.tmpl license.tmpl lists.tmpl bugs.tmpl demos.tmpl		\
 cgi-overview.tmpl tutorial.tmpl lib-overview.tmpl COPYING COPYING.LIB	\
-- 
