diff -ruN newt-0.52.11-old/Makefile.in newt-0.52.11/Makefile.in
--- newt-0.52.11-old/Makefile.in	2009-04-20 21:44:14.000000000 +0100
+++ newt-0.52.11/Makefile.in	2009-04-20 21:49:26.000000000 +0100
@@ -1,10 +1,10 @@
 LIBS = -lslang
-LIBTCL = -ltcl8.4
+LIBTCL = -ltcl8.5
 
 CC = @CC@
 CPP = @CPP@
 CFLAGS = @CFLAGS@
-CPPFLAGS = -D_GNU_SOURCE -I/usr/include/slang @CPPFLAGS@ 
+CPPFLAGS = -D_GNU_SOURCE -I/usr/include/slang -I/usr/include/tcl8.5@CPPFLAGS@ 
 
 SHLIBFLAGS= -Wl,--version-script,newt.0.52.ver
 VERSION = @VERSION@
@@ -49,7 +49,7 @@
 TARGET=depend $(PROGS)
 endif
 
-all:	$(TARGET) _snackmodule.so
+all:	$(TARGET) _snackmodule.so whiptcl.so
 
 test:	test.o $(LIBNEWT)
 	$(CC) -g -o test test.o $(LIBNEWT) $(LIBS)
