diff -ruN newt-0.52.11-old/Makefile.in newt-0.52.11/Makefile.in
--- newt-0.52.11-old/Makefile.in	2009-04-30 14:37:20.000000000 +0200
+++ newt-0.52.11/Makefile.in	2009-04-30 14:39:08.000000000 +0200
@@ -72,6 +72,8 @@
 	    	mkdir -p $$ver ;\
 	        $(CC) $(CFLAGS) -I/usr/include/$$ver $(SHCFLAGS) -c -o $$ver/snackmodule.o snackmodule.c ;\
 		$(CC) --shared $(SHCFLAGS) -o $$ver/_snackmodule.so $$ver/snackmodule.o -L .  -lnewt ;\
+	        $(CC) $(subst -O2,-O0,$(CFLAGS)) -I/usr/include/$${ver}_d -fPIC -c -o $$ver/snackmodule_d.o snackmodule.c ;\
+		$(CC) --shared $(SHCFLAGS) -o $$ver/_snackmodule_d.so $$ver/snackmodule_d.o -L .  -lnewt  ;\
 	done
 	touch $@
 
@@ -130,6 +132,7 @@
 	for ver in $(PYTHONVERS) ; do \
 	   [ -d $(instroot)/$(libdir)/$$ver/site-packages ] || install -m 755 -d $(instroot)/$(libdir)/$$ver/site-packages ;\
 	   install -m 755 $$ver/_snackmodule.so $(instroot)/$(libdir)/$$ver/site-packages ;\
++	   install -m 755 $$ver/_snackmodule_d.so $(instroot)/$(libdir)/$$ver/site-packages ;\
 	   install -m 644 snack.py $(instroot)/$(libdir)/$$ver/site-packages ;\
 	done
 
