From: Alberto Gonzalez Iniesta <agi@inittab.org>

--- a/utils/gssd/svcgssd.man	2010-02-18 23:35:00.000000000 +1100
+++ b/utils/gssd/svcgssd.man	2010-04-06 16:48:19.000000000 +1000
@@ -35,6 +35,11 @@ increases the verbosity of the output (c
 .B -i
 If the nfsidmap library supports setting debug level,
 increases the verbosity of the output (can be specified multiple times).
+.TP
+.B -n
+Do not search for nfs/<your.host>@<YOUR.REALM> at start up, but instead
+accept requests for any key in the keytab file. Useful for HA clusters
+where the NFS service hostname is different from the node hostname.
 
 .SH SEE ALSO
 .BR rpc.gssd(8),
