From c72cd53e4c8f96f3608be1ccbd8f2d68c540e02b Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard10@gmail.com>
Date: Fri, 30 Jan 2015 12:06:35 -0800
Subject: [PATCH 03/19] 05_share_ldb_module

---
 source4/param/wscript_build | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/source4/param/wscript_build b/source4/param/wscript_build
index 4585a83..3b845b7 100644
--- a/source4/param/wscript_build
+++ b/source4/param/wscript_build
@@ -18,7 +18,7 @@ bld.SAMBA_MODULE('share_classic',
 	source='share_classic.c',
 	subsystem='share',
 	init_function='share_classic_init',
-	deps='samba-util'
+	deps='samba-util samba-hostconfig'
 	)
 
 
@@ -26,7 +26,8 @@ bld.SAMBA_MODULE('share_ldb',
 	source='share_ldb.c',
 	subsystem='share',
 	init_function='share_ldb_init',
-	deps='ldbsamba auth_system_session'
+	deps='ldbsamba auth_system_session',
+	internal_module=False,
 	)
 
 
-- 
1.9.1

