Fix compilation warnings

Author: Hai Zaar <haizaar@haizaar.com>

--- a/main.c.orig	2012-10-26 17:53:41.000000000 -0700
+++ b/main.c	2012-10-26 17:56:53.186848331 -0700
@@ -43,14 +43,16 @@ int haveuid= 0;
 int server_uids[]= {SERVER_UIDS, 0};
 #endif
 
+void snooze(int seconds); /* defined in snooze.c */
+int check_auth(char *login, char *passwd); /* defined in auth_*.c */
 
-main(int argc, char **argv)
+int main(int argc, char **argv)
 {
 #ifdef ENV_METHOD
     char *login, *passwd;
 #else
     char login[BFSZ+1], passwd[BFSZ+1];
-    char *c, *strchr();
+    char *c;
 #endif
     int uid,i;
     int status;
