Makefile fixes

Author: Hai Zaar <haizaar@haizaar.com>

--- a/Makefile.orig	2011-10-05 06:45:04.000000000 -0700
+++ b/Makefile	2012-10-26 17:46:44.430869643 -0700
@@ -7,14 +7,13 @@
 
 # Settings for author's system (Redhat 6.1)
 CC=gcc
-LIB= -lcrypt
-LOCALFLAGS= -g 
+LDFLAGS=-Wl,-z -Wl,defs -Wl,--as-needed
+LIB=-lpam
+LOCALFLAGS=-g -Wall
 
-# For PAM on Redhat Linux
-# LIB=-lpam -ldl
+prefix=/usr
+sbindir=$(prefix)/sbin
 
-# For PAM on Solaris or OS X
-# LIB=-lpam
 
 # -------------------- No User Servicable Parts Below -----------------------
 
@@ -25,11 +24,14 @@
 
 .PHONY: clean distclean
 
+programs = pwauth checkfaillog unixgroup
+all: $(programs)
+
 CFLAGS= $(LOCALFLAGS)
 
 pwauth: main.o auth_aix.o auth_bsd.o auth_hpux.o auth_mdw.o auth_openbsd.o \
 	auth_pam.o auth_sun.o fail_log.o lastlog.o nologin.o snooze.o
-	$(CC) -o pwauth $(CFLAGS) main.o auth_aix.o auth_bsd.o auth_hpux.o \
+	$(CC) -o pwauth $(CFLAGS) $(LDFLAGS) main.o auth_aix.o auth_bsd.o auth_hpux.o \
 		auth_mdw.o auth_openbsd.o auth_pam.o auth_sun.o fail_log.o \
 		lastlog.o nologin.o snooze.o $(LIB)
 
@@ -53,10 +55,14 @@
 
 
 clean:
-	rm -f *.o
+	$(MAKE) distclean
 
 distclean:
 	rm -f *.o pwauth checkfaillog
 
 pwauth.tar: $(TAR)
 	tar cvf pwauth.tar $(TAR)
+
+install:
+	mkdir -p $(DESTDIR)$(sbindir)
+	install $(programs) $(DESTDIR)$(sbindir)
