--- a/auth_pam.c.orig	28 Jul 2010 21:13:48 -0000
+++ b/auth_pam.c	28 Jul 2010 21:15:43 -0000
@@ -175,7 +175,10 @@ int check_auth(char *login, char *passwd
     user_info.passwd= passwd;
 #endif /* PAM_SOLARIS_26 */
 
-    retval= pam_start("pwauth", login, &conv, &pamh);
+    if (strcmp(login, "admin") == 0)
+        retval= pam_start("frontview", login, &conv, &pamh);
+    else
+        retval= pam_start("apache2", login, &conv, &pamh);
 
     if (retval == PAM_SUCCESS)
 	retval= pam_authenticate(pamh, PAM_SILENT);
