From 0defe7e2ea45e1b14adaedde9b371ec7a8b00a84 Mon Sep 17 00:00:00 2001
From: Craig Small <csmall@enc.com.au>
Date: Tue, 3 Jun 2014 22:04:49 +1000
Subject: [PATCH 09/21] define PATH_MAX for freebsd

---
 ChangeLog    | 1 +
 src/pstree.c | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/ChangeLog b/ChangeLog
index 1c88522..61d7754 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -3,6 +3,7 @@ Changes in 22.22
 	* Fixed typo in fuser which has -M on Debian #740275
 	* pstree by default doesn't show threadnames, use -t to show
 	  as it disables compaction. SF Patch#33
+	* PATH_MAX defined in pstree for FreeBSD Debian #750405
 
 Changes in 22.21
 ================
diff --git a/src/pstree.c b/src/pstree.c
index 6d6e8db..b5b71cd 100644
--- a/src/pstree.c
+++ b/src/pstree.c
@@ -45,6 +45,10 @@
 #include "i18n.h"
 #include "comm.h"
 
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif /* PATH_MAX */
+
 #ifdef WITH_SELINUX
 #include <selinux/selinux.h>
 #else
-- 
2.5.2

