From acbd2f4986f5598862cc6fc65ba059aaea5457f9 Mon Sep 17 00:00:00 2001
From: Craig Small <csmall@enc.com.au>
Date: Tue, 3 Jun 2014 22:14:06 +1000
Subject: [PATCH 11/21] Note S patch #34 for pstree

---
 ChangeLog | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ChangeLog b/ChangeLog
index 61d7754..f2a01f6 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -4,6 +4,7 @@ Changes in 22.22
 	* pstree by default doesn't show threadnames, use -t to show
 	  as it disables compaction. SF Patch#33
 	* PATH_MAX defined in pstree for FreeBSD Debian #750405
+	* pstree ignores disappeared processes SF Patch#34
 
 Changes in 22.21
 ================
-- 
2.5.2

