--- a/modules/mod_core.c	2010-03-11 22:33:56.000000000 +0000
+++ b/modules/mod_core.c	2010-03-11 22:34:51.000000000 +0000
@@ -297,6 +297,7 @@ MODRET add_include(cmd_rec *cmd) {
   CHECK_ARGS(cmd, 1);
   CHECK_CONF(cmd, CONF_ROOT|CONF_VIRTUAL|CONF_ANON|CONF_GLOBAL|CONF_DIR);
 
+  PRIVS_ROOT
   /* Make sure the given path is a valid path. */
   if (pr_fs_valid_path(cmd->argv[1]) < 0) {
     CONF_ERROR(cmd, pstrcat(cmd->tmp_pool,
@@ -313,6 +314,7 @@ MODRET add_include(cmd_rec *cmd) {
         "': ", strerror(errno), NULL));
     }
   }
+  PRIVS_RELINQUISH
 
   return PR_HANDLED(cmd);
 }
