Index: libedit/src/readline.c
===================================================================
--- libedit.orig/src/readline.c
+++ libedit/src/readline.c
@@ -1,4 +1,4 @@
-/*	$NetBSD: readline.c,v 1.110 2014/01/21 13:51:44 christos Exp $	*/
+/*	$NetBSD: readline.c,v 1.111 2014/07/06 18:09:04 christos Exp $	*/
 
 /*-
  * Copyright (c) 1997 The NetBSD Foundation, Inc.
@@ -31,7 +31,7 @@
 
 #include "config.h"
 #if !defined(lint) && !defined(SCCSID)
-__RCSID("$NetBSD: readline.c,v 1.110 2014/01/21 13:51:44 christos Exp $");
+__RCSID("$NetBSD: readline.c,v 1.111 2014/07/06 18:09:04 christos Exp $");
 #endif /* not lint && not SCCSID */
 
 #include <sys/types.h>
@@ -1966,7 +1966,7 @@ rl_callback_read_char(void)
 		} else
 			wbuf = NULL;
 		(*(void (*)(const char *))rl_linefunc)(wbuf);
-		//el_set(e, EL_UNBUFFERED, 1);
+		el_set(e, EL_UNBUFFERED, 1);
 	}
 }
 
