Index: libedit/configure.ac
===================================================================
--- libedit.orig/configure.ac	2014-07-06 19:09:05.727059614 +0200
+++ libedit/configure.ac	2014-07-06 19:09:19.119442966 +0200
@@ -35,7 +35,7 @@
 AC_PROG_LIBTOOL
 
 # libtool -version-info
-AC_SUBST(LT_VERSION, [0:51:0])
+AC_SUBST(LT_VERSION, [2:51:0])
 
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
