From: Gert Michael Kulyk <gkulyk@yahoo.de>
Subject: This patch fixes the build flags, so libdbi can be built in
 Debian again.

Index: libdbi-0.8.3+really0.8.2/configure
===================================================================
--- libdbi-0.8.3+really0.8.2.orig/configure	2011-04-24 13:12:32.163876930 +0200
+++ libdbi-0.8.3+really0.8.2/configure	2011-04-24 13:14:20.472414007 +0200
@@ -19257,8 +19257,8 @@
         case $host in
         *-*-linux*)
                 DEBUG="-g -Wall -D_REENTRANT -D__NO_MATH_INLINES -fsigned-char -std=gnu99"
-                CFLAGS="-O20 -ffast-math -D_REENTRANT -fsigned-char -std=gnu99"
-                PROFILE="-pg -g -O20 -ffast-math -D_REENTRANT -fsigned-char -std=gnu99";;
+                CFLAGS="-D_REENTRANT -fsigned-char -std=gnu99"
+                PROFILE="-pg -g -D_REENTRANT -fsigned-char -std=gnu99";;
         sparc-sun-*)
                 DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char -mv8 -std=gnu99"
                 CFLAGS="-O20 -ffast-math -D__NO_MATH_INLINES -fsigned-char -mv8 -std=gnu99"
Index: libdbi-0.8.3+really0.8.2/configure.in
===================================================================
--- libdbi-0.8.3+really0.8.2.orig/configure.in	2011-04-24 13:12:37.351902651 +0200
+++ libdbi-0.8.3+really0.8.2/configure.in	2011-04-24 13:13:31.296170140 +0200
@@ -55,8 +55,8 @@
         case $host in
         *-*-linux*)
                 DEBUG="-g -Wall -D_REENTRANT -D__NO_MATH_INLINES -fsigned-char -std=gnu99"
-                CFLAGS="-O20 -ffast-math -D_REENTRANT -fsigned-char -std=gnu99"
-                PROFILE="-pg -g -O20 -ffast-math -D_REENTRANT -fsigned-char -std=gnu99";;
+                CFLAGS="-D_REENTRANT -fsigned-char -std=gnu99"
+                PROFILE="-pg -g -D_REENTRANT -fsigned-char -std=gnu99";;
         sparc-sun-*)
                 DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char -mv8 -std=gnu99"
                 CFLAGS="-O20 -ffast-math -D__NO_MATH_INLINES -fsigned-char -mv8 -std=gnu99"
