--- a/prog/pwm/pwmconfig
+++ b/prog/pwm/pwmconfig
@@ -461,8 +461,8 @@
 				fanactive="$j ${fanactive}"
 				fanactive_min="$S ${fanactive_min}"
 			else
-				fanactive="$j+${fanactive}" #not supported yet by fancontrol
-				fanactive_min="$S+${fanactive_min}"
+				fanactive="$(($j+${fanactive}))" #not supported yet by fancontrol
+				fanactive_min="$(($S+${fanactive_min}))"
 			fi
 			sleep $DELAY
 			if [ $? -ne 0 ]
