--- a/configure.in
+++ b/configure.in
@@ -1445,11 +1445,7 @@
 fi
 
 ################################################################################
-AC_PATH_TOOL(MODPROBE_CMD, modprobe)
-
-if test -n "$MODPROBE_CMD"; then
-	AC_DEFINE_UNQUOTED([MODPROBE_CMD], ["$MODPROBE_CMD"], [The path to 'modprobe', if available.])
-fi
+AC_DEFINE_UNQUOTED([MODPROBE_CMD], ["/sbin/modprobe"], [The path to 'modprobe', if available.])
 
 
 lvm_exec_prefix=$exec_prefix
