From 945d41b4fc141d8263bcf76f6e66e5d05908f9e8 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard10@gmail.com>
Date: Wed, 16 Apr 2014 10:11:52 -0700
Subject: [PATCH 2/2] main: default to no backtrace, and let the OS core dump
 instead

---
 src/main.c |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/src/main.c b/src/main.c
index 4f635de..0ba75b6 100644
--- a/src/main.c
+++ b/src/main.c
@@ -457,7 +457,7 @@ static gchar *option_noplugin = NULL;
 static gchar *option_wifi = NULL;
 static gboolean option_detach = TRUE;
 static gboolean option_dnsproxy = TRUE;
-static gboolean option_backtrace = TRUE;
+static gboolean option_backtrace = FALSE;
 static gboolean option_version = FALSE;
 
 static bool parse_debug(const char *key, const char *value,
@@ -494,9 +494,8 @@ static GOptionEntry options[] = {
 	{ "nodnsproxy", 'r', G_OPTION_FLAG_REVERSE,
 				G_OPTION_ARG_NONE, &option_dnsproxy,
 				"Don't enable DNS Proxy" },
-	{ "nobacktrace", 0, G_OPTION_FLAG_REVERSE,
-				G_OPTION_ARG_NONE, &option_backtrace,
-				"Don't print out backtrace information" },
+	{ "backtrace", 0, 0, G_OPTION_ARG_NONE, &option_backtrace,
+				"Print out backtrace information" },
 	{ "version", 'v', 0, G_OPTION_ARG_NONE, &option_version,
 				"Show version information and exit" },
 	{ NULL },
-- 
1.7.10.4

