From 3490c9d4bae627f104e0cfb4893c95b931196e92 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard10@gmail.com>
Date: Wed, 15 Oct 2014 15:53:17 -0700
Subject: [PATCH] disable-unused-features.patch

---
 src/main.c |   14 --------------
 1 file changed, 14 deletions(-)

diff --git a/src/main.c b/src/main.c
index 21d1e06..ba2cee9 100644
--- a/src/main.c
+++ b/src/main.c
@@ -652,10 +652,6 @@ int main(int argc, char *argv[])
 	__connman_device_init(option_device, option_nodevice);
 
 	__connman_ippool_init();
-	__connman_iptables_init();
-	__connman_firewall_init();
-	__connman_nat_init();
-	__connman_tethering_init();
 	__connman_counter_init();
 	__connman_manager_init();
 	__connman_stats_init();
@@ -676,9 +672,6 @@ int main(int argc, char *argv[])
 	__connman_rtnl_start();
 	__connman_dhcp_init();
 	__connman_dhcpv6_init();
-	__connman_wpad_init();
-	__connman_wispr_init();
-	__connman_rfkill_init();
 	__connman_machine_init();
 
 	g_free(option_config);
@@ -692,9 +685,6 @@ int main(int argc, char *argv[])
 	g_source_remove(signal);
 
 	__connman_machine_cleanup();
-	__connman_rfkill_cleanup();
-	__connman_wispr_cleanup();
-	__connman_wpad_cleanup();
 	__connman_dhcpv6_cleanup();
 	__connman_session_cleanup();
 	__connman_plugin_cleanup();
@@ -712,10 +702,6 @@ int main(int argc, char *argv[])
 	__connman_config_cleanup();
 	__connman_manager_cleanup();
 	__connman_counter_cleanup();
-	__connman_tethering_cleanup();
-	__connman_nat_cleanup();
-	__connman_firewall_cleanup();
-	__connman_iptables_cleanup();
 	__connman_peer_service_cleanup();
 	__connman_peer_cleanup();
 	__connman_ippool_cleanup();
-- 
1.7.10.4

