diff --git a/src/dnsproxy.c- b/src/dnsproxy.c
index 741cd45..cd29024 100644
--- a/src/dnsproxy.c-
+++ b/src/dnsproxy.c
@@ -2602,12 +2602,8 @@ static struct server_data *create_server(int index,
 	}
 
 	if (protocol == IPPROTO_UDP) {
-		if (__connman_service_index_is_default(data->index) ||
-				__connman_service_index_is_split_routing(
-								data->index)) {
-			data->enabled = true;
-			DBG("Adding DNS server %s", data->server);
-		}
+		data->enabled = true;
+		DBG("Adding DNS server %s", data->server);
 
 		server_list = g_slist_append(server_list, data);
 	}
