--- connman-1.21.orig/src/connection.c
+++ connman-1.21/src/connection.c
@@ -968,10 +968,10 @@ void __connman_connection_gateway_remove
 		return;
 
 	if (do_ipv4 && data->ipv4_gateway)
-		set_default4 = data->ipv4_gateway->vpn;
+		set_default4 = true;
 
 	if (do_ipv6 && data->ipv6_gateway)
-		set_default6 = data->ipv6_gateway->vpn;
+		set_default6 = true;
 
 	DBG("ipv4 gateway %s ipv6 gateway %s vpn %d/%d",
 		data->ipv4_gateway ? data->ipv4_gateway->gateway : "<null>",
