Description: Spell error
Author: Herbert Parentes Fortes Neto <hpfn@ig.com.br>
Last-Update: 2015-07-11
Index: libgphoto2-2.5.8/camlibs/sierra/epson-desc.c
===================================================================
--- libgphoto2-2.5.8.orig/camlibs/sierra/epson-desc.c
+++ libgphoto2-2.5.8/camlibs/sierra/epson-desc.c
@@ -237,7 +237,7 @@ static const ValueNameType ep3000z_reg_5
 	{ { 0x06 }, N_("Italian") },
 	{ { 0x07 }, N_("Japanese") },
 	{ { 0x08 }, N_("Spanish") },
-	{ { 0x09 }, N_("Portugese") },
+	{ { 0x09 }, N_("Portuguese") },
 
 };
 static const RegisterDescriptorType ep3000z_reg_53[] = {
