diff --git a/modules/pam_tally2/pam_tally2.c b/modules/pam_tally2/pam_tally2.c
index 09e8585..1d2350e 100644
--- a/modules/pam_tally2/pam_tally2.c
+++ b/modules/pam_tally2/pam_tally2.c
@@ -583,7 +583,11 @@ tally_check (tally_t oldcnt, time_t oldtime, pam_handle_t *pamh, uid_t uid,
                    "user %s (%lu) tally %hu, deny %hu",
 		   user, (unsigned long)uid, tally->fail_cnt, opts->deny);
 	}
-        rv = PAM_AUTH_ERR;                 /* Only unconditional failure   */
+        if (opts->deny != 0 && tally->fail_cnt > opts->deny) {
+          rv = PAM_MAXTRIES;
+        } else {
+          rv = PAM_AUTH_ERR;                 /* Only unconditional failure   */
+        }
         goto cleanup;
     }
 
