Index: b/Makefile.in
===================================================================
--- a/Makefile.in	2013-05-08 09:39:45.925902114 +0000
+++ b/Makefile.in	2013-05-08 09:40:17.986114783 +0000
@@ -116,7 +116,7 @@
 VSNFLAG = -version-info @LIBCURRENT@:@LIBREVISION@:@LIBAGE@
 
 ### autoconf this?
-LTFLAGS = --silent
+LTFLAGS = --verbose
 
 COMPILE = $(CC) $(INCLUDES) $(CFLAGS) $(DEFS) $(CPPFLAGS)
 CXXCOMPILE = $(CXX) $(INCLUDES) $(CXXFLAGS) $(DEFS) $(CPPFLAGS)
