commit f59d7b2eea1c1498d1a6cbd914608874b1621fc3
Author: Michael Li <michael.li@netgear.com>
Date:   Wed Sep 5 17:51:17 2018 +0800

    Increase default queue size to 1024.

diff --git a/init.d/audispd.conf b/init.d/audispd.conf
index 4471c6c..066e5b4 100644
--- a/init.d/audispd.conf
+++ b/init.d/audispd.conf
@@ -3,7 +3,7 @@
 # dispatcher daemon, audispd.
 #
 
-q_depth = 250
+q_depth = 1024
 overflow_action = SYSLOG
 priority_boost = 4
 max_restarts = 10
