Index: src/io.c
===================================================================
--- src/io.c	(revision 1676)
+++ src/io.c	(revision 1677)
@@ -2148,9 +2148,6 @@
     struct sockaddr_in client;
     SOCKET_T child_fd;
     IO_SOCKET_PRIV *priv;
-#ifdef WIN32
-    long blocking = 0;
-#endif
 
     ASSERT(phandle);
     ASSERT(pchild);
@@ -2179,12 +2176,6 @@
         return FALSE;
     }
 
-#ifdef WIN32
-    if(ioctlsocket(child_fd,FIONBIO,&blocking)) {
-        io_err_printf(IO_LOG_LOG,"Couldn't set socket to blocking\n");
-    }
-#endif
-
     io_err_printf(IO_LOG_DEBUG,"Got listen socket %d\n",child_fd);
 
     /* copy host, if passed a buffer */
@@ -2289,6 +2280,7 @@
     ssize_t byteswritten=0;
     uint32_t totalbytes;
     unsigned char *bufp;
+    long blocking = 0;
 
     ASSERT(phandle);
     ASSERT(phandle->private);
@@ -2321,9 +2313,11 @@
 
 #ifdef WIN32
         if(WSAGetLastError() == WSAEWOULDBLOCK) {
-            io_err_printf(IO_LOG_DEBUG,"WSAEWOULDBLOCK hack\n");
             byteswritten = 0;
-            Sleep(50);
+
+            if(ioctlsocket(priv->fd,FIONBIO,&blocking)) {
+                io_err_printf(IO_LOG_LOG,"Couldn't set socket to blocking\n");
+            }
         }
 #endif
         if((byteswritten == -1 ) && (errno != EINTR)) {
