Index: src/webserver.c
===================================================================
--- src/webserver.c	(revision 1677)
+++ src/webserver.c	(revision 1678)
@@ -815,7 +815,7 @@
             last=first;
             strsep(&last,":");
 
-            if(last==first) {
+            if(!last) {
                 ws_dprintf(L_WS_WARN,"Thread %d: Invalid header: %s\n",
                         pwsc->threadno,first);
             } else {
