--- trunk/src/webserver.c	2007/10/21 22:45:32	1679
+++ trunk/src/webserver.c	2007/10/21 22:51:01	1680
@@ -1134,8 +1134,8 @@
                 if((auth) && (ws_decodepassword(auth,&username, &password))) {
                     if(auth_handler(pwsc,username,password))
                         can_dispatch=1;
-                    ws_addarg(&pwsc->request_vars,"HTTP_USER",username);
-                    ws_addarg(&pwsc->request_vars,"HTTP_PASSWD",password);
+                    ws_addarg(&pwsc->request_vars,"HTTP_USER","%s",username);
+                    ws_addarg(&pwsc->request_vars,"HTTP_PASSWD","%s",password);
                     free(username); /* this frees password too */
                 }
 
