diff --git a/src/core/mount.c.orig b/src/core/mount.c
index 05cbaba..b1c6915 100644
--- a/src/core/mount.c.orig
+++ b/src/core/mount.c
@@ -1002,6 +1002,24 @@ fail:
         mount_enter_mounted(m, MOUNT_FAILURE_RESOURCES);
 }
 
+static int mount_ro(void) {
+	FILE *fp;
+	char buf[16];
+	int ret = 0;
+
+	fp = fopen("/dev/boot_reason", "r");
+	if (!fp)
+		return 0;
+	if (fgets(buf, sizeof(buf), fp) != NULL)
+	{
+		if (strcmp(buf, "read_only") == 0)
+			ret = 1;
+	}
+	fclose(fp);
+
+	return ret;
+}
+
 static void mount_enter_mounting(Mount *m) {
         int r;
         MountParameters *p;
@@ -1035,6 +1053,7 @@ static void mount_enter_mounting(Mount *m) {
                                 m->control_command,
                                 "/bin/mount",
                                 m->where,
+                                mount_ro() ? "-o" : NULL, "ro",
                                 NULL);
         else
                 r = -ENOENT;
