Description: add missing link with gpg-error library
Author: Laszlo Boszormenyi (GCS) <gcs@debian.org>
Last-Update: 2014-10-05

---

--- ntfs-3g-2014.2.15AR.2.orig/ntfsprogs/Makefile.am
+++ ntfs-3g-2014.2.15AR.2/ntfsprogs/Makefile.am
@@ -131,7 +131,7 @@ ntfsfallocate_LDFLAGS	= $(AM_LFLAGS)
 if ENABLE_CRYPTO
 ntfsdecrypt_SOURCES	= ntfsdecrypt.c utils.c utils.h
 ntfsdecrypt_LDADD	= $(AM_LIBS) $(GNUTLS_LIBS) $(LIBGCRYPT_LIBS)
-ntfsdecrypt_LDFLAGS	= $(AM_LFLAGS)
+ntfsdecrypt_LDFLAGS	= $(AM_LFLAGS) -lgpg-error
 ntfsdecrypt_CFLAGS	= $(GNUTLS_CFLAGS) $(LIBGCRYPT_CFLAGS)
 endif
 
