diff -ru net-snmp-5.2.3.orig/acconfig.h net-snmp-5.2.3/acconfig.h
--- net-snmp-5.2.3.orig/acconfig.h	2006-05-26 18:37:05.000000000 +0200
+++ net-snmp-5.2.3/acconfig.h	2006-08-01 11:18:38.000000000 +0200
@@ -617,6 +617,9 @@
 /* define if you have getservbyname */
 #undef HAVE_GETSERVBYNAME
 
+/* define if you have dlopen */
+#undef HAVE_DLOPEN
+
 /* printing system */
 #undef HAVE_LPSTAT
 #undef LPSTAT_PATH
diff -ru net-snmp-5.2.3.orig/agent/Makefile.in net-snmp-5.2.3/agent/Makefile.in
--- net-snmp-5.2.3.orig/agent/Makefile.in	2005-12-06 18:51:17.000000000 +0100
+++ net-snmp-5.2.3/agent/Makefile.in	2006-08-01 11:18:38.000000000 +0200
@@ -98,7 +98,7 @@
 	 agent_handler.o @agentgroup_list_o@ @OTHERAGENTLIBOBJS@
 LLIBAGENTOBJS=snmp_agent.lo snmp_vars.lo agent_read_config.lo \
 	agent_registry.lo agent_index.lo agent_trap.lo kernel.lo \
-	agent_handler.lo @agentgroup_list_lo@ @OTHERAGENTLIBLOBJS@
+	agent_handler.lo @agentgroup_list_lo@ @OTHERAGENTLIBLOBJS@ @WRAPLIBS@
 
 # The agent objects.
 AGENTOBJS=snmpd.o @other_agentobjs@
@@ -137,7 +137,7 @@
 	$(RANLIB) libnetsnmpagent.$(LIB_EXTENSION)$(LIB_VERSION)
 
 libnetsnmpmibs.$(LIB_EXTENSION)$(LIB_VERSION):    ${LMIBOBJS}
-	$(LIB_LD_CMD) libnetsnmpmibs.$(LIB_EXTENSION)$(LIB_VERSION) ${LMIBOBJS}  $(LIB_LD_LIBS)
+	$(LIB_LD_CMD) libnetsnmpmibs.$(LIB_EXTENSION)$(LIB_VERSION) ${LMIBOBJS}  $(LIB_LD_LIBS) $(LIBSENSORS)
 	$(RANLIB) libnetsnmpmibs.$(LIB_EXTENSION)$(LIB_VERSION)
 
 agentlib: libnetsnmpagent.$(LIB_EXTENSION)$(LIB_VERSION)
diff -ru net-snmp-5.2.3.orig/configure.in net-snmp-5.2.3/configure.in
--- net-snmp-5.2.3.orig/configure.in	2006-07-14 01:29:49.000000000 +0200
+++ net-snmp-5.2.3/configure.in	2006-08-01 11:18:38.000000000 +0200
@@ -2193,12 +2193,13 @@
          AC_CHECK_LIB(RSAglue, RSA_PKCS1_RSAref)
       fi
 
-      AC_CHECK_LIB(crypto, EVP_md5)
+      AC_CHECK_LIB(crypto, EVP_md5, [CRYPTOLIBS=-lcrypto], , )
       AC_CHECK_FUNCS(AES_cfb128_encrypt)
     fi
 elif test "x$askedpkcs" = "xyes"; then
     AC_CHECK_LIB(pkcs11, C_Initialize)
 fi
+AC_SUBST(CRYPTOLIBS)
 
 # Check for libraries that the agent needs
 # saving old libraries
@@ -2321,8 +2322,9 @@
 # Replace `main' with a function in -lkvm:
 AC_CHECK_LIB(kvm, kvm_read)
 # DYNAMIC MODULE SUPPORT
-AC_CHECK_LIB(dl, dlopen)
-AC_CHECK_FUNCS(dlopen)
+AC_CHECK_FUNC(dlopen, AC_DEFINE(HAVE_DLOPEN),
+	AC_CHECK_LIB(dl, dlopen, AC_DEFINE(HAVE_DLOPEN)
+	LIBS="${LIBS} -ldl"))
 
 AC_CHECK_FUNCS(kvm_openfiles nlist knlist kvm_getprocs kvm_getswapinfo)
 echo " $module_list " | grep " ucd-snmp/diskio " >/dev/null
@@ -2766,7 +2768,7 @@
 
 AC_MSG_RESULT([$ac_cv_type_ssize_t])
 if test $ac_cv_type_ssize_t = yes; then
-  AC_DEFINE(HAVE_SSIZE_T)
+  AC_DEFINE(HAVE_SSIZE_T, 1, [Define if type ssize_t is available])
 fi
 
 # Check ps args
diff -ru net-snmp-5.2.3.orig/local/Makefile.in net-snmp-5.2.3/local/Makefile.in
--- net-snmp-5.2.3.orig/local/Makefile.in	2005-10-10 20:47:48.000000000 +0200
+++ net-snmp-5.2.3/local/Makefile.in	2006-08-01 11:18:38.000000000 +0200
@@ -97,7 +97,7 @@
 
 mib2c.made: $(srcdir)/mib2c
 	if test "x$(PERL)" != "x" ; then \
-	  $(PERL) -p -e 's%^#!.*/perl.*%#!$(PERL)%;s#/usr/local/share/snmp#$(snmplibdir)#;' ${srcdir}/mib2c > mib2c.made; \
+	  $(PERL) -p -e 's%^#!.*/perl.*%#!$(PERL)%;s#/usr/local/share/snmp#$(snmplibdir)#;s#/usr/local/etc/snmp#$(SNMPCONFPATH)#;' ${srcdir}/mib2c > mib2c.made; \
 	else \
 	  touch mib2c.made; \
         fi
diff -ru net-snmp-5.2.3.orig/Makefile.rules net-snmp-5.2.3/Makefile.rules
--- net-snmp-5.2.3.orig/Makefile.rules	2005-10-10 20:47:48.000000000 +0200
+++ net-snmp-5.2.3/Makefile.rules	2006-08-01 11:18:38.000000000 +0200
@@ -315,7 +315,7 @@
 # cleaning targets
 #
 clean: cleansubdirs $(OTHERCLEANTODOS)
-	rm -f core ${OBJS} ${LOBJS} $(STANDARDCLEANTARGETS) $(OTHERCLEANTARGETS)
+	rm -f -- core ${OBJS} ${LOBJS} $(STANDARDCLEANTARGETS) $(OTHERCLEANTARGETS)
 	rm -rf .libs
 
 cleansubdirs:
diff -ru net-snmp-5.2.3.orig/Makefile.top net-snmp-5.2.3/Makefile.top
--- net-snmp-5.2.3.orig/Makefile.top	2006-06-09 23:49:31.000000000 +0200
+++ net-snmp-5.2.3/Makefile.top	2006-08-01 11:19:04.000000000 +0200
@@ -26,6 +26,7 @@
 snmplibdir	= $(datadir)/snmp
 mibdir		= $(snmplibdir)/mibs
 persistentdir	= @PERSISTENT_DIRECTORY@
+sysconfdir	= @sysconfdir@
 DESTDIR         = @INSTALL_PREFIX@
 INSTALL_PREFIX  = $(DESTDIR)
 
diff -ru net-snmp-5.2.3.orig/mibs/Makefile.in net-snmp-5.2.3/mibs/Makefile.in
--- net-snmp-5.2.3.orig/mibs/Makefile.in	2005-01-26 15:19:10.000000000 +0100
+++ net-snmp-5.2.3/mibs/Makefile.in	2006-08-01 11:18:38.000000000 +0200
@@ -47,11 +47,15 @@
 UCDMIBS = UCD-SNMP-MIB.txt UCD-DEMO-MIB.txt UCD-IPFWACC-MIB.txt \
 	UCD-DLMOD-MIB.txt UCD-DISKIO-MIB.txt
 
+EXTRAMIBS = BGP4-MIB.txt BRIDGE-MIB.txt GNOME-SMI.txt OSPF-MIB.txt \
+	OSPF-TRAP-MIB.txt RIPv2-MIB.txt SOURCE-ROUTING-MIB.txt \
+	LM-SENSORS-MIB.txt
+
 DEFAULTMIBS = @default_mibs_install@
 
 MIBS	= $(V1MIBS) $(V2MIBS) $(V3MIBS) $(RFCMIBS) \
 	$(AGENTMIBS) $(IANAMIBS) \
-	$(NETSNMPMIBS) $(UCDMIBS) $(DEFAULTMIBS)
+	$(NETSNMPMIBS) $(UCDMIBS) $(DEFAULTMIBS) $(EXTRAMIBS)
 
 all: standardall
 
diff -ru net-snmp-5.2.3.orig/net-snmp-config.in net-snmp-5.2.3/net-snmp-config.in
--- net-snmp-5.2.3.orig/net-snmp-config.in	2006-01-11 16:10:24.000000000 +0100
+++ net-snmp-5.2.3/net-snmp-config.in	2006-08-01 11:18:38.000000000 +0200
@@ -148,13 +148,13 @@
     #################################################### client lib
     --libs)
       # use this one == --netsnmp-libs + --external-libs
-      echo $NSC_LDFLAGS $NSC_LIBDIR -lnetsnmp $NSC_LIBS
+      echo $NSC_LDFLAGS $NSC_LIBDIR -lnetsnmp $NSC_LIBS @WRAPLIBS@
       ;;
     --netsnmp-libs)
       echo $NSC_LIBDIR -lnetsnmp
       ;;
     --external-libs)
-      echo $NSC_LDFLAGS $NSC_LIBS
+      echo $NSC_LDFLAGS $NSC_LIBS @WRAPLIBS@
       ;;
     #################################################### agent lib
     --base-agent-libs)
@@ -165,13 +165,13 @@
       ;;
     --agent-libs)
       # use this one == --netsnmp-agent-libs + --external-libs
-      echo $NSC_LDFLAGS $NSC_LIBDIR $NSC_BASE_AGENT_LIBS $NSC_AGENTLIBS
+      echo $NSC_LDFLAGS $NSC_LIBDIR $NSC_BASE_AGENT_LIBS $NSC_AGENTLIBS @WRAPLIBS@
       ;;
     --netsnmp-agent-libs)
       echo $NSC_LIBDIR $NSC_BASE_AGENT_LIBS
       ;;
     --external-agent-libs)
-      echo $NSC_LDFLAGS $NSC_AGENTLIBS
+      echo $NSC_LDFLAGS $NSC_AGENTLIBS @WRAPLIBS@
       ;;
     ####################################################
     --version|--ver*)
@@ -461,7 +461,7 @@
         echo "Ack.  Can't create $tmpfile."
 	exit 1
       fi
-      cmd="@CC@ $cflags @CFLAGS@ @DEVFLAGS@ -I. -I${NSC_INCLUDEDIR} -o $outname $tmpfile $cfiles $NSC_LDFLAGS $NSC_LIBDIR $NSC_BASE_AGENT_LIBS $NSC_AGENTLIBS $ldflags"
+      cmd="@CC@ $cflags @CFLAGS@ @DEVFLAGS@ -I. -I${NSC_INCLUDEDIR} -o $outname $tmpfile $cfiles $NSC_LDFLAGS $NSC_LIBDIR $NSC_BASE_AGENT_LIBS $NSC_AGENTLIBS @WRAPLIBS@ $ldflags"
       echo "running: $cmd"
       `$cmd`
       if test "x$norm" != "x1" ; then
diff -ru net-snmp-5.2.3.orig/snmplib/Makefile.in net-snmp-5.2.3/snmplib/Makefile.in
--- net-snmp-5.2.3.orig/snmplib/Makefile.in	2005-05-06 02:13:37.000000000 +0200
+++ net-snmp-5.2.3/snmplib/Makefile.in	2006-08-01 11:18:38.000000000 +0200
@@ -185,11 +185,11 @@
 
 # how to build the libraries.
 libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION):    ${TOBJS}
-	$(LIB_LD_CMD) libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION) ${TOBJS}
+	$(LIB_LD_CMD) libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION) ${TOBJS} @CRYPTOLIBS@
 	$(RANLIB) libnetsnmp.$(LIB_EXTENSION)$(LIB_VERSION)
 
 libsnmp.$(LIB_EXTENSION)$(LIB_VERSION):    ${TOBJS}
-	$(LIB_LD_CMD) libsnmp.$(LIB_EXTENSION)$(LIB_VERSION) ${TOBJS}
+	$(LIB_LD_CMD) libsnmp.$(LIB_EXTENSION)$(LIB_VERSION) ${TOBJS} @CRYPTOLIBS@
 	$(RANLIB) libsnmp.$(LIB_EXTENSION)$(LIB_VERSION)
 
 #
