--- net-snmp-5.2.2.orig/snmplib/snmp_parse_args.c
+++ net-snmp-5.2.2/snmplib/snmp_parse_args.c
@@ -13,6 +13,7 @@
  */
 
 #include <net-snmp/net-snmp-config.h>
+#include <errno.h>
 
 #if HAVE_STDLIB_H
 #include <stdlib.h>
@@ -401,21 +402,33 @@
 #define SNMPV3_CMD_OPTIONS
 #ifdef  SNMPV3_CMD_OPTIONS
         case 'Z':
-            session->engineBoots = strtoul(optarg, NULL, 10);
-            if (session->engineBoots == 0 || !isdigit(optarg[0])) {
+            errno = 0;
+            session->engineBoots = strtoul(optarg, &cp, 10);
+            if (errno || cp == optarg) {
                 fprintf(stderr, "Need engine boots value after -Z flag.\n");
                 return (-1);
             }
-            cp = strchr(optarg, ',');
-            if (cp && *(++cp) && isdigit(*cp))
-                session->engineTime = strtoul(cp, NULL, 10);
+            if (*cp == ',') {
+              char *endptr;
+              cp ++;
+              session->engineTime = strtoul(cp, &endptr, 10);
+              if (errno || cp == endptr) {
+                fprintf(stderr, "Need engine time after \"-Z engineBoot,\".\n");
+                return (-1);
+              }
+            }
             /*
              * Handle previous '-Z boot time' syntax 
              */
-            else if ((optind < argc) && isdigit(argv[optind][0]))
-                session->engineTime = strtoul(argv[optind], NULL, 10);
+            else if (optind < argc) {
+              session->engineTime = strtoul(argv[optind], &cp, 10);
+              if (errno || cp == argv[optind]) {
+                fprintf(stderr, "Need engine boots value after -Z flag.\n");
+                return (-1);
+              }
+            }
             else {
-                fprintf(stderr, "Need engine time value after -Z flag.\n");
+                fprintf(stderr, "Need engine time value after \"-Z engineBoot\".\n");
                 return (-1);
             }
             break;
--- net-snmp-5.2.2.orig/snmplib/snmpv3.c
+++ net-snmp-5.2.2/snmplib/snmpv3.c
@@ -3,6 +3,7 @@
  */
 
 #include <net-snmp/net-snmp-config.h>
+#include <errno.h>
 #ifdef HAVE_LIMITS_H
 #include <limits.h>
 #endif
@@ -262,17 +263,24 @@
     switch (*cp) {
 
     case 'Z':
-        session->engineBoots = strtoul(optarg, NULL, 10);
-        if (session->engineBoots == 0 || !isdigit(optarg[0])) {
+        errno = 0;
+        session->engineBoots = strtoul(optarg, &cp, 10);
+        if (errno || cp == optarg) {
             fprintf(stderr, "Need engine boots value after -3Z flag.\n");
             return (-1);
         }
-        cp = strchr(optarg, ',');
-        if (cp && *(++cp) && isdigit(*cp))
-            session->engineTime = strtoul(cp, NULL, 10);
-        else {
-            fprintf(stderr, "Need engine time value after -3Z flag.\n");
+        if (*cp == ',') {
+          char *endptr;
+          cp ++;
+          session->engineTime = strtoul(cp, &endptr, 10);
+          if (errno || cp == endptr) {
+            fprintf(stderr, "Need engine time after \"-3Z engineBoot,\".\n");
             return (-1);
+          }
+        }
+        else {
+          fprintf(stderr, "Need engine time after \"-3Z engineBoot\".\n");
+          return (-1);
         }
         break;
 
