From e6e073d31e89089a731870799b43de797480306d Mon Sep 17 00:00:00 2001
From: Howard Chu <hyc@openldap.org>
Date: Wed, 29 Oct 2014 18:05:00 +0000
Subject: [PATCH] ITS#7975 fix onelevel search

---
 servers/slapd/back-mdb/search.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/servers/slapd/back-mdb/search.c b/servers/slapd/back-mdb/search.c
index 98b5f25..9089016 100644
--- a/servers/slapd/back-mdb/search.c
+++ b/servers/slapd/back-mdb/search.c
@@ -837,6 +837,7 @@ loop_begin:
 			}
 			/* Fall-thru */
 		case LDAP_SCOPE_ONELEVEL:
+			if ( id == base->e_id ) break;
 			isc.id = id;
 			isc.nscope = 0;
 			rs->sr_err = mdb_idscopes( op, &isc );
-- 
1.9.1

