--- rsync-3.0.9.orig/rsync.c
+++ rsync-3.0.9/rsync.c
@@ -473,10 +473,6 @@
 				    full_fname(fname));
 				goto cleanup;
 			}
-			if (uid == (uid_t)-1 && sxp->st.st_uid != (uid_t)-1)
-				rprintf(FERROR_XFER, "uid 4294967295 (-1) is impossible to set on %s\n", full_fname(fname));
-			if (gid == (gid_t)-1 && sxp->st.st_gid != (gid_t)-1)
-				rprintf(FERROR_XFER, "gid 4294967295 (-1) is impossible to set on %s\n", full_fname(fname));
 			/* A lchown had been done, so we need to re-stat if
 			 * the destination had the setuid or setgid bits set
 			 * (due to the side effect of the chown call). */
