From 4d28fbb6675c3f5ab60e8fc64d017715d0ec9e88 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard10@gmail.com>
Date: Fri, 30 Jan 2015 12:06:25 -0800
Subject: [PATCH 01/16] torture_libnetapi

---
 source4/torture/libnetapi/wscript_build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/source4/torture/libnetapi/wscript_build b/source4/torture/libnetapi/wscript_build
index 4f579c9820d..07420781d9c 100644
--- a/source4/torture/libnetapi/wscript_build
+++ b/source4/torture/libnetapi/wscript_build
@@ -7,5 +7,6 @@ bld.SAMBA_MODULE('TORTURE_LIBNETAPI',
 	init_function='torture_libnetapi_init',
 	deps='POPT_CREDENTIALS netapi',
 	internal_module=True,
+	enabled=False,
 	)
 
-- 
2.15.1

