From a73537d142ffdac0005deec78f53d15741f02a6b Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard10@gmail.com>
Date: Fri, 30 Jan 2015 12:06:35 -0800
Subject: [PATCH 02/16] share_ldb_module

---
 source4/param/wscript_build | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/source4/param/wscript_build b/source4/param/wscript_build
index 4a2b3b10be2..6555dde8886 100644
--- a/source4/param/wscript_build
+++ b/source4/param/wscript_build
@@ -19,7 +19,7 @@ bld.SAMBA_MODULE('share_classic',
 	source='share_classic.c',
 	subsystem='share',
 	init_function='share_classic_init',
-	deps='samba-util'
+	deps='samba-util samba-hostconfig'
 	)
 
 
@@ -27,7 +27,8 @@ bld.SAMBA_MODULE('share_ldb',
 	source='share_ldb.c',
 	subsystem='share',
 	init_function='share_ldb_init',
-	deps='ldbsamba auth_system_session'
+	deps='ldbsamba auth_system_session',
+	internal_module=False,
 	)
 
 
-- 
2.15.1

