From ad43aa8369f54cafacb1c486d46c6ed006fc0b1a Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard10@gmail.com>
Date: Fri, 30 Jan 2015 12:10:08 -0800
Subject: [PATCH 07/16] reduce-max-debug.patch

---
 source3/include/local.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/source3/include/local.h b/source3/include/local.h
index 6a4331e555a..a5e6652e8be 100644
--- a/source3/include/local.h
+++ b/source3/include/local.h
@@ -214,4 +214,7 @@
 #define HAVE_LINUX_SPLICE_FROM_SOCKET 1
 #endif
 
+/* Reduce max debug level to improve performance */
+#define MAX_DEBUG_LEVEL 5
+
 #endif
-- 
2.15.1

