From 1a1bc8d9702691dfa32e95db197d827e722b2021 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Wed, 28 Sep 2016 18:45:30 +0000
Subject: [PATCH 1/7] netgear: use hostid for homehost

Conflicts:
	mdadm.c
---
 mapfile.c | 6 ++----
 mdadm.c   | 6 ++----
 2 files changed, 4 insertions(+), 8 deletions(-)

diff --git a/mapfile.c b/mapfile.c
index c89d403..9285867 100644
--- a/mapfile.c
+++ b/mapfile.c
@@ -356,10 +356,8 @@ void RebuildMap(void)
 	char *homehost = conf_get_homehost(&require_homehost);
 
 	if (homehost == NULL || strcmp(homehost, "<system>")==0) {
-		if (gethostname(sys_hostname, sizeof(sys_hostname)) == 0) {
-			sys_hostname[sizeof(sys_hostname)-1] = 0;
-			homehost = sys_hostname;
-		}
+		sprintf(sys_hostname, "%08lx", (long)gethostid());
+		homehost = sys_hostname;
 	}
 
 	for (md = mdstat ; md ; md = md->next) {
diff --git a/mdadm.c b/mdadm.c
index c3a265b..b21144a 100644
--- a/mdadm.c
+++ b/mdadm.c
@@ -1326,10 +1326,8 @@ int main(int argc, char *argv[])
 	if (c.homehost == NULL && c.require_homehost)
 		c.homehost = conf_get_homehost(&c.require_homehost);
 	if (c.homehost == NULL || strcasecmp(c.homehost, "<system>") == 0) {
-		if (gethostname(sys_hostname, sizeof(sys_hostname)) == 0) {
-			sys_hostname[sizeof(sys_hostname)-1] = 0;
-			c.homehost = sys_hostname;
-		}
+		sprintf(sys_hostname, "%08lx", (long)gethostid());
+		c.homehost = sys_hostname;
 	}
 	if (c.homehost &&
 	    (!c.homehost[0] || strcasecmp(c.homehost, "<none>") == 0)) {
-- 
2.13.0

