From b1b50a62b7fbb2f324d4fd7b0df74c4c2a779b54 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard10@gmail.com>
Date: Thu, 5 Jun 2014 10:59:52 -0700
Subject: [PATCH 6/7] netgear: create: disable automatically enabled
 write-intent bitmap

---
 Create.c | 8 --------
 1 file changed, 8 deletions(-)

Index: mdadm-4.1/Create.c
===================================================================
--- mdadm-4.1.orig/Create.c
+++ mdadm-4.1/Create.c
@@ -519,16 +519,6 @@ int Create(struct supertype *st, char *m
 		}
 	}
 
-	if (!s->bitmap_file &&
-	    s->level >= 1 &&
-	    st->ss->add_internal_bitmap &&
-	    (s->consistency_policy != CONSISTENCY_POLICY_RESYNC &&
-	     s->consistency_policy != CONSISTENCY_POLICY_PPL) &&
-	    (s->write_behind || s->size > 100*1024*1024ULL)) {
-		if (c->verbose > 0)
-			pr_err("automatically enabling write-intent bitmap on large array\n");
-		s->bitmap_file = "internal";
-	}
 	if (s->bitmap_file && strcmp(s->bitmap_file, "none") == 0)
 		s->bitmap_file = NULL;
 
