From 318a07f6e9c30002c16818caa8c2537fb5bbfc52 Mon Sep 17 00:00:00 2001
From: Justin Ding <justin.ding@netgear.com>
Date: Wed, 2 Jan 2019 18:53:14 -0800
Subject: [PATCH] Remove two error logs.

---
 Create.c | 5 -----
 mdopen.c | 1 -
 2 files changed, 6 deletions(-)

diff --git a/Create.c b/Create.c
index 04b1dfc..d059b66 100644
--- a/Create.c
+++ b/Create.c
@@ -466,11 +466,6 @@ int Create(struct supertype *st, char *mddev,
 			    s->level == 1 &&
 			    (warn & 1024) == 0) {
 				warn |= 1024;
-				pr_err("Note: this array has metadata at the start and\n"
-					"    may not be suitable as a boot device.  If you plan to\n"
-					"    store '/boot' on this device please ensure that\n"
-					"    your boot-loader understands md/v1.x metadata, or use\n"
-					"    --metadata=0.90\n");
 			}
 			close(fd);
 		}
diff --git a/mdopen.c b/mdopen.c
index 4ec13f5..9fa8383 100644
--- a/mdopen.c
+++ b/mdopen.c
@@ -118,7 +118,6 @@ int create_named_array(char *devnm)
 		close(fd);
 	}
 	if (fd < 0 || n != (int)strlen(devnm)) {
-		pr_err("Fail create %s when using %s\n", devnm, new_array_file);
 		return 0;
 	}
 
-- 
2.1.4

