From 18832eb075a3dab4f3b30f7a9a1b26bcdd49f8e2 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard10@gmail.com>
Date: Fri, 30 Jan 2015 12:06:25 -0800
Subject: [PATCH 02/19] 02_torture_libnetapi

---
 source4/torture/libnetapi/wscript_build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/source4/torture/libnetapi/wscript_build b/source4/torture/libnetapi/wscript_build
index 017a163..0649807 100644
--- a/source4/torture/libnetapi/wscript_build
+++ b/source4/torture/libnetapi/wscript_build
@@ -8,5 +8,6 @@ bld.SAMBA_MODULE('TORTURE_LIBNETAPI',
 	init_function='torture_libnetapi_init',
 	deps='POPT_CREDENTIALS netapi',
 	internal_module=True,
+	enabled=False,
 	)
 
-- 
1.9.1

