From d2523829dde8db275ebe2350d0880f27ecc293f4 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard10@gmail.com>
Date: Fri, 30 Jan 2015 12:09:51 -0800
Subject: [PATCH 09/19] dyson.patch

---
 source3/wscript_build | 19 +++++++++++++------
 1 file changed, 13 insertions(+), 6 deletions(-)

diff --git a/source3/wscript_build b/source3/wscript_build
index e1964a3..65982cb 100755
--- a/source3/wscript_build
+++ b/source3/wscript_build
@@ -55,12 +55,19 @@ bld.SAMBA3_LIBRARY('netapi',
                     pc_files='libnet/netapi.pc',
                     vnum='0')
 
-bld.SAMBA3_LIBRARY('nss_wins',
-                  source='../nsswitch/wins.c',
-                  deps='''param libsmb LIBTSOCKET''',
-                  realname='libnss_wins.so.2',
-                  soname='libnss_wins.so',
-                  vnum='2')
+if Utils.unversioned_sys_platform() == 'sunos':
+        bld.SAMBA3_LIBRARY('nss_wins',
+                          source='../nsswitch/wins.c ../nsswitch/wins_nss_solaris.c',
+                          deps='''param libsmb LIBTSOCKET''',
+                          realname='nss_wins.so.1',
+                          vnum='1')
+else:
+        bld.SAMBA3_LIBRARY('nss_wins',
+                          source='../nsswitch/wins.c',
+                          deps='''param libsmb LIBTSOCKET''',
+                          realname='libnss_wins.so.2',
+                          soname='libnss_wins.so',
+                          vnum='2')
 
 bld.SAMBA3_LIBRARY('gse',
                    source='librpc/crypto/gse_krb5.c librpc/crypto/gse.c',
-- 
1.9.1

