Index: dbus-1.2.14/dbus/dbus-internals.c
===================================================================
--- dbus-1.2.14.orig/dbus/dbus-internals.c	2009-04-17 21:45:29.000000000 +0200
+++ dbus-1.2.14/dbus/dbus-internals.c	2009-05-07 00:16:33.000000000 +0200
@@ -193,7 +193,7 @@
 
 static dbus_bool_t warn_initted = FALSE;
 static dbus_bool_t fatal_warnings = FALSE;
-static dbus_bool_t fatal_warnings_on_check_failed = TRUE;
+static dbus_bool_t fatal_warnings_on_check_failed = FALSE;
 
 static void
 init_warnings(void)
