--- a/rsync.c	Mon Dec  5 22:09:28 2011
+++ b/rsync.c	Mon Dec  5 22:04:35 2011
@@ -473,10 +473,12 @@ int set_file_attrs(const char *fname, st
 				    full_fname(fname));
 				goto cleanup;
 			}
+#if 0 // disable the dreaded "UID is impossible to set" message
 			if (uid == (uid_t)-1 && sxp->st.st_uid != (uid_t)-1)
 				rprintf(FERROR_XFER, "uid 4294967295 (-1) is impossible to set on %s\n", full_fname(fname));
 			if (gid == (gid_t)-1 && sxp->st.st_gid != (gid_t)-1)
 				rprintf(FERROR_XFER, "gid 4294967295 (-1) is impossible to set on %s\n", full_fname(fname));
+#endif
 			/* A lchown had been done, so we need to re-stat if
 			 * the destination had the setuid or setgid bits set
 			 * (due to the side effect of the chown call). */
