Description: Exclude .svn directories from library scan
Author: Colin Watson <cjwatson@debian.org>
Forwarded: not-needed
Last-Update: 2010-03-02

Index: b/Makefile.PL
===================================================================
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -113,6 +113,13 @@
 ";
 }
 
+# Exclude .svn directories.
+sub MY::libscan {
+  my($self,$path) = @_;
+  return '' if $path =~ m:(\b(RCS|CVS|SCCS)|\.svn)\b: ;
+  $path;
+}
+
 sub MY::postamble {
   q{
 setversion:
