From: Daniel Veillard <veillard@redhat.com>
Date: Wed, 28 Nov 2012 22:38:50 +0800
Subject: Fix potential out of bound access

---
 parser.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/parser.c b/parser.c
index 9863275..e1b0364 100644
--- a/parser.c
+++ b/parser.c
@@ -3932,7 +3932,7 @@ xmlParseAttValueComplex(xmlParserCtxtPtr ctxt, int *attlen, int normalize) {
 	c = CUR_CHAR(l);
     }
     if ((in_space) && (normalize)) {
-        while (buf[len - 1] == 0x20) len--;
+        while ((len > 0) && (buf[len - 1] == 0x20)) len--;
     }
     buf[len] = 0;
     if (RAW == '<') {
-- 
