diff -uNr netatalk-2.1.1/libatalk/adouble/ad_open.c netatalk-2.1.1.ntgr/libatalk/adouble/ad_open.c
--- netatalk-2.1.1/libatalk/adouble/ad_open.c	2010-04-29 09:17:50.000000000 +0000
+++ netatalk-2.1.1.ntgr/libatalk/adouble/ad_open.c	2010-05-26 22:49:16.000000000 +0000
@@ -1284,7 +1284,17 @@
                     admode = mode;
                 }
             }
-                
+#ifdef __sparc__
+            if ((mode == 0x1B6) && (*path == '.')) {
+                if (access(path, F_OK) < 0) {
+                    char oldpath[MAXPATHLEN+1];
+		    sprintf(oldpath, ":2e%s", path+1);
+                    if(rename(oldpath, path) == 0) {
+                        LOG(log_info, logtype_default, "Converted [%s] from CAP-encoded name", path);
+                    }
+                }
+            }
+#endif
             ad->ad_data_fork.adf_fd =open( path, hoflags | O_NOFOLLOW, admode );
             
             if (ad->ad_data_fork.adf_fd == -1) {
