diff -uNr netatalk-2.1.1/etc/afpd/switch.c netatalk-2.1.1.ntgr/etc/afpd/switch.c
--- netatalk-2.1.1/etc/afpd/switch.c	2010-04-29 09:17:50.000000000 +0000
+++ netatalk-2.1.1.ntgr/etc/afpd/switch.c	2010-05-26 21:55:39.000000000 +0000
@@ -82,7 +82,7 @@
     NULL, NULL, NULL, NULL,
     NULL, NULL, NULL, NULL,					/*  64 -  71 */
     NULL, NULL, NULL, NULL,
-    NULL, NULL, NULL, NULL,					/*  72 -  79 */
+    NULL, NULL, afp_syncdir, afp_syncfork,			/*  72 -  79 */
     NULL, NULL, NULL, NULL,
     NULL, NULL, NULL, NULL,					/*  80 -  87 */
     NULL, NULL, NULL, NULL,
diff -uNr netatalk-2.1.1/etc/atalkd/main.c netatalk-2.1.1.ntgr/etc/atalkd/main.c
--- netatalk-2.1.1/etc/atalkd/main.c	2010-04-29 09:17:50.000000000 +0000
+++ netatalk-2.1.1.ntgr/etc/atalkd/main.c	2010-05-26 21:55:39.000000000 +0000
@@ -1075,7 +1075,8 @@
     openlog( prog, LOG_PID );
 #else /* ultrix */
     set_processname(prog);
-    syslog_setup(log_debug, logtype_default, logoption_pid, logfacility_daemon );
+    //JM: Don't use syslog// syslog_setup(log_debug, logtype_default, logoption_pid, logfacility_daemon );
+    log_init();
 #endif /* ultrix */
 
     LOG(log_info, logtype_atalkd, "restart (%s)", version );
diff -uNr netatalk-2.1.1/etc/papd/Makefile.am netatalk-2.1.1.ntgr/etc/papd/Makefile.am
--- netatalk-2.1.1/etc/papd/Makefile.am	2010-04-29 09:17:50.000000000 +0000
+++ netatalk-2.1.1.ntgr/etc/papd/Makefile.am	2010-05-26 21:55:39.000000000 +0000
@@ -9,7 +9,7 @@
 papd_SOURCES = main.c printcap.c session.c file.c comment.c lp.c ppd.c \
 	       magics.c headers.c queries.c auth.c uam.c print_cups.c
 
-papd_LDADD = $(top_builddir)/libatalk/libatalk.la @CUPS_LIBS@ @LIBADD_DL@
+papd_LDADD = $(top_builddir)/libatalk/libatalk.la @CUPS_LIBS@ -ldl 
 papd_LDFLAGS = -export-dynamic @CUPS_LDFLAGS@
 
 showppd_SOURCES = showppd.c ppd.c
