diff -uNr netatalk-2.1.1/etc/afpd/file.c netatalk-2.1.1.ntgr/etc/afpd/file.c
--- netatalk-2.1.1/etc/afpd/file.c	2010-05-20 08:25:06.000000000 +0000
+++ netatalk-2.1.1.ntgr/etc/afpd/file.c	2010-05-26 22:24:01.000000000 +0000
@@ -585,6 +585,15 @@
                 return AFPERR_ACCESS;
             case EIO:
                 LOG(log_error, logtype_afpd, "getfilparams(%s): bad resource fork", upath);
+                LOG(log_error, logtype_afpd, "trying to delete bad resource fork for '%s'", fullpathname(upath));
+		char badfork[ MAXPATHLEN + 1];
+		if ( getcwd( badfork , MAXPATHLEN) ) {
+			strlcat(badfork, "/.AppleDouble/", MAXPATHLEN);
+			strlcat(badfork, upath, MAXPATHLEN);
+			LOG(log_error, logtype_afpd, " deleting bad resource fork '%s'", badfork);
+			if(unlink(badfork) == -1)
+				LOG(log_error, logtype_afpd, " deletion of '%s' failed: %s", fullpathname(upath), strerror(errno));
+		}
                 /* fall through */
             case ENOENT:
             default:
