diff --git a/libatalk/cnid/dbd/cnid_dbd.c b/libatalk/cnid/dbd/cnid_dbd.c
index 4c45eb1..a3b2386 100644
--- a/libatalk/cnid/dbd/cnid_dbd.c
+++ b/libatalk/cnid/dbd/cnid_dbd.c
@@ -67,6 +67,7 @@ static void delay(int sec)
 
 static int tsock_getfd(const char *host, const char *port)
 {
+    static int tried_start_cnid_metad = 0;
     int sock = -1;
     int attr;
     int err;
@@ -86,6 +87,7 @@ static int tsock_getfd(const char *host, const char *port)
         return -1;
     }
 
+retry_cnid_connection:
     /* loop through all the results and bind to the first we can */
     for (p = servinfo; p != NULL; p = p->ai_next) {
         if ((sock = socket(p->ai_family, p->ai_socktype, p->ai_protocol)) == -1) {
@@ -154,6 +156,23 @@ static int tsock_getfd(const char *host, const char *port)
                         errno = optval;
                         LOG(log_error, logtype_cnid, "getfd: getsockopt says: %s",
                             strerror(errno));
+                        // NETGEAR - JM: If we get Connection refused, it may be because cnid_metad isn't running.  Try to start it.
+                        if (!p->ai_next && errno == ECONNREFUSED) {
+                            close(sock);
+                            sock = -1;
+                            if (tried_start_cnid_metad) {
+                                LOG(log_error, logtype_cnid, "Failed to start CNID daemon, please contact ReadyNAS technical support");
+                            } else {
+                                tried_start_cnid_metad = 1;
+                                LOG(log_error, logtype_cnid, "Retrying after restarting cnid_metad...");
+                                uid_t uid = geteuid();
+                                seteuid(0);
+                                system("/usr/sbin/cnid_metad");
+                                seteuid(uid);
+                                sleep(2);
+                                goto retry_cnid_connection;
+                            }
+                        }
                     }
                     close(sock);
                     sock = -1;
