--- bvi-1.3.2.orig/Makefile.in
+++ bvi-1.3.2/Makefile.in
@@ -19,6 +19,9 @@
 #
 ###############################################
 
+# Edited for Debian/GNU Linux
+DESTDIR = 
+
 srcdir = @srcdir@
 prefix = @prefix@
 exec_prefix = @exec_prefix@
@@ -33,18 +36,22 @@
 HEADER =  bvi.h set.h
 BMOBJ  =  bmore.o bm_unix.o recomp.o
 CC     =  @CC@
-CFLAGS =  @CFLAGS@ @CPPFLAGS@
+CFLAGS =  @CFLAGS@ -Wall @CPPFLAGS@
 DEFS   =  @DEFS@
 LDFLAGS=  @LDFLAGS@
 LIBS   =  @LIBS@
 SHELL  =  /bin/sh
 
 
-
 INSTALL = @INSTALL@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_DATA = @INSTALL_DATA@
 
+# Check for Debian environment strings to build with debug
+# (compilation is done with '-g' option by default if available)
+ifeq (,$(findstring nostrip,$(DEB_BUILD_OPTIONS)))
+	INSTALL_PROGRAM += -s
+endif
 
 all: bvi bmore
 
@@ -64,13 +71,13 @@
 install: all installdirs
 	@echo "Installing bvi"
 	@$(INSTALL_DATA) bvi.1 bmore.1 $(man1dir)
-	@$(INSTALL_PROGRAM) -s bvi $(bindir)
-	@$(INSTALL_PROGRAM) -s bmore $(bindir)
+	@$(INSTALL_PROGRAM) bvi $(bindir)
+	@$(INSTALL_PROGRAM) bmore $(bindir)
 	@$(INSTALL_DATA) bmore.help $(libdir)
 	@if [ -f $(bindir)/bview ]; then rm $(bindir)/bview; fi
 	@if [ -f $(bindir)/bvedit ]; then rm $(bindir)/bvedit; fi
-	@ln $(bindir)/bvi $(bindir)/bview
-	@ln $(bindir)/bvi $(bindir)/bvedit
+	@ln -s bvi $(bindir)/bview
+	@ln -s bvi $(bindir)/bvedit
 
 installdirs:
 	$(SHELL) ${srcdir}/mkinstalldirs $(bindir) $(libdir) $(man1dir)
