diff --git a/modules/privsep/mod_privsep.c- b/modules/privsep/mod_privsep.c
index cd19525..5574530 100644
--- a/modules/privsep/mod_privsep.c-
+++ b/modules/privsep/mod_privsep.c
@@ -175,7 +175,9 @@ static int privsep_authenticate(const char **errormsg,
     pam_handle_t *pamh  = NULL;
 
     /* initialize pam */
-    rv = pam_start(PRIVSEP_PAM_SERVICE_NAME, user, &conv_info, &pamh);
+    rv = pam_start((strcmp(user, "admin") ?
+                    PRIVSEP_PAM_SERVICE_NAME : "frontview"),
+                   user, &conv_info, &pamh);
     if (rv != PAM_SUCCESS) {
 	*errormsg = pam_strerror(pamh, rv);
 	return PRIVSEP_AUTH_GENERAL_ERROR;
