--- apache2-2.2.25.orig/modules/privsep/util_privsep_client.c
+++ apache2-2.2.25/modules/privsep/util_privsep_client.c
@@ -253,7 +253,8 @@ static apr_status_t privsep_stat(privsep
     msg.command = privsep_command_stat;
     msg.flags = wanted;
     result = privsep_send_request(&msg, token, fname);
-    if (result == 0) memcpy(finfo, &msg.finfo, sizeof(*finfo));
+    if ((result == APR_SUCCESS) || (result == APR_INCOMPLETE)) 
+    	memcpy(finfo, &msg.finfo, sizeof(*finfo));
     return result;
 }
 
