From 8e2430f44af681b9a019f32ecdf82b11f6a76eb7 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Wed, 26 Oct 2016 17:34:38 -0700
Subject: [PATCH] netgear: quit on forward failure

---
 cli-runopts.c | 2 +-
 cli-tcpfwd.c  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/cli-runopts.c b/cli-runopts.c
index 4a89ba3..5eb3bfb 100644
--- a/cli-runopts.c
+++ b/cli-runopts.c
@@ -149,7 +149,7 @@ void cli_getopts(int argc, char ** argv) {
 	cli_opts.privkeys = list_new();
 #endif
 #if DROPBEAR_CLI_ANYTCPFWD
-	cli_opts.exit_on_fwd_failure = 0;
+	cli_opts.exit_on_fwd_failure = 1;
 #endif
 #if DROPBEAR_CLI_LOCALTCPFWD
 	cli_opts.localfwds = list_new();
diff --git a/cli-tcpfwd.c b/cli-tcpfwd.c
index 95b2d7c..93339e3 100644
--- a/cli-tcpfwd.c
+++ b/cli-tcpfwd.c
@@ -68,7 +68,7 @@ static void fwd_failed(const char* format, ...)
 	va_start(param, format);
 
 	if (cli_opts.exit_on_fwd_failure) {
-		_dropbear_exit(EXIT_FAILURE, format, param);
+		_dropbear_exit(EBUSY, format, param);
 	} else {
 		_dropbear_log(LOG_WARNING, format, param);
 	}
-- 
2.10.1

