diff -urN db-2.7.7.orig/mutex/68020.gcc db-2.7.7/mutex/68020.gcc
--- db-2.7.7.orig/mutex/68020.gcc	Tue Nov 10 11:50:44 1998
+++ db-2.7.7/mutex/68020.gcc	Mon Mar 26 23:46:01 2001
@@ -5,7 +5,7 @@
  */
 #define TSL_SET(tsl) ({							\
 	register tsl_t *__l = (tsl);					\
-	int __r;							\
+	unsigned char __r;							\
 	    asm volatile("tas  %1; \n					\
 			  seq  %0"					\
 		: "=dm" (__r), "=m" (*__l)				\
