diff -uNr db-2.7.7/cxx/cxx_app.cpp db-2.7.7.new/cxx/cxx_app.cpp
--- db-2.7.7/cxx/cxx_app.cpp	Wed Dec 16 12:53:47 1998
+++ db-2.7.7.new/cxx/cxx_app.cpp	Mon Apr  2 19:55:02 2001
@@ -20,6 +20,8 @@
 #include <stdio.h>              // needed for setErrorStream
 #include <string.h>
 
+using namespace std;
+
 ////////////////////////////////////////////////////////////////////////
 //                                                                    //
 //                            DbEnv                                   //
@@ -125,7 +127,7 @@
 // db_env triggered the call.  A user that has multiple DB_ENVs
 // will simply not be able to have different streams for each one.
 //
-void DbEnv::set_error_stream(class ostream *stream)
+void DbEnv::set_error_stream(ostream *stream)
 {
     error_stream_ = stream;
 
diff -uNr db-2.7.7/include/db_cxx.h db-2.7.7.new/include/db_cxx.h
--- db-2.7.7/include/db_cxx.h	Tue Aug 31 11:45:43 1999
+++ db-2.7.7.new/include/db_cxx.h	Mon Apr  2 19:54:32 2001
@@ -52,6 +52,8 @@
 #include <iostream.h>
 #include <db.h>
 
+using namespace std;
+
 class Db;                                        // forward
 class Dbc;                                       // forward
 class DbEnv;                                     // forward
@@ -717,7 +719,7 @@
     // call set_error_stream() to force all errors to a C++ stream.
     // It is unwise to mix these approaches.
     //
-    void set_error_stream(class ostream*);
+    void set_error_stream(ostream*);
 
     // used internally
     static int runtime_error(const char *caller, int err,
