--- db-2.7.7/dist/Makefile.in.orig	Fri Apr  5 22:18:55 2002
+++ db-2.7.7/dist/Makefile.in	Fri Apr  5 22:24:00 2002
@@ -67,7 +67,14 @@
 	$(ar) $@ $(COBJS)
 	test ! -f $(ranlib) || $(ranlib) $@
 
-libdb2.so: db.h $(SHARED_OBJS)
+Versions: $(SHARED_OBJS)
+	rm -f $@
+	echo "DB2 {" > $@
+	echo "  global:" >> $@
+	nm $(SHARED_OBJS) | grep " [Tt] " | cut -d" " -f3 | sed -e 's/$$/;/' >> $@
+	echo "};" >> $@
+
+libdb2.so: db.h $(SHARED_OBJS) Versions
 	$(rm) -f $@*
 	$(CC) -shared -o $@ $(LDFLAGS) -Wl,-Bsymbolic -Wl,-soname,$@.$(MAJOR) \
 		-Wl,--version-script=Versions $(SHARED_OBJS) $(LIBS)

