--- a/startpar.c
+++ b/startpar.c
@@ -397,7 +397,6 @@
   if (p->pid != 0)
     return;
 
-  (void)signal(SIGINT,  SIG_DFL);
   (void)signal(SIGQUIT, SIG_DFL);
   (void)signal(SIGSEGV, SIG_DFL);
   (void)signal(SIGTERM, SIG_DFL);
@@ -511,7 +510,9 @@
 
   if (pid == 0)
     {
+      /* Interactive init scripts may be interrupted */
       (void)signal(SIGINT,  SIG_DFL);
+
       (void)signal(SIGQUIT, SIG_DFL);
       (void)signal(SIGSEGV, SIG_DFL);
       (void)signal(SIGTERM, SIG_DFL);
@@ -700,6 +701,9 @@
   char *splashopt = 0;
   int notty = 0;
 
+  /* By default, init scripts should not be interruptible */
+  (void)signal(SIGINT,  SIG_IGN);
+
   (void)signal(SIGUSR1, sighandler_preload);
   (void)signal(SIGUSR2, sighandler_nopreload);
 
