--- cifs-utils-6.1.orig/mount.cifs.c
+++ cifs-utils-6.1/mount.cifs.c
@@ -2065,6 +2065,9 @@ mount_retry:
 
 		switch (errno) {
 		case EOPNOTSUPP:
+			if (strstr(options, "noserverino") == NULL)
+				strlcat(parsed_info->options, ",noserverino",
+					sizeof(parsed_info->options));
 			if (strstr(options, "nounix") == NULL) {
 				strlcat(parsed_info->options, ",nounix",
 					sizeof(parsed_info->options));
