From 6d3707522a004c2b469b2405a71dec7c222ddec9 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Thu, 28 Jul 2016 12:08:14 -0700
Subject: [PATCH 1/9] force pam multiarch

---
 pam/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pam/Makefile.am b/pam/Makefile.am
index befccf8..c089b59 100644
--- a/pam/Makefile.am
+++ b/pam/Makefile.am
@@ -8,7 +8,7 @@ AM_CFLAGS = -D_GNU_SOURCE -Wwrite-strings
 
 AM_CPPFLAGS = -I$(top_srcdir) $(DBUS_CFLAGS)
 
-securelibdir = $(shell echo /`basename $(libdir)`/security)
+securelibdir = $(shell echo /lib/`basename $(libdir)`/security)
 
 securelib_LTLIBRARIES = pam_snapper.la
 
-- 
2.9.3

