From 93e889e56f2857650b49501ae18ce55422e7d9d7 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Wed, 16 Sep 2015 13:20:59 -0700
Subject: [PATCH 3/9] syslog

---
 server/snapperd.cc | 26 +++++++++++++++++++++++---
 1 file changed, 23 insertions(+), 3 deletions(-)

diff --git a/server/snapperd.cc b/server/snapperd.cc
index ab69492..b5669bc 100644
--- a/server/snapperd.cc
+++ b/server/snapperd.cc
@@ -22,6 +22,7 @@
 
 #include <stdlib.h>
 #include <getopt.h>
+#include <syslog.h>
 #include <iostream>
 #include <string>
 
@@ -40,7 +41,7 @@ using namespace std;
 const seconds idle_time(120);
 const seconds snapper_cleanup_time(60);
 
-bool log_stdout = false;
+bool log_stdout = true;
 bool log_debug = false;
 
 
@@ -169,7 +170,26 @@ void
 log_do(LogLevel level, const string& component, const char* file, const int line, const char* func,
        const string& text)
 {
-    cout << text << endl;
+    int syslog_level;
+
+    switch (level) {
+    case DEBUG:
+        syslog_level = LOG_DEBUG;
+        break;
+    case MILESTONE:
+        syslog_level = LOG_INFO;
+        break;
+    case WARNING:
+        syslog_level = LOG_WARNING;
+        break;
+    case ERROR:
+        syslog_level = LOG_ERR;
+        break;
+    default:
+        syslog_level = LOG_CRIT;
+        break;
+    }
+    syslog(syslog_level, "%s", text.c_str());
 }
 
 
@@ -227,7 +247,7 @@ main(int argc, char** argv)
 	switch (c)
 	{
 	    case 's':
-		log_stdout = true;
+		log_stdout = false;
 		break;
 
 	    case 'd':
-- 
2.9.3

