Otherwise we quickly crash.
---

--- tracker-1.12.3.orig/src/libtracker-common/tracker-seccomp.c
+++ tracker-1.12.3/src/libtracker-common/tracker-seccomp.c
@@ -70,7 +70,6 @@ tracker_seccomp_init (void)
 	ALLOW_RULE (mprotect);
 	ALLOW_RULE (madvise);
 	ERROR_RULE (mlock, EPERM);
-	ERROR_RULE (mlock2, EPERM);
 	ERROR_RULE (munlock, EPERM);
 	ERROR_RULE (mlockall, EPERM);
 	ERROR_RULE (munlockall, EPERM);
