--- tracker-1.12.3/src/libtracker-common/tracker-seccomp.c	2018-01-16 02:45:42.000000000 -0800
+++ tracker-1.12.3-patched/src/libtracker-common/tracker-seccomp.c	2018-01-17 21:41:48.000000000 -0800
@@ -139,6 +139,9 @@
 	ALLOW_RULE (lseek);
 	ALLOW_RULE (_llseek);
 	ALLOW_RULE (fadvise64);
+#ifdef __arm__
+	ALLOW_RULE (arm_fadvise64_64);
+#endif
 	ALLOW_RULE (write);
 	ALLOW_RULE (writev);
 	ALLOW_RULE (dup);
