diff -urN shadow-20000902.orig/src/Makefile.am shadow-20000902/src/Makefile.am
--- shadow-20000902.orig/src/Makefile.am	Fri Jun 23 10:27:12 2000
+++ shadow-20000902/src/Makefile.am	Wed Sep 27 18:42:26 2000
@@ -76,7 +76,7 @@
 # chage chfn chsh expiry gpasswd newgrp passwd \
 # sulogin
 
-shlibs = ../lib/libshadow.la
+shlibs = #../lib/libshadow.la
 # With glibc2, almost all programs need libcrypt for some reason,
 # even those that don't actually use crypt().
 LDADD = ${shlibs} ../libmisc/libmisc.a ../lib/libshadow.a @INTLLIBS@ @LIBCRYPT@ @LIBTCFS@ @LIBSKEY@ @LIBMD@
