--- shadow-19990827/src/lastlog.c~	Sat Oct  9 11:38:06 1999
+++ shadow-19990827/src/lastlog.c	Sat Oct  9 11:52:54 1999
@@ -64,6 +64,15 @@
 static struct stat statbuf;	/* fstat buffer for file size */
 static struct passwd *pwent;
 
+#include <getopt.h> 
+static struct option const longopts[] =
+{
+  {"user", required_argument, 0, 'u'},
+  {"time", required_argument, 0, 't'},
+  {"help", no_argument, 0, 'h'},
+  {0, 0, 0, 0}
+};
+
 extern	char	*optarg;
 
 #define	NOW	(time ((time_t *) 0))
@@ -86,7 +94,7 @@
 		perror (LASTLOG_FILE);
 		exit (1);
 	}
-	while ((c = getopt (argc, argv, "u:t:")) != EOF) {
+	while ((c = getopt_long (argc, argv, "u:t:h", longopts, NULL)) != -1) {
 		switch (c) {
 			case 'u':
 				pwent = getpwnam (optarg);
@@ -104,6 +112,14 @@
 				seconds = days * DAY;
 				tflg++;
 				break;
+			case 'h':
+				fprintf(stdout, "Usage: %s [-u login-name] [-t days]\n",
+					argv[0]);
+				exit(0);
+			default:
+				fprintf(stdout, "Usage: %s [-u login-name] [-t days]\n",
+					argv[0]);
+				exit(1);
 		}
 	}
 	print ();
