diff -urN shadow-20000902.orig/src/useradd.c shadow-20000902/src/useradd.c
--- shadow-20000902.orig/src/useradd.c	Sat Sep  2 14:40:44 2000
+++ shadow-20000902/src/useradd.c	Mon Oct 23 11:54:40 2000
@@ -75,7 +75,7 @@
 static gid_t def_group = 100;
 static const char *def_gname = "other";
 static const char *def_home = "/home";
-static const char *def_shell = "";
+static const char *def_shell = "/bin/bash";
 static const char *def_template = SKEL_DIR;
 #ifdef SHADOWPWD
 static long def_inactive = -1;
@@ -451,6 +451,7 @@
 	char	buf[1024];
 	static	char	new_file[] = NEW_USER_FILE;
 	char	*cp;
+	int	ofd;
 	int	out_group = 0;
 	int	out_home = 0;
 	int	out_inactive = 0;
@@ -465,9 +466,14 @@
 	 * Create a temporary file to copy the new output to.
 	 */
 
-	mktemp (new_file);
-	if (!(ofp = fopen (new_file, "w"))) {
+	if ((ofd = mkstemp (new_file)) == -1) {
 		fprintf(stderr, _("%s: cannot create new defaults file\n"),
+			Prog);
+		return -1;
+	}
+
+	if (!(ofp = fdopen(ofd, "w"))) {
+		fprintf(stderr, _("%s: cannot open new defaults file\n"),
 			Prog);
 		return -1;
 	}
