--- shadow-19990827/libmisc/chowntty.c~	Sat Jan  8 22:17:28 2000
+++ shadow-19990827/libmisc/chowntty.c	Sat Jan  8 22:18:04 2000
@@ -37,6 +37,7 @@
 
 #include <stdio.h>
 #include <grp.h>
+#include <errno.h>
 
 #include "prototypes.h"
 #include "defines.h"
@@ -105,8 +106,8 @@
 		exit (1);
 	}
 	
-	if (chown(tty, info->pw_uid, gid) ||
-			chmod(tty, getdef_num("TTYPERM", 0600))) {
+	if ((chown(tty, info->pw_uid, gid) && errno != EROFS) ||
+		    (chmod(tty, getdef_num("TTYPERM", 0600)) && errno != EROFS)) {
 		snprintf(buf, sizeof buf, _("Unable to change tty %s"), tty);
 		SYSLOG((LOG_WARN, "unable to change tty `%s' for user `%s'\n",
 			tty, info->pw_name));
