diff -urN shadow-19990827.orig/libmisc/pam_pass.c shadow-19990827/libmisc/pam_pass.c
--- shadow-19990827.orig/libmisc/pam_pass.c	Mon Jun  7 12:40:44 1999
+++ shadow-19990827/libmisc/pam_pass.c	Sat Apr 29 10:26:33 2000
@@ -50,7 +50,7 @@
 		pam_end(pamh, ret);
 		exit(10);  /* XXX */
 	}
-
+	fprintf(stderr, "passwd: password updated successfully\n");
 	pam_end(pamh, PAM_SUCCESS);
 }
 #else /* !USE_PAM */
diff -urN shadow-19990827.orig/src/passwd.c shadow-19990827/src/passwd.c
--- shadow-19990827.orig/src/passwd.c	Fri Aug 27 15:02:51 1999
+++ shadow-19990827/src/passwd.c	Sat Apr 29 10:22:55 2000
@@ -1286,8 +1286,8 @@
 	 */
 
 	if (!amroot && pw->pw_uid != getuid ()) {
-		fprintf(stderr, _("You may not change the password for %s.\n"),
-			name);
+		fprintf(stderr, _("%s: You may not view or modify password information for %s.\n"),
+			Prog, name);
 		SYSLOG((LOG_WARN, NOPERM2, name));
 		closelog();
 		exit(E_NOPERM);
