--- shadow-19990827/doc/README.debian~	Sat Oct  9 00:44:45 1999
+++ shadow-19990827/doc/README.debian	Sat Oct  9 00:49:04 1999
@@ -9,9 +9,12 @@
 password support, you'll gain the ability to set password ages and
 expirations with chage(1).
 
-You may want to install the secure-su package which allows more
-restrictions on su, for example a wheel group.
-
+NOTE: If you use the nscd package, you may have problems with a
+slight delay in updating the password information. You may notice
+this during upgrades of certain packages that try to add a system
+user and then access the users information immediately afterwards.
+To avoid this, it is suggested that you stop the nscd daemon before
+upgrades, then restart it again.
 
 ---General configuration
 
@@ -19,26 +16,14 @@
 /etc/login.defs.  See login.defs(5).  The defaults are quite
 reasonable.
 
-
----MD5 Encryption
-
-If you set MD5_CRYPT_ENAB=yes in /etc/login.defs, passwords will be
-encrypted with an MD5-based algorithm.  It also supports of passwords
-of unlimited length and longer salt strings.
+Also see the /etc/pam.d/* files for each program to configure the PAM
+support. PAM documentation is available in several formats in the
+libpam-doc package.
 
 
----Login and resource control
-
-/etc/login.access and /etc/porttime control who may login to which
-ports and when they may login.  To enforce time restrictions, you'll
-need to run logoutd.  /etc/init.d/logoutd will start it on bootup if
-there are non-comment lines in /etc/portttime.
-
-The lastlog and faillog commands will report the last time a user had
-a successful and failed login, respectively.
+---MD5 Encryption
 
-You may set per-user resource limits by editing /etc/limits.  See
-limits(5).
+This is enabled now using the /etc/pam.d/* files. Examples are given.
 
 
 ---Adding users and groups
@@ -66,3 +51,6 @@
 Password HOWTO (included in the doc-linux package).  A large portion
 of these files deals with getting shadow installed.  You can, of
 course, ignore those parts.
+
+Also, the libpam-doc package will go a long way to allowing you to take
+full advantage of the PAM authentication scheme.
