--- shadow-20000902/libmisc/sulog.c~	Mon Oct 23 11:29:47 2000
+++ shadow-20000902/libmisc/sulog.c	Mon Oct 23 11:30:15 2000
@@ -65,7 +65,7 @@
 	time(&now);
 	tm = localtime(&now);
 
-	fprintf(fp, "SU %.02d/%.02d %.02d:%.02d %c %.6s %s-%s\n",
+	fprintf(fp, "SU %.02d/%.02d %.02d:%.02d %c %s %s-%s\n",
 		tm->tm_mon + 1, tm->tm_mday, tm->tm_hour, tm->tm_min,
 		success ? '+':'-', tty, oldname, name);
 
