diff -urN shadow-20000902.orig/src/su.c shadow-20000902/src/su.c
--- shadow-20000902.orig/src/su.c	Sat Mar  3 00:07:59 2001
+++ shadow-20000902/src/su.c	Sat Mar  3 00:18:56 2001
@@ -374,8 +374,10 @@
 
 	if (optind < argc)
 		strncpy(name, argv[optind++], sizeof(name) - 1);
-	else
-		strcpy(name, "root");
+	else {
+		struct passwd *root_pw = getpwuid(0);
+		strcpy(name, root_pw->pw_name);
+	}
 
 	if (optind < argc)
 		additional_args = argv + optind;
