diff -urN shadow-20000902/src/su.c shadow-20000902.new/src/su.c
--- shadow-20000902/src/su.c	Wed Aug 22 11:15:15 2001
+++ shadow-20000902.new/src/su.c	Wed Aug 22 11:14:21 2001
@@ -751,6 +751,7 @@
 #if 1  /* Suggested by Joey Hess.  XXX - is this right?  */
 	else if (change_environment || restricted_shell(pwent.pw_shell)) {
 		addenv("HOME", pwent.pw_dir);
+		addenv("USER", pwent.pw_name);
 		addenv("SHELL", shell);
 	}
 #endif
