diff -urN shadow-20000902/src/grpck.c shadow-20000902.new/src/grpck.c
--- shadow-20000902/src/grpck.c	Sat Sep  2 14:40:44 2000
+++ shadow-20000902.new/src/grpck.c	Wed Aug 22 12:30:33 2001
@@ -159,6 +159,7 @@
 	int	errors = 0;
 	int	deleted = 0;
 	int	i;
+	int	prune = 0;
 	struct	commonio_entry	*gre, *tgre;
 	struct	group	*grp;
 #ifdef	SHADOWGRP
@@ -183,7 +184,7 @@
 	 * Parse the command line arguments
 	 */
 
-	while ((arg = getopt(argc, argv, "qr")) != EOF) {
+	while ((arg = getopt(argc, argv, "qpr")) != EOF) {
 		switch (arg) {
 		case 'q':
 			/* quiet - ignored for now */
@@ -191,6 +192,9 @@
 		case 'r':
 			read_only = 1;
 			break;
+		case 'p':
+			prune = 1;
+			break;
 		default:
 			usage();
 		}
@@ -532,6 +536,10 @@
 		if (!gr_locate(sgr->sg_name)) {
 			puts(_("no matching group file entry\n"));
 			printf(_("delete line `%s'? "), sge->line);
+			if (prune) {
+				printf("Yes\n");
+				goto delete_sg;
+			}
 			errors++;
 			if (yes_or_no())
 				goto delete_sg;
diff -urN shadow-20000902/src/userdel.c shadow-20000902.new/src/userdel.c
--- shadow-20000902/src/userdel.c	Sat Sep  2 14:40:44 2000
+++ shadow-20000902.new/src/userdel.c	Wed Aug 22 12:31:08 2001
@@ -145,6 +145,7 @@
 	const struct group *grp;
 	struct group *ngrp;
 #ifdef	SHADOWGRP
+	int deleted_user_group = 0;
 	const struct sgrp *sgrp;
 	struct sgrp *nsgrp;
 #endif	/* SHADOWGRP */
@@ -205,6 +206,10 @@
 
 		gr_remove(grp->gr_name);
 
+#ifdef SHADOWGRP
+		deleted_user_group = 1;
+#endif
+
 		/*
 		 * Update the DBM group file with the new entry as well.
 		 */
@@ -272,6 +277,10 @@
 		SYSLOG((LOG_INFO, "delete `%s' from shadow group `%s'\n",
 			user_name, nsgrp->sg_name));
 	}
+
+	if (deleted_user_group)
+		sgr_remove(user_name);
+
 #ifdef	NDBM
 	endsgent ();
 #endif	/* NDBM */
