diff -urN shadow-20000902/src/shadowconfig.sh shadow-20000902.new/src/shadowconfig.sh
--- shadow-20000902/src/shadowconfig.sh	Wed Aug 22 12:03:27 2001
+++ shadow-20000902.new/src/shadowconfig.sh	Wed Aug 22 12:03:15 2001
@@ -7,7 +7,7 @@
 bash<<- EOF
     set -e
     pwck -q
-    grpck
+    grpck -p
     pwconv
     grpconv
     cd /etc
@@ -22,7 +22,7 @@
 bash<<- EOF
     set -e
     pwck -q
-    grpck
+    grpck -p
     pwunconv
     grpunconv
     cd /etc
