diff -urN shadow-20000902/etc/login.defs.hurd shadow-20000902.new/etc/login.defs.hurd
--- shadow-20000902/etc/login.defs.hurd	Sat Aug 26 14:27:10 2000
+++ shadow-20000902.new/etc/login.defs.hurd	Sun Oct 21 14:43:48 2001
@@ -85,7 +85,7 @@
 # phone, home phone).  If not defined, no changes are allowed.
 # For backward compatibility, "yes" = "rwh" and "no" = "frwh".
 #
-CHFN_RESTRICT		rwh
+CHFN_RESTRICT		frwh
 
 #
 # Password prompt (%s will be replaced by user name).
diff -urN shadow-20000902/etc/login.defs.linux shadow-20000902.new/etc/login.defs.linux
--- shadow-20000902/etc/login.defs.linux	Sat Aug 26 14:27:10 2000
+++ shadow-20000902.new/etc/login.defs.linux	Sun Oct 21 14:43:36 2001
@@ -282,7 +282,7 @@
 # phone, home phone).  If not defined, no changes are allowed.
 # For backward compatibility, "yes" = "rwh" and "no" = "frwh".
 # 
-CHFN_RESTRICT		rwh
+CHFN_RESTRICT		frwh
 
 #
 # Password prompt (%s will be replaced by user name).
