diff -urN shadow-20000902/lib/commonio.c shadow-20000902.new/lib/commonio.c
--- shadow-20000902/lib/commonio.c	Sat Sep  2 14:40:42 2000
+++ shadow-20000902.new/lib/commonio.c	Sun Oct 21 14:54:48 2001
@@ -469,13 +469,15 @@
 		goto cleanup;
 
 	while (db->ops->fgets(buf, buflen, db->fp)) {
+		int rlen;
 		while (!(cp = strrchr(buf, '\n')) && !feof(db->fp)) {
 			buflen += BUFLEN;
 			cp = (char *) realloc(buf, buflen);
 			if (!cp)
 				goto cleanup_buf;
 			buf = cp;
-			db->ops->fgets(buf + buflen - BUFLEN, BUFLEN, db->fp);
+			rlen = strlen(buf);
+			db->ops->fgets(buf + rlen, BUFLEN, db->fp);
 		}
 		if ((cp = strrchr(buf, '\n')))
 			*cp = '\0';
