diff -urN shadow-20000902/configure shadow-20000902.new/configure
--- shadow-20000902/configure	Fri Sep  1 17:07:50 2000
+++ shadow-20000902.new/configure	Sun Oct 21 15:13:08 2001
@@ -3663,7 +3663,7 @@
 
 echo $ac_n "checking location of shared mail directory""... $ac_c" 1>&6
 echo "configure:3666: checking location of shared mail directory" >&5
-for maildir in /var/spool/mail /var/mail /usr/spool/mail /usr/mail NONE; do
+for maildir in /var/mail /var/spool/mail /usr/spool/mail /usr/mail NONE; do
 	if test "$maildir" = "NONE"; then
 		echo "$ac_t""None" 1>&6
 	elif test -d $maildir; then
diff -urN shadow-20000902/configure.in shadow-20000902.new/configure.in
--- shadow-20000902/configure.in	Fri Sep  1 15:02:19 2000
+++ shadow-20000902.new/configure.in	Sun Oct 21 15:13:41 2001
@@ -141,7 +141,7 @@
 fi
 
 AC_MSG_CHECKING(location of shared mail directory)
-for maildir in /var/spool/mail /var/mail /usr/spool/mail /usr/mail NONE; do
+for maildir in /var/mail /var/spool/mail /usr/spool/mail /usr/mail NONE; do
 	if test "$maildir" = "NONE"; then
 		AC_MSG_RESULT(None)
 	elif test -d $maildir; then
