Description: Fix XSS in CUPS web interface
Author: Michael Sweet <msweet@apple.com>
Bug-CVE: https://security-tracker.debian.org/tracker/CVE-2014-2856
Bug: http://www.cups.org/str.php?L4356
Last-Update: 2014-02-19
--- a/scheduler/client.c
+++ b/scheduler/client.c
@@ -4075,6 +4075,14 @@
     return (0);
 
  /*
+  * Check for "<" or quotes in the path and reject since this is probably
+  * someone trying to inject HTML...
+  */
+
+  if (strchr(path, '<') != NULL || strchr(path, '\"') != NULL || strchr(path, '\'') != NULL)
+    return (0);
+
+ /*
   * Check for "/.." in the path...
   */
 
