Description: Fix Japanese and Russian templates translations typos
Bug-Debian: http://bugs.debian.org/698521
Author: Didier Raboud <odyx@debian.org>
Last-Update: 2013-01-20

--- a/templates/ja/help-header.tmpl
+++ b/templates/ja/help-header.tmpl
@@ -3,7 +3,7 @@
 {TOPIC?<INPUT TYPE="HIDDEN" NAME="TOPIC" VALUE="{TOPIC}">:}
 
 <P ALIGN="CENTER"><B>
-{HELPTITLE?{HELPTITLE}:{TOPIC?{TOPIC}:すべてのドキュメント}内を検索:</B> <INPUT
+{HELPTITLE?{HELPTITLE}:{TOPIC?{TOPIC}:すべてのドキュメント}内を検索}:</B> <INPUT
 TYPE="SEARCH" NAME="QUERY" VALUE="{?QUERY}" SIZE="40" PLACEHOLDER=""
 AUTOSAVE="org.cups.help" RESULTS="20">
 <INPUT TYPE="SUBMIT" NAME="SEARCH" VALUE="検索">
--- a/templates/ru/classes-header.tmpl
+++ b/templates/ru/classes-header.tmpl
@@ -1 +1 @@
-<P ALIGN="CENTER">{total=0?Нет групп:Группа {#printer_name} из {total}.</P>
+<P ALIGN="CENTER">{total=0?Нет групп:Группа {#printer_name} из {total}}.</P>
--- a/templates/ru/jobs-header.tmpl
+++ b/templates/ru/jobs-header.tmpl
@@ -2,5 +2,5 @@
 {?which_jobs=completed?:<FORM ACTION="{?printer_name=?/jobs:{printer_uri_supported}}" METHOD="GET"><INPUT TYPE="HIDDEN" NAME="which_jobs" VALUE="completed"><INPUT TYPE="SUBMIT" VALUE="Показать завершенные задания"></FORM>}
 {?which_jobs=all?:<FORM ACTION="{?printer_name=?/jobs:{printer_uri_supported}}" METHOD="GET"><INPUT TYPE="HIDDEN" NAME="which_jobs" VALUE="all"><INPUT TYPE="SUBMIT" VALUE="Показать все задания"></FORM>}</DIV>
 
-<P ALIGN="CENTER">{total=0?Нет заданий:Задание {#job_id} из {total} {?which_jobs=?активных:{which_jobs=all?:завершенных}}.</P>
+<P ALIGN="CENTER">{total=0?Нет заданий:Задание {#job_id} из {total} {?which_jobs=?активных:{which_jobs=all?:завершенных}}}.</P>
 
--- a/templates/ru/printers-header.tmpl
+++ b/templates/ru/printers-header.tmpl
@@ -1 +1 @@
-<P ALIGN="CENTER">{total=0?Нет принтеров:Принтер {#printer_name} из {total}.</P>
+<P ALIGN="CENTER">{total=0?Нет принтеров:Принтер {#printer_name} из {total}}.</P>
