From a13b1d96a9753e057713391a1a5e7d852d53bd6f Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Wed, 7 Dec 2016 16:31:54 -0800
Subject: [PATCH 85/85] hwmon: (it87) Add GPIO feature

---
 include/linux/gpio/driver.h |  2 ++
 1 file changed, 18 insertions(+)

diff --git a/include/linux/gpio/driver.h b/include/linux/gpio/driver.h
index d1baebf..0dd1718 100644
--- a/include/linux/gpio/driver.h
+++ b/include/linux/gpio/driver.h
@@ -160,6 +160,8 @@ struct gpio_chip {
 	 */
 	struct list_head pin_ranges;
 #endif
+	/* Generic API for very chip-specific operations. */
+	int (*control)(struct gpio_chip *gc, unsigned int cmd, void *arg);
 };
 
 extern const char *gpiochip_is_requested(struct gpio_chip *chip,
-- 
1.9.1

