From 1478b31329ba97e1e4a781386c9b0cfab8e3ec4d Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Wed, 21 Oct 2015 11:48:31 -0700
Subject: [PATCH] gpio: relax non-fatal conflict checking of ICH sensor to
 enable GPIO.

---
 drivers/mfd/lpc_ich.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/mfd/lpc_ich.c b/drivers/mfd/lpc_ich.c
index b514f3c..2b5f4b4 100644
--- a/drivers/mfd/lpc_ich.c
+++ b/drivers/mfd/lpc_ich.c
@@ -945,7 +945,9 @@ gpe0_done:
 	if (ret < 0) {
 		/* this isn't necessarily fatal for the GPIO */
 		acpi_conflict = true;
+#if 0	/* Not fatal - keep GPIO alive. */
 		goto gpio_done;
+#endif
 	}
 	lpc_chipset_info[priv->chipset].use_gpio = ret;
 	lpc_ich_enable_gpio_space(dev);
-- 
1.9.1

