From cdfaebea5944d51cc097f8138dcd7a549cc17a4f Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Mon, 29 Jun 2015 18:16:46 -0700
Subject: [PATCH 121/121] armada: Support U-Boot adjusted clock settings.

---
 drivers/clk/mvebu/common.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/clk/mvebu/common.c b/drivers/clk/mvebu/common.c
index 28aac67..e5cfb56 100644
--- a/drivers/clk/mvebu/common.c
+++ b/drivers/clk/mvebu/common.c
@@ -62,6 +62,8 @@ u32 kirkwood_fix_sscg_deviation(u32 system_clk)
 		goto out;
 	}
 
+	sscg_reg = readl(sscg_map + 0x3EC) & 0xFF;
+	system_clk -= (0x2F - sscg_reg) * 25000000;
 	sscg_reg = readl(sscg_map);
 	high_bound = SSCG_CONF_HIGH(sscg_reg);
 	low_bound = SSCG_CONF_LOW(sscg_reg);
-- 
1.9.1

