From e7f21d6a765182b4be3fe4589ebe6a598ce68f07 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Wed, 11 Jan 2017 15:58:37 -0800
Subject: [PATCH 130/131] gpio: it87: Add IT8613 to supported chip list

---
 drivers/gpio/gpio-it87.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpio/gpio-it87.c b/drivers/gpio/gpio-it87.c
index 21f6f7c..1248168 100644
--- a/drivers/gpio/gpio-it87.c
+++ b/drivers/gpio/gpio-it87.c
@@ -34,6 +34,7 @@
 
 /* Chip Id numbers */
 #define NO_DEV_ID	0xffff
+#define IT8613_ID	0x8613
 #define IT8728_ID	0x8728
 #define IT8732_ID	0x8732
 #define IT8761_ID	0x8761
@@ -307,6 +308,7 @@ static int __init it87_gpio_init(void)
 	it87_gpio->chip = it87_template_chip;
 
 	switch (chip_type) {
+	case IT8613_ID:
 	case IT8728_ID:
 	case IT8732_ID:
 		gpio_ba_reg = 0x62;
-- 
1.9.1

