From ee3f447d7de0854d66bcd89c516e40b8c261b599 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Tue, 18 Aug 2015 11:25:27 -0700
Subject: [PATCH 1003/1016] btrfs: Make remounts quieter

---
 fs/btrfs/super.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/fs/btrfs/super.c b/fs/btrfs/super.c
index 335c688..40cda4b 100644
--- a/fs/btrfs/super.c
+++ b/fs/btrfs/super.c
@@ -791,8 +791,6 @@ int btrfs_parse_options(struct btrfs_root *root, char *options)
 		}
 	}
 out:
-	if (!ret && btrfs_test_opt(root, SPACE_CACHE))
-		btrfs_info(root->fs_info, "disk space caching is enabled");
 	kfree(orig);
 	return ret;
 }
-- 
1.9.1

