From 5c6ee7965f0d3f07e3896fd3ef48ea206f4e271c Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Wed, 10 Aug 2016 14:11:29 -0700
Subject: [PATCH 2/7] continue on local fs failure

---
 units/local-fs.target | 2 --
 1 file changed, 2 deletions(-)

diff --git a/units/local-fs.target b/units/local-fs.target
index d2e5429..f624033 100644
--- a/units/local-fs.target
+++ b/units/local-fs.target
@@ -11,5 +11,3 @@ Documentation=man:systemd.special(7)
 DefaultDependencies=no
 Conflicts=shutdown.target
 After=local-fs-pre.target
-OnFailure=emergency.target
-OnFailureJobMode=replace-irreversibly
-- 
2.9.2

