From 89c8906ee1a02c699e6c85a74f8cccd3d03c76a5 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Wed, 10 Aug 2016 14:44:53 -0700
Subject: [PATCH 6/7] ignore power button by default

---
 src/login/logind.conf.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/login/logind.conf.in b/src/login/logind.conf.in
index 6095e48..dba5f32 100644
--- a/src/login/logind.conf.in
+++ b/src/login/logind.conf.in
@@ -18,7 +18,7 @@
 #KillOnlyUsers=
 #KillExcludeUsers=root
 #InhibitDelayMaxSec=5
-#HandlePowerKey=poweroff
+HandlePowerKey=ignore
 #HandleSuspendKey=suspend
 #HandleHibernateKey=hibernate
 #HandleLidSwitch=suspend
-- 
2.9.2

