From cb45ec6b21b31248ad1f4a227de160ba93531ba1 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard@netgear.com>
Date: Thu, 22 Sep 2016 10:44:38 -0700
Subject: [PATCH] Extend mount timeout to 10 minutes (from 90 seconds).

---
 src/core/mount.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/mount.c b/src/core/mount.c
index 2fc9709..fb09039 100644
--- a/src/core/mount.c
+++ b/src/core/mount.c
@@ -125,7 +125,7 @@ static void mount_init(Unit *u) {
         assert(u);
         assert(u->load_state == UNIT_STUB);
 
-        m->timeout_usec = u->manager->default_timeout_start_usec;
+        m->timeout_usec = 600 * USEC_PER_SEC;
         m->directory_mode = 0755;
 
         if (unit_has_name(u, "-.mount")) {
-- 
2.9.3

