From 59c2a3d45829a2f68be088d2fa90c51ed2a1de6a Mon Sep 17 00:00:00 2001
From: Justin Ding <justin.ding@netgear.com>
Date: Wed, 31 Oct 2018 00:54:27 -0700
Subject: [PATCH] Extend mount timeout to 30 minutes (from 10 minutes)

---
 src/core/mount.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/mount.c b/src/core/mount.c
index f485243..9785fcc 100644
--- a/src/core/mount.c
+++ b/src/core/mount.c
@@ -161,7 +161,7 @@ static void mount_init(Unit *u) {
         assert(u);
         assert(u->load_state == UNIT_STUB);
 
-        m->timeout_usec = 600 * USEC_PER_SEC;
+        m->timeout_usec = 1800 * USEC_PER_SEC;
         m->directory_mode = 0755;
 
         if (unit_has_name(u, "-.mount")) {
-- 
2.1.4

