--- nut-2.6.5/drivers/richcomm_usb.c    2012-07-31 19:38:59.000000000 +0200
+++ nut-2.6.5-patched/drivers/richcomm_usb.c    2014-04-17 13:03:38.985900814 +0200
@@ -507,4 +507,6 @@

 void upsdrv_makevartable(void)
 {
+	addvar(VAR_VALUE, "vendorid", "Regular expression to match UPS Manufacturer numerical ID (4 digits hexadecimal)");
+	addvar(VAR_VALUE, "productid", "Regular expression to match UPS Product numerical ID (4 digits hexadecimal)");
 }
