Index: libapache2-mod-perl2/lib/Apache2/Build.pm
===================================================================
--- libapache2-mod-perl2.orig/lib/Apache2/Build.pm	2006-11-20 01:31:40.000000000 +0200
+++ libapache2-mod-perl2/lib/Apache2/Build.pm	2007-08-21 13:33:39.000000000 +0300
@@ -449,7 +449,7 @@
         $xlibs .= ' -lintl';
     }
 
-    return " -lgtop -lgtop_sysdeps -lgtop_common $xlibs";
+    return " -lgtop-2.0 -lgtop_sysdeps-2.0 -lgtop_common-2.0 $xlibs";
 }
 
 sub gtop_ldopts {
