Index: libapache2-mod-perl2/lib/ModPerl/TestReport.pm
===================================================================
--- libapache2-mod-perl2.orig/lib/ModPerl/TestReport.pm	2007-08-21 13:16:40.000000000 +0300
+++ libapache2-mod-perl2/lib/ModPerl/TestReport.pm	2007-08-21 13:35:30.000000000 +0300
@@ -97,7 +97,11 @@
     return $package;
 }
 
-sub report_to { 'modperl@perl.apache.org' }
-
+# We are reporting to the Debian BTS, not directly to the upstream (Apache)
+sub report_to { 'submit@bugs.debian.org' }
+sub postit_note {
+    # No, no postit notes required :) This report will be typically
+    # embedded in reportbug's, so...
+}
 
 1;
