Index: libapache2-mod-perl2/lib/ModPerl/BuildMM.pm
===================================================================
--- libapache2-mod-perl2.orig/lib/ModPerl/BuildMM.pm	2007-08-21 13:16:41.000000000 +0300
+++ libapache2-mod-perl2/lib/ModPerl/BuildMM.pm	2007-08-21 13:39:47.000000000 +0300
@@ -281,7 +281,7 @@
                 $man =~ s!/!::!g;
 
                 push @target,
-                    '$(NOECHO) $(POD2MAN_EXE) --section=3 ' .
+                    '$(NOECHO) $(POD2MAN_EXE) --section=3pm ' .
                     "$podpath \$(INST_MAN3DIR)/$man.\$(MAN3EXT)"
             }
         }
