Index: libapache2-mod-perl2/t/apache/content_length_header.t
===================================================================
--- libapache2-mod-perl2.orig/t/apache/content_length_header.t	2007-08-18 15:55:45.000000000 +0300
+++ libapache2-mod-perl2/t/apache/content_length_header.t	2007-08-18 16:06:38.000000000 +0300
@@ -65,11 +65,12 @@
         my $res = $method->($uri);
 
         my $cl      = 0;
-        my $head_cl = undef;
+        # We are actually 2.2, so HEAD also gets the set content length
+        my $head_cl = 25;
            
         ok t_cmp $res->code, 200, "$method $uri code";
         ok t_cmp ($res->header('Content-Length'),
-                  $method eq 'GET' ? $cl : $head_cl,
+                  $method eq 'GET' ? length($res->content) : $head_cl,
                   "$method $uri C-L header");
         ok t_cmp $res->content, "", "$method $uri content";
     }
