--- nss-llmnr-0.10.orig/src/nss.c
+++ nss-llmnr-0.10/src/nss.c
@@ -283,7 +283,7 @@ enum nss_status _nss_llmnr_gethostbyname
     ipv6_func = af == AF_INET6 ? ipv6_callback : NULL;
 #endif
 
-    name_allowed = 1;
+    name_allowed = !strchr(name, '.');
 
     {
         if ((fd = llmnr_open_socket()) < 0) {
--- nss-llmnr-0.10.orig/src/query.c
+++ nss-llmnr-0.10/src/query.c
@@ -481,7 +481,7 @@ static int process_name_response(int fd,
 /*                     else  */
 #endif                        
 #ifndef NSS_IPV4_ONLY
-                    if (ipv6_func &&
+                    if (!done && ipv6_func &&
                         type == DNS_TYPE_AAAA &&
                         class == DNS_CLASS_IN &&
                         !domain_cmp(name, pname) &&
