Index: b/extensions/libxt_length.man
===================================================================
--- a/extensions/libxt_length.man	2009-11-28 12:55:48.104766607 -0500
+++ b/extensions/libxt_length.man	2009-11-28 12:58:39.258651778 -0500
@@ -1,5 +1,5 @@
 This module matches the length of the layer-3 payload (e.g. layer-4 packet)
-f a packet against a specific value
+of a packet against a specific value
 or range of values.
 .TP
 [\fB!\fP] \fB\-\-length\fP \fIlength\fP[\fB:\fP\fIlength\fP]
